// @aws-sdk/client-bedrock-agent/dist-cjs/index.js
'use strict';

var middlewareHostHeader = require('@aws-sdk/middleware-host-header');
var middlewareLogger = require('@aws-sdk/middleware-logger');
var middlewareRecursionDetection = require('@aws-sdk/middleware-recursion-detection');
var middlewareUserAgent = require('@aws-sdk/middleware-user-agent');
var configResolver = require('@smithy/config-resolver');
var core = require('@smithy/core');
var schema = require('@smithy/core/schema');
var middlewareContentLength = require('@smithy/middleware-content-length');
var middlewareEndpoint = require('@smithy/middleware-endpoint');
var middlewareRetry = require('@smithy/middleware-retry');
var smithyClient = require('@smithy/smithy-client');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var regionConfigResolver = require('@aws-sdk/region-config-resolver');
var protocolHttp = require('@smithy/protocol-http');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "bedrock",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(regionConfigResolver.getAwsRegionExtensionConfiguration(runtimeConfig), smithyClient.getDefaultExtensionConfiguration(runtimeConfig), protocolHttp.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, regionConfigResolver.resolveAwsRegionExtensionConfiguration(extensionConfiguration), smithyClient.resolveDefaultRuntimeConfig(extensionConfiguration), protocolHttp.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class BedrockAgentClient extends smithyClient.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = middlewareUserAgent.resolveUserAgentConfig(_config_1);
        const _config_3 = middlewareRetry.resolveRetryConfig(_config_2);
        const _config_4 = configResolver.resolveRegionConfig(_config_3);
        const _config_5 = middlewareHostHeader.resolveHostHeaderConfig(_config_4);
        const _config_6 = middlewareEndpoint.resolveEndpointConfig(_config_5);
        const _config_7 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(middlewareUserAgent.getUserAgentPlugin(this.config));
        this.middlewareStack.use(middlewareRetry.getRetryPlugin(this.config));
        this.middlewareStack.use(middlewareContentLength.getContentLengthPlugin(this.config));
        this.middlewareStack.use(middlewareHostHeader.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(middlewareLogger.getLoggerPlugin(this.config));
        this.middlewareStack.use(middlewareRecursionDetection.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultBedrockAgentHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class BedrockAgentServiceException extends smithyClient.ServiceException {
    constructor(options) {
        super(options);
        Object.setPrototypeOf(this, BedrockAgentServiceException.prototype);
    }
}

class AccessDeniedException extends BedrockAgentServiceException {
    name = "AccessDeniedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "AccessDeniedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AccessDeniedException.prototype);
    }
}
class ConflictException extends BedrockAgentServiceException {
    name = "ConflictException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictException.prototype);
    }
}
class InternalServerException extends BedrockAgentServiceException {
    name = "InternalServerException";
    $fault = "server";
    constructor(opts) {
        super({
            name: "InternalServerException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalServerException.prototype);
    }
}
class ResourceNotFoundException extends BedrockAgentServiceException {
    name = "ResourceNotFoundException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
    }
}
class ServiceQuotaExceededException extends BedrockAgentServiceException {
    name = "ServiceQuotaExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ServiceQuotaExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ServiceQuotaExceededException.prototype);
    }
}
class ThrottlingException extends BedrockAgentServiceException {
    name = "ThrottlingException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ThrottlingException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ThrottlingException.prototype);
    }
}
class ValidationException extends BedrockAgentServiceException {
    name = "ValidationException";
    $fault = "client";
    fieldList;
    constructor(opts) {
        super({
            name: "ValidationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ValidationException.prototype);
        this.fieldList = opts.fieldList;
    }
}

const _A = "Agent";
const _AA = "AgentAlias";
const _AAC = "AssociateAgentCollaborator";
const _AACR = "AssociateAgentCollaboratorRequest";
const _AACRs = "AssociateAgentCollaboratorResponse";
const _AAG = "AgentActionGroup";
const _AAHE = "AgentAliasHistoryEvent";
const _AAHEg = "AgentAliasHistoryEvents";
const _AAKB = "AssociateAgentKnowledgeBase";
const _AAKBR = "AssociateAgentKnowledgeBaseRequest";
const _AAKBRs = "AssociateAgentKnowledgeBaseResponse";
const _AARC = "AgentAliasRoutingConfiguration";
const _AARCLI = "AgentAliasRoutingConfigurationListItem";
const _AAS = "AgentAliasSummary";
const _AASg = "AgentAliasSummaries";
const _AC = "AgentCollaborator";
const _ACS = "AgentCollaboratorSummary";
const _ACSg = "AgentCollaboratorSummaries";
const _ACu = "AudioConfiguration";
const _ACud = "AudioConfigurations";
const _AD = "AgentDescriptor";
const _ADE = "AccessDeniedException";
const _AFNC = "AgentFlowNodeConfiguration";
const _AGE = "ActionGroupExecutor";
const _AGS = "ActionGroupSummary";
const _AGSc = "ActionGroupSummaries";
const _AKB = "AgentKnowledgeBase";
const _AKBS = "AgentKnowledgeBaseSummary";
const _AKBSg = "AgentKnowledgeBaseSummaries";
const _APIS = "APISchema";
const _AS = "AgentSummary";
const _ASC = "AudioSegmentationConfiguration";
const _ASg = "AgentSummaries";
const _ATC = "AnyToolChoice";
const _ATCu = "AutoToolChoice";
const _AV = "AgentVersion";
const _AVS = "AgentVersionSummary";
const _AVSg = "AgentVersionSummaries";
const _BCB = "ByteContentBlob";
const _BCD = "ByteContentDoc";
const _BDAC = "BedrockDataAutomationConfiguration";
const _BEMC = "BedrockEmbeddingModelConfiguration";
const _BFMC = "BedrockFoundationModelConfiguration";
const _BFMCEC = "BedrockFoundationModelContextEnrichmentConfiguration";
const _BPT = "BasePromptTemplate";
const _CA = "CreateAgent";
const _CAA = "CreateAgentAlias";
const _CAAG = "CreateAgentActionGroup";
const _CAAGR = "CreateAgentActionGroupRequest";
const _CAAGRr = "CreateAgentActionGroupResponse";
const _CAAR = "CreateAgentAliasRequest";
const _CAARr = "CreateAgentAliasResponse";
const _CAR = "CreateAgentRequest";
const _CARr = "CreateAgentResponse";
const _CB = "ContentBlocks";
const _CBo = "ContentBlock";
const _CC = "ChunkingConfiguration";
const _CCC = "ConfluenceCrawlerConfiguration";
const _CCFVD = "CyclicConnectionFlowValidationDetails";
const _CCu = "CustomContent";
const _CDI = "CustomDocumentIdentifier";
const _CDS = "CreateDataSource";
const _CDSC = "ConfluenceDataSourceConfiguration";
const _CDSR = "CreateDataSourceRequest";
const _CDSRr = "CreateDataSourceResponse";
const _CE = "ConflictException";
const _CEC = "ContextEnrichmentConfiguration";
const _CF = "CreateFlow";
const _CFA = "CreateFlowAlias";
const _CFAR = "CreateFlowAliasRequest";
const _CFARr = "CreateFlowAliasResponse";
const _CFC = "CrawlFilterConfiguration";
const _CFNC = "CollectorFlowNodeConfiguration";
const _CFNCo = "ConditionFlowNodeConfiguration";
const _CFR = "CreateFlowRequest";
const _CFRr = "CreateFlowResponse";
const _CFV = "CreateFlowVersion";
const _CFVR = "CreateFlowVersionRequest";
const _CFVRr = "CreateFlowVersionResponse";
const _CI = "CollaborationInstruction";
const _CKB = "CreateKnowledgeBase";
const _CKBR = "CreateKnowledgeBaseRequest";
const _CKBRr = "CreateKnowledgeBaseResponse";
const _CO = "CustomOrchestration";
const _CP = "CreatePrompt";
const _CPB = "CachePointBlock";
const _CPR = "CreatePromptRequest";
const _CPRr = "CreatePromptResponse";
const _CPTC = "ChatPromptTemplateConfiguration";
const _CPV = "CreatePromptVersion";
const _CPVR = "CreatePromptVersionRequest";
const _CPVRr = "CreatePromptVersionResponse";
const _CQ = "CuratedQuery";
const _CQu = "CuratedQueries";
const _CSC = "ConfluenceSourceConfiguration";
const _CSL = "CustomS3Location";
const _CTC = "CustomTransformationConfiguration";
const _D = "Data";
const _DA = "DeleteAgent";
const _DAA = "DeleteAgentAlias";
const _DAAG = "DeleteAgentActionGroup";
const _DAAGR = "DeleteAgentActionGroupRequest";
const _DAAGRe = "DeleteAgentActionGroupResponse";
const _DAAR = "DeleteAgentAliasRequest";
const _DAARe = "DeleteAgentAliasResponse";
const _DAC = "DisassociateAgentCollaborator";
const _DACR = "DisassociateAgentCollaboratorRequest";
const _DACRi = "DisassociateAgentCollaboratorResponse";
const _DAKB = "DisassociateAgentKnowledgeBase";
const _DAKBR = "DisassociateAgentKnowledgeBaseRequest";
const _DAKBRi = "DisassociateAgentKnowledgeBaseResponse";
const _DAR = "DeleteAgentRequest";
const _DARe = "DeleteAgentResponse";
const _DAV = "DeleteAgentVersion";
const _DAVR = "DeleteAgentVersionRequest";
const _DAVRe = "DeleteAgentVersionResponse";
const _DC = "DocumentContent";
const _DCEFVD = "DuplicateConditionExpressionFlowValidationDetails";
const _DCFVD = "DuplicateConnectionsFlowValidationDetails";
const _DDS = "DeleteDataSource";
const _DDSR = "DeleteDataSourceRequest";
const _DDSRe = "DeleteDataSourceResponse";
const _DF = "DeleteFlow";
const _DFA = "DeleteFlowAlias";
const _DFAR = "DeleteFlowAliasRequest";
const _DFARe = "DeleteFlowAliasResponse";
const _DFR = "DeleteFlowRequest";
const _DFRe = "DeleteFlowResponse";
const _DFV = "DeleteFlowVersion";
const _DFVR = "DeleteFlowVersionRequest";
const _DFVRe = "DeleteFlowVersionResponse";
const _DI = "DocumentIdentifier";
const _DIo = "DocumentIdentifiers";
const _DKB = "DeleteKnowledgeBase";
const _DKBD = "DeleteKnowledgeBaseDocuments";
const _DKBDR = "DeleteKnowledgeBaseDocumentsRequest";
const _DKBDRe = "DeleteKnowledgeBaseDocumentsResponse";
const _DKBR = "DeleteKnowledgeBaseRequest";
const _DKBRe = "DeleteKnowledgeBaseResponse";
const _DM = "DocumentMetadata";
const _DP = "DeletePrompt";
const _DPR = "DeletePromptRequest";
const _DPRe = "DeletePromptResponse";
const _DS = "DataSource";
const _DSC = "DataSourceConfiguration";
const _DSS = "DataSourceSummary";
const _DSSa = "DataSourceSummaries";
const _EMC = "EmbeddingModelConfiguration";
const _ESC = "EnrichmentStrategyConfiguration";
const _F = "Function";
const _FACC = "FlowAliasConcurrencyConfiguration";
const _FARC = "FlowAliasRoutingConfiguration";
const _FARCLI = "FlowAliasRoutingConfigurationListItem";
const _FAS = "FlowAliasSummary";
const _FASl = "FlowAliasSummaries";
const _FC = "FlowCondition";
const _FCC = "FlowConnectionConfiguration";
const _FCCC = "FlowConditionalConnectionConfiguration";
const _FCE = "FlowConditionExpression";
const _FCl = "FlowConnection";
const _FClo = "FlowConditions";
const _FClow = "FlowConnections";
const _FD = "FlowDefinition";
const _FDCC = "FlowDataConnectionConfiguration";
const _FFR = "FieldForReranking";
const _FFRi = "FieldsForReranking";
const _FL = "FilterList";
const _FN = "FlowNode";
const _FNC = "FlowNodeConfiguration";
const _FNI = "FlowNodeInput";
const _FNIE = "FlowNodeInputExpression";
const _FNIl = "FlowNodeInputs";
const _FNO = "FlowNodeOutput";
const _FNOl = "FlowNodeOutputs";
const _FNl = "FlowNodes";
const _FOT = "FilteredObjectType";
const _FP = "FilterPattern";
const _FS = "FlowSummary";
const _FSCC = "FixedSizeChunkingConfiguration";
const _FSl = "FlowSummaries";
const _FSu = "FunctionSchema";
const _FV = "FlowValidation";
const _FVD = "FlowValidationDetails";
const _FVS = "FlowVersionSummary";
const _FVSl = "FlowVersionSummaries";
const _FVl = "FlowValidations";
const _Fu = "Functions";
const _GA = "GraphArn";
const _GAA = "GetAgentAlias";
const _GAAG = "GetAgentActionGroup";
const _GAAGR = "GetAgentActionGroupRequest";
const _GAAGRe = "GetAgentActionGroupResponse";
const _GAAR = "GetAgentAliasRequest";
const _GAARe = "GetAgentAliasResponse";
const _GAC = "GetAgentCollaborator";
const _GACR = "GetAgentCollaboratorRequest";
const _GACRe = "GetAgentCollaboratorResponse";
const _GAKB = "GetAgentKnowledgeBase";
const _GAKBR = "GetAgentKnowledgeBaseRequest";
const _GAKBRe = "GetAgentKnowledgeBaseResponse";
const _GAR = "GetAgentRequest";
const _GARe = "GetAgentResponse";
const _GAV = "GetAgentVersion";
const _GAVR = "GetAgentVersionRequest";
const _GAVRe = "GetAgentVersionResponse";
const _GAe = "GetAgent";
const _GC = "GuardrailConfiguration";
const _GDS = "GetDataSource";
const _GDSR = "GetDataSourceRequest";
const _GDSRe = "GetDataSourceResponse";
const _GF = "GetFlow";
const _GFA = "GetFlowAlias";
const _GFAR = "GetFlowAliasRequest";
const _GFARe = "GetFlowAliasResponse";
const _GFR = "GetFlowRequest";
const _GFRe = "GetFlowResponse";
const _GFV = "GetFlowVersion";
const _GFVR = "GetFlowVersionRequest";
const _GFVRe = "GetFlowVersionResponse";
const _GIJ = "GetIngestionJob";
const _GIJR = "GetIngestionJobRequest";
const _GIJRe = "GetIngestionJobResponse";
const _GKB = "GetKnowledgeBase";
const _GKBD = "GetKnowledgeBaseDocuments";
const _GKBDR = "GetKnowledgeBaseDocumentsRequest";
const _GKBDRe = "GetKnowledgeBaseDocumentsResponse";
const _GKBR = "GetKnowledgeBaseRequest";
const _GKBRe = "GetKnowledgeBaseResponse";
const _GP = "GetPrompt";
const _GPR = "GetPromptRequest";
const _GPRe = "GetPromptResponse";
const _HCC = "HierarchicalChunkingConfiguration";
const _HCLC = "HierarchicalChunkingLevelConfiguration";
const _HCLCi = "HierarchicalChunkingLevelConfigurations";
const _I = "Instruction";
const _IA = "IndexArn";
const _IC = "InlineCode";
const _ICDTFVD = "IncompatibleConnectionDataTypeFlowValidationDetails";
const _ICFNC = "InlineCodeFlowNodeConfiguration";
const _ICn = "InferenceConfiguration";
const _ICnl = "InlineContent";
const _IFNC = "InputFlowNodeConfiguration";
const _IFNCt = "IteratorFlowNodeConfiguration";
const _IJ = "IngestionJob";
const _IJF = "IngestionJobFilter";
const _IJFn = "IngestionJobFilters";
const _IJS = "IngestionJobStatistics";
const _IJSB = "IngestionJobSortBy";
const _IJSn = "IngestionJobSummary";
const _IJSng = "IngestionJobSummaries";
const _IKBD = "IngestKnowledgeBaseDocuments";
const _IKBDR = "IngestKnowledgeBaseDocumentsRequest";
const _IKBDRn = "IngestKnowledgeBaseDocumentsResponse";
const _ILBFVD = "InvalidLoopBoundaryFlowValidationDetails";
const _IN = "IndexName";
const _IS = "IntermediateStorage";
const _ISE = "InternalServerException";
const _K = "Key";
const _KB = "KnowledgeBase";
const _KBC = "KnowledgeBaseConfiguration";
const _KBD = "KnowledgeBaseDocument";
const _KBDD = "KnowledgeBaseDocumentDetail";
const _KBDDn = "KnowledgeBaseDocumentDetails";
const _KBDn = "KnowledgeBaseDocuments";
const _KBFNC = "KnowledgeBaseFlowNodeConfiguration";
const _KBOC = "KnowledgeBaseOrchestrationConfiguration";
const _KBPT = "KnowledgeBasePromptTemplate";
const _KBS = "KnowledgeBaseSummary";
const _KBSn = "KnowledgeBaseSummaries";
const _KBTP = "KnowledgeBaseTextPrompt";
const _KKBC = "KendraKnowledgeBaseConfiguration";
const _LA = "ListAgents";
const _LAA = "ListAgentAliases";
const _LAAG = "ListAgentActionGroups";
const _LAAGR = "ListAgentActionGroupsRequest";
const _LAAGRi = "ListAgentActionGroupsResponse";
const _LAAR = "ListAgentAliasesRequest";
const _LAARi = "ListAgentAliasesResponse";
const _LAC = "ListAgentCollaborators";
const _LACR = "ListAgentCollaboratorsRequest";
const _LACRi = "ListAgentCollaboratorsResponse";
const _LAKB = "ListAgentKnowledgeBases";
const _LAKBR = "ListAgentKnowledgeBasesRequest";
const _LAKBRi = "ListAgentKnowledgeBasesResponse";
const _LAR = "ListAgentsRequest";
const _LARi = "ListAgentsResponse";
const _LAV = "ListAgentVersions";
const _LAVR = "ListAgentVersionsRequest";
const _LAVRi = "ListAgentVersionsResponse";
const _LCFNC = "LoopControllerFlowNodeConfiguration";
const _LDS = "ListDataSources";
const _LDSR = "ListDataSourcesRequest";
const _LDSRi = "ListDataSourcesResponse";
const _LF = "ListFlows";
const _LFA = "ListFlowAliases";
const _LFAR = "ListFlowAliasesRequest";
const _LFARi = "ListFlowAliasesResponse";
const _LFFNC = "LambdaFunctionFlowNodeConfiguration";
const _LFNC = "LexFlowNodeConfiguration";
const _LFNCo = "LoopFlowNodeConfiguration";
const _LFR = "ListFlowsRequest";
const _LFRi = "ListFlowsResponse";
const _LFV = "ListFlowVersions";
const _LFVR = "ListFlowVersionsRequest";
const _LFVRi = "ListFlowVersionsResponse";
const _LIFNC = "LoopInputFlowNodeConfiguration";
const _LIJ = "ListIngestionJobs";
const _LIJR = "ListIngestionJobsRequest";
const _LIJRi = "ListIngestionJobsResponse";
const _LINTFVD = "LoopIncompatibleNodeTypeFlowValidationDetails";
const _LKB = "ListKnowledgeBases";
const _LKBD = "ListKnowledgeBaseDocuments";
const _LKBDR = "ListKnowledgeBaseDocumentsRequest";
const _LKBDRi = "ListKnowledgeBaseDocumentsResponse";
const _LKBR = "ListKnowledgeBasesRequest";
const _LKBRi = "ListKnowledgeBasesResponse";
const _LP = "ListPrompts";
const _LPR = "ListPromptsRequest";
const _LPRi = "ListPromptsResponse";
const _LTFR = "ListTagsForResource";
const _LTFRR = "ListTagsForResourceRequest";
const _LTFRRi = "ListTagsForResourceResponse";
const _M = "Message";
const _MA = "MetadataAttribute";
const _MAV = "MetadataAttributeValue";
const _MAe = "MetadataAttributes";
const _MC = "MemoryConfiguration";
const _MCCFVD = "MissingConnectionConfigurationFlowValidationDetails";
const _MCEFVD = "MalformedConditionExpressionFlowValidationDetails";
const _MCFR = "MetadataConfigurationForReranking";
const _MDAC = "MongoDbAtlasConfiguration";
const _MDAFM = "MongoDbAtlasFieldMapping";
const _MDCFVD = "MissingDefaultConditionFlowValidationDetails";
const _MENFVD = "MissingEndingNodesFlowValidationDetails";
const _MLCNFVD = "MissingLoopControllerNodeFlowValidationDetails";
const _MLCNFVDu = "MultipleLoopControllerNodesFlowValidationDetails";
const _MLINFVD = "MissingLoopInputNodeFlowValidationDetails";
const _MLINFVDu = "MultipleLoopInputNodesFlowValidationDetails";
const _MNCFVD = "MissingNodeConfigurationFlowValidationDetails";
const _MNICFVD = "MultipleNodeInputConnectionsFlowValidationDetails";
const _MNIEFVD = "MalformedNodeInputExpressionFlowValidationDetails";
const _MNIFVD = "MissingNodeInputFlowValidationDetails";
const _MNITFVD = "MismatchedNodeInputTypeFlowValidationDetails";
const _MNOFVD = "MissingNodeOutputFlowValidationDetails";
const _MNOTFVD = "MismatchedNodeOutputTypeFlowValidationDetails";
const _MSNFVD = "MissingStartingNodesFlowValidationDetails";
const _Me = "Messages";
const _NAC = "NeptuneAnalyticsConfiguration";
const _NAFM = "NeptuneAnalyticsFieldMapping";
const _NV = "NumberValue";
const _OE = "OrchestrationExecutor";
const _OFNC = "OutputFlowNodeConfiguration";
const _OSMCC = "OpenSearchManagedClusterConfiguration";
const _OSMCFM = "OpenSearchManagedClusterFieldMapping";
const _OSMCIN = "OpenSearchManagedClusterIndexName";
const _OSSC = "OpenSearchServerlessConfiguration";
const _OSSFM = "OpenSearchServerlessFieldMapping";
const _P = "Payload";
const _PA = "PrepareAgent";
const _PAR = "PrepareAgentRequest";
const _PARr = "PrepareAgentResponse";
const _PARro = "PromptAgentResource";
const _PC = "ParsingConfiguration";
const _PCe = "PerformanceConfiguration";
const _PCi = "PineconeConfiguration";
const _PCr = "PromptConfiguration";
const _PCro = "PromptConfigurations";
const _PD = "ParameterDetail";
const _PF = "PrepareFlow";
const _PFM = "PineconeFieldMapping";
const _PFNC = "PromptFlowNodeConfiguration";
const _PFNIC = "PromptFlowNodeInlineConfiguration";
const _PFNRC = "PromptFlowNodeResourceConfiguration";
const _PFNSC = "PromptFlowNodeSourceConfiguration";
const _PFR = "PrepareFlowRequest";
const _PFRr = "PrepareFlowResponse";
const _PGAR = "PromptGenAiResource";
const _PIC = "PromptInferenceConfiguration";
const _PIV = "PromptInputVariable";
const _PIVL = "PromptInputVariablesList";
const _PM = "ParameterMap";
const _PME = "PromptMetadataEntry";
const _PMIC = "PromptModelInferenceConfiguration";
const _PMK = "PromptMetadataKey";
const _PML = "PromptMetadataList";
const _PMV = "PromptMetadataValue";
const _POC = "PromptOverrideConfiguration";
const _POF = "PatternObjectFilter";
const _POFC = "PatternObjectFilterConfiguration";
const _POFL = "PatternObjectFilterList";
const _PP = "ParsingPrompt";
const _PS = "PromptSummary";
const _PSr = "PromptSummaries";
const _PTC = "PromptTemplateConfiguration";
const _PV = "PromptVariant";
const _PVL = "PromptVariantList";
const _QGC = "QueryGenerationColumn";
const _QGCu = "QueryGenerationConfiguration";
const _QGCue = "QueryGenerationContext";
const _QGCuer = "QueryGenerationColumns";
const _QGT = "QueryGenerationTable";
const _QGTu = "QueryGenerationTables";
const _RC = "RdsConfiguration";
const _RCe = "RedshiftConfiguration";
const _RECC = "RedisEnterpriseCloudConfiguration";
const _RECFM = "RedisEnterpriseCloudFieldMapping";
const _RFM = "RdsFieldMapping";
const _RFNC = "RetrievalFlowNodeConfiguration";
const _RFNSC = "RetrievalFlowNodeS3Configuration";
const _RFNSCe = "RetrievalFlowNodeServiceConfiguration";
const _RMSMC = "RerankingMetadataSelectiveModeConfiguration";
const _RNFE = "ResourceNotFoundException";
const _RPAC = "RedshiftProvisionedAuthConfiguration";
const _RPC = "RedshiftProvisionedConfiguration";
const _RQEADCSC = "RedshiftQueryEngineAwsDataCatalogStorageConfiguration";
const _RQEC = "RedshiftQueryEngineConfiguration";
const _RQERSC = "RedshiftQueryEngineRedshiftStorageConfiguration";
const _RQESC = "RedshiftQueryEngineStorageConfiguration";
const _RQESCe = "RedshiftQueryEngineStorageConfigurations";
const _RSAC = "RedshiftServerlessAuthConfiguration";
const _RSC = "RedshiftServerlessConfiguration";
const _SC = "S3Content";
const _SCB = "SystemContentBlocks";
const _SCBy = "SystemContentBlock";
const _SCC = "SalesforceCrawlerConfiguration";
const _SCCe = "SemanticChunkingConfiguration";
const _SCt = "StorageConfiguration";
const _SDSC = "S3DataSourceConfiguration";
const _SDSCa = "SalesforceDataSourceConfiguration";
const _SDSCu = "SupplementalDataStorageConfiguration";
const _SDSL = "SupplementalDataStorageLocation";
const _SDSLu = "SupplementalDataStorageLocations";
const _SFNC = "StorageFlowNodeConfiguration";
const _SFNSC = "StorageFlowNodeS3Configuration";
const _SFNSCt = "StorageFlowNodeServiceConfiguration";
const _SI = "S3Identifier";
const _SIJ = "StartIngestionJob";
const _SIJR = "StartIngestionJobRequest";
const _SIJRt = "StartIngestionJobResponse";
const _SIJRto = "StopIngestionJobRequest";
const _SIJRtop = "StopIngestionJobResponse";
const _SIJt = "StopIngestionJob";
const _SKBC = "SqlKnowledgeBaseConfiguration";
const _SL = "S3Location";
const _SLV = "StringListValue";
const _SP = "S3Prefix";
const _SPCC = "SharePointCrawlerConfiguration";
const _SPDSC = "SharePointDataSourceConfiguration";
const _SPSC = "SharePointSourceConfiguration";
const _SPr = "S3Prefixes";
const _SQEE = "ServiceQuotaExceededException";
const _SSC = "SalesforceSourceConfiguration";
const _SSCe = "SessionSummaryConfiguration";
const _SSEC = "ServerSideEncryptionConfiguration";
const _STC = "SpecificToolChoice";
const _SU = "SeedUrl";
const _SUe = "SeedUrls";
const _SV = "StringValue";
const _SVC = "S3VectorsConfiguration";
const _T = "Transformation";
const _TC = "ToolConfiguration";
const _TCD = "TextContentDoc";
const _TCo = "ToolChoice";
const _TE = "ThrottlingException";
const _TF = "TransformationFunction";
const _TIS = "ToolInputSchema";
const _TLC = "TransformationLambdaConfiguration";
const _TP = "TextPrompt";
const _TPTC = "TextPromptTemplateConfiguration";
const _TR = "TagResource";
const _TRR = "TagResourceRequest";
const _TRRa = "TagResourceResponse";
const _TS = "ToolSpecification";
const _To = "Tools";
const _Too = "Tool";
const _Tr = "Transformations";
const _UA = "UserAgent";
const _UAA = "UpdateAgentAlias";
const _UAAG = "UpdateAgentActionGroup";
const _UAAGR = "UpdateAgentActionGroupRequest";
const _UAAGRp = "UpdateAgentActionGroupResponse";
const _UAAR = "UpdateAgentAliasRequest";
const _UAARp = "UpdateAgentAliasResponse";
const _UAC = "UpdateAgentCollaborator";
const _UACR = "UpdateAgentCollaboratorRequest";
const _UACRp = "UpdateAgentCollaboratorResponse";
const _UAH = "UserAgentHeader";
const _UAKB = "UpdateAgentKnowledgeBase";
const _UAKBR = "UpdateAgentKnowledgeBaseRequest";
const _UAKBRp = "UpdateAgentKnowledgeBaseResponse";
const _UAR = "UpdateAgentRequest";
const _UARp = "UpdateAgentResponse";
const _UAp = "UpdateAgent";
const _UC = "UrlConfiguration";
const _UCCFVD = "UnknownConnectionConditionFlowValidationDetails";
const _UCCFVDn = "UnsatisfiedConnectionConditionsFlowValidationDetails";
const _UCSFVD = "UnknownConnectionSourceFlowValidationDetails";
const _UCSOFVD = "UnknownConnectionSourceOutputFlowValidationDetails";
const _UCTFVD = "UnknownConnectionTargetFlowValidationDetails";
const _UCTIFVD = "UnknownConnectionTargetInputFlowValidationDetails";
const _UDS = "UpdateDataSource";
const _UDSR = "UpdateDataSourceRequest";
const _UDSRp = "UpdateDataSourceResponse";
const _UF = "UpdateFlow";
const _UFA = "UpdateFlowAlias";
const _UFAR = "UpdateFlowAliasRequest";
const _UFARp = "UpdateFlowAliasResponse";
const _UFR = "UpdateFlowRequest";
const _UFRp = "UpdateFlowResponse";
const _UFVD = "UnspecifiedFlowValidationDetails";
const _UKB = "UpdateKnowledgeBase";
const _UKBR = "UpdateKnowledgeBaseRequest";
const _UKBRp = "UpdateKnowledgeBaseResponse";
const _UNFVD = "UnreachableNodeFlowValidationDetails";
const _UNIFVD = "UnfulfilledNodeInputFlowValidationDetails";
const _UNIFVDn = "UnknownNodeInputFlowValidationDetails";
const _UNOFVD = "UnknownNodeOutputFlowValidationDetails";
const _UP = "UpdatePrompt";
const _UPR = "UpdatePromptRequest";
const _UPRp = "UpdatePromptResponse";
const _UR = "UntagResource";
const _URR = "UntagResourceRequest";
const _URRn = "UntagResourceResponse";
const _VBA = "VectorBucketArn";
const _VC = "VideoConfiguration";
const _VCi = "VideoConfigurations";
const _VE = "ValidationException";
const _VEF = "ValidationExceptionField";
const _VEFL = "ValidationExceptionFieldList";
const _VFD = "ValidateFlowDefinition";
const _VFDR = "ValidateFlowDefinitionRequest";
const _VFDRa = "ValidateFlowDefinitionResponse";
const _VIC = "VectorIngestionConfiguration";
const _VKBC = "VectorKnowledgeBaseConfiguration";
const _VSBRC = "VectorSearchBedrockRerankingConfiguration";
const _VSBRMC = "VectorSearchBedrockRerankingModelConfiguration";
const _VSC = "VideoSegmentationConfiguration";
const _VSRC = "VectorSearchRerankingConfiguration";
const _WCC = "WebCrawlerConfiguration";
const _WCL = "WebCrawlerLimits";
const _WDSC = "WebDataSourceConfiguration";
const _WSC = "WebSourceConfiguration";
const _a = "audio";
const _aA = "agentArn";
const _aAA = "agentAliasArn";
const _aAG = "agentActionGroup";
const _aAHE = "agentAliasHistoryEvents";
const _aAI = "agentAliasId";
const _aAN = "agentAliasName";
const _aAS = "agentAliasStatus";
const _aASg = "agentAliasSummaries";
const _aAg = "agentAlias";
const _aAl = "aliasArn";
const _aC = "agentCollaboration";
const _aCS = "agentCollaboratorSummaries";
const _aCg = "agentCollaborator";
const _aCu = "authConfiguration";
const _aD = "agentDescriptor";
const _aDCC = "awsDataCatalogConfiguration";
const _aGE = "actionGroupExecutor";
const _aGI = "actionGroupId";
const _aGN = "actionGroupName";
const _aGS = "actionGroupState";
const _aGSc = "actionGroupSummaries";
const _aI = "agentId";
const _aIS = "aliasInvocationState";
const _aIg = "agentIdentifier";
const _aIl = "aliasIdentifier";
const _aKB = "agentKnowledgeBase";
const _aKBS = "agentKnowledgeBaseSummaries";
const _aMRF = "additionalModelRequestFields";
const _aN = "agentName";
const _aRRA = "agentResourceRoleArn";
const _aS = "agentStatus";
const _aSg = "agentSummaries";
const _aSp = "apiSchema";
const _aT = "authType";
const _aV = "agentVersion";
const _aVS = "agentVersionSummaries";
const _ag = "agent";
const _an = "any";
const _ar = "arn";
const _at = "attribute";
const _au = "auto";
const _bA = "bucketArn";
const _bAA = "botAliasArn";
const _bC = "byteContent";
const _bDAC = "bedrockDataAutomationConfiguration";
const _bEMC = "bedrockEmbeddingModelConfiguration";
const _bFMC = "bedrockFoundationModelConfiguration";
const _bN = "bucketName";
const _bOAI = "bucketOwnerAccountId";
const _bPT = "basePromptTemplate";
const _bPTr = "breakpointPercentileThreshold";
const _bRC = "bedrockRerankingConfiguration";
const _bS = "bufferSize";
const _bV = "booleanValue";
const _c = "client";
const _cA = "createdAt";
const _cAo = "collectionArn";
const _cC = "crawlerConfiguration";
const _cCh = "chunkingConfiguration";
const _cCo = "concurrencyConfiguration";
const _cCon = "confluenceConfiguration";
const _cCont = "continueCondition";
const _cCu = "customControl";
const _cCy = "cyclicConnection";
const _cDI = "customDocumentIdentifier";
const _cEC = "contextEnrichmentConfiguration";
const _cEKA = "customerEncryptionKeyArn";
const _cI = "collaboratorId";
const _cIl = "clusterIdentifier";
const _cIo = "collaborationInstruction";
const _cL = "crawlerLimits";
const _cMF = "customMetadataField";
const _cN = "collaboratorName";
const _cNo = "collectionName";
const _cO = "customOrchestration";
const _cP = "cachePoint";
const _cQ = "curatedQueries";
const _cS = "chunkingStrategy";
const _cSA = "credentialsSecretArn";
const _cSo = "connectionString";
const _cT = "clientToken";
const _cTC = "customTransformationConfiguration";
const _ca = "category";
const _cau = "cause";
const _ch = "chat";
const _co = "conditions";
const _cod = "code";
const _col = "columns";
const _coll = "collector";
const _con = "connection";
const _cond = "condition";
const _condi = "conditional";
const _conf = "configuration";
const _conn = "connections";
const _cont = "content";
const _cu = "custom";
const _d = "description";
const _dA = "domainArn";
const _dC = "duplicateConnections";
const _dCE = "duplicateConditionExpression";
const _dD = "documentDetails";
const _dDP = "dataDeletionPolicy";
const _dE = "domainEndpoint";
const _dI = "documentIdentifiers";
const _dN = "databaseName";
const _dS = "dataSource";
const _dSC = "dataSourceConfiguration";
const _dSI = "dataSourceId";
const _dSS = "dataSourceSummaries";
const _dST = "dataSourceType";
const _dU = "databaseUser";
const _dV = "defaultVariant";
const _da = "data";
const _de = "definition";
const _det = "details";
const _di = "dimensions";
const _do = "documents";
const _dom = "domain";
const _e = "error";
const _eD = "endDate";
const _eDT = "embeddingDataType";
const _eF = "exclusionFilters";
const _eMA = "embeddingModelArn";
const _eMC = "embeddingModelConfiguration";
const _eMT = "enabledMemoryTypes";
const _eRA = "executionRoleArn";
const _eSC = "enrichmentStrategyConfiguration";
const _eSN = "endpointServiceName";
const _eT = "expectedType";
const _eTS = "executionTimeoutSeconds";
const _en = "endpoint";
const _ex = "executor";
const _exp = "expression";
const _f = "filters";
const _fAS = "flowAliasSummaries";
const _fC = "filterConfiguration";
const _fI = "flowIdentifier";
const _fIl = "flowId";
const _fL = "fieldList";
const _fLD = "fixedLengthDuration";
const _fM = "foundationModel";
const _fMi = "fieldMapping";
const _fN = "fieldName";
const _fR = "failureReasons";
const _fS = "functionSchema";
const _fSCC = "fixedSizeChunkingConfiguration";
const _fSl = "flowSummaries";
const _fTE = "fieldsToExclude";
const _fTI = "fieldsToInclude";
const _fV = "flowVersion";
const _fVS = "flowVersionSummaries";
const _fu = "functions";
const _gA = "graphArn";
const _gAR = "genAiResource";
const _gC = "guardrailConfiguration";
const _gCe = "generationContext";
const _gI = "guardrailIdentifier";
const _gV = "guardrailVersion";
const _h = "http";
const _hCC = "hierarchicalChunkingConfiguration";
const _hE = "httpError";
const _hQ = "httpQuery";
const _hT = "hostType";
const _hU = "hostUrl";
const _i = "instruction";
const _iA = "inlineAttributes";
const _iAn = "indexArn";
const _iC = "inlineContent";
const _iCDT = "incompatibleConnectionDataType";
const _iCn = "inferenceConfiguration";
const _iCnf = "inferenceConfig";
const _iCnl = "inlineCode";
const _iF = "inclusionFilters";
const _iJ = "ingestionJob";
const _iJI = "ingestionJobId";
const _iJS = "ingestionJobSummaries";
const _iLB = "invalidLoopBoundary";
const _iN = "indexName";
const _iNN = "incompatibleNodeName";
const _iNT = "incompatibleNodeType";
const _iP = "inclusionPrefixes";
const _iS = "intermediateStorage";
const _iSTTLIS = "idleSessionTTLInSeconds";
const _iSn = "inputSchema";
const _iV = "inputVariables";
const _id = "id";
const _ide = "identifier";
const _in = "inputs";
const _inc = "inclusion";
const _inl = "inline";
const _inp = "input";
const _it = "iterator";
const _j = "json";
const _k = "key";
const _kB = "knowledgeBase";
const _kBA = "knowledgeBaseArn";
const _kBC = "knowledgeBaseConfiguration";
const _kBI = "knowledgeBaseId";
const _kBS = "knowledgeBaseState";
const _kBSn = "knowledgeBaseSummaries";
const _kIA = "kendraIndexArn";
const _kKA = "kmsKeyArn";
const _kKBC = "kendraKnowledgeBaseConfiguration";
const _l = "language";
const _lA = "lambdaArn";
const _lAV = "latestAgentVersion";
const _lC = "levelConfigurations";
const _lCo = "loopController";
const _lF = "lambdaFunction";
const _lI = "localeId";
const _lINT = "loopIncompatibleNodeType";
const _lIo = "loopInput";
const _lN = "loopNode";
const _lUA = "lastUpdatedAt";
const _la = "latency";
const _lam = "lambda";
const _le = "lex";
const _lo = "loop";
const _m = "message";
const _mA = "modelArn";
const _mC = "memoryConfiguration";
const _mCC = "missingConnectionConfiguration";
const _mCE = "malformedConditionExpression";
const _mCa = "maxConcurrency";
const _mCe = "metadataConfiguration";
const _mCo = "modelConfiguration";
const _mDAC = "mongoDbAtlasConfiguration";
const _mDC = "missingDefaultCondition";
const _mEN = "missingEndingNodes";
const _mF = "metadataField";
const _mI = "modelId";
const _mIa = "maxIterations";
const _mL = "maximumLength";
const _mLCN = "missingLoopControllerNode";
const _mLCNu = "multipleLoopControllerNodes";
const _mLIN = "missingLoopInputNode";
const _mLINu = "multipleLoopInputNodes";
const _mNC = "missingNodeConfiguration";
const _mNI = "missingNodeInput";
const _mNIC = "multipleNodeInputConnections";
const _mNIE = "malformedNodeInputExpression";
const _mNIT = "mismatchedNodeInputType";
const _mNO = "missingNodeOutput";
const _mNOT = "mismatchedNodeOutputType";
const _mP = "maxPages";
const _mR = "maxResults";
const _mRS = "maxRecentSessions";
const _mSN = "missingStartingNodes";
const _mT = "mimeType";
const _mTa = "maxTokens";
const _me = "messages";
const _met = "method";
const _meta = "metadata";
const _n = "name";
const _nAC = "neptuneAnalyticsConfiguration";
const _nL = "naturalLanguage";
const _nODD = "numberOfDocumentsDeleted";
const _nODF = "numberOfDocumentsFailed";
const _nODS = "numberOfDocumentsScanned";
const _nOMDI = "numberOfModifiedDocumentsIndexed";
const _nOMDM = "numberOfMetadataDocumentsModified";
const _nOMDS = "numberOfMetadataDocumentsScanned";
const _nONDI = "numberOfNewDocumentsIndexed";
const _nOR = "numberOfResults";
const _nORR = "numberOfRerankedResults";
const _nT = "nextToken";
const _nV = "numberValue";
const _na = "namespace";
const _no = "node";
const _nod = "nodes";
const _o = "outputs";
const _oC = "orchestrationConfiguration";
const _oL = "overrideLambda";
const _oMCC = "opensearchManagedClusterConfiguration";
const _oP = "overlapPercentage";
const _oSC = "opensearchServerlessConfiguration";
const _oT = "orchestrationType";
const _oTb = "objectType";
const _oTv = "overlapTokens";
const _op = "operator";
const _or = "order";
const _ou = "output";
const _p = "parameters";
const _pA = "preparedAt";
const _pAGS = "parentActionGroupSignature";
const _pAGSP = "parentActionGroupSignatureParams";
const _pAS = "parentActionSignature";
const _pAr = "promptArn";
const _pC = "performanceConfig";
const _pCM = "promptCreationMode";
const _pCa = "parsingConfiguration";
const _pCi = "pineconeConfiguration";
const _pCr = "promptConfigurations";
const _pCro = "provisionedConfiguration";
const _pI = "promptIdentifier";
const _pKF = "primaryKeyField";
const _pM = "parsingModality";
const _pMa = "parserMode";
const _pOC = "promptOverrideConfiguration";
const _pOF = "patternObjectFilter";
const _pP = "parsingPrompt";
const _pPT = "parsingPromptText";
const _pS = "promptSummaries";
const _pSa = "parsingStrategy";
const _pSr = "promptState";
const _pT = "provisionedThroughput";
const _pTr = "promptTemplate";
const _pTro = "promptType";
const _pV = "promptVersion";
const _pa = "payload";
const _pr = "prompt";
const _qEC = "queryEngineConfiguration";
const _qGC = "queryGenerationConfiguration";
const _r = "role";
const _rA = "recommendedActions";
const _rAe = "resourceArn";
const _rAo = "roleArn";
const _rC = "routingConfiguration";
const _rCH = "relayConversationHistory";
const _rCd = "rdsConfiguration";
const _rCe = "requireConfirmation";
const _rCed = "redshiftConfiguration";
const _rCer = "rerankingConfiguration";
const _rECC = "redisEnterpriseCloudConfiguration";
const _rL = "rateLimit";
const _re = "required";
const _res = "resource";
const _ret = "retrieval";
const _s = "system";
const _sA = "startedAt";
const _sB = "sortBy";
const _sBN = "s3BucketName";
const _sC = "segmentationConfiguration";
const _sCC = "semanticChunkingConfiguration";
const _sCa = "salesforceConfiguration";
const _sCe = "serverlessConfiguration";
const _sCer = "serviceConfiguration";
const _sCo = "sourceConfiguration";
const _sCon = "s3Configuration";
const _sCt = "storageConfiguration";
const _sCto = "storageConfigurations";
const _sD = "startDate";
const _sDSC = "supplementalDataStorageConfiguration";
const _sDt = "storageDays";
const _sKBC = "sqlKnowledgeBaseConfiguration";
const _sL = "s3Location";
const _sLV = "stringListValue";
const _sLt = "storageLocations";
const _sM = "selectionMode";
const _sMC = "selectiveModeConfiguration";
const _sO = "sourceOutput";
const _sOK = "s3ObjectKey";
const _sPC = "sharePointConfiguration";
const _sR = "statusReason";
const _sRIUC = "skipResourceInUseCheck";
const _sS = "stopSequences";
const _sSC = "sessionSummaryConfiguration";
const _sSEC = "serverSideEncryptionConfiguration";
const _sT = "sourceType";
const _sTA = "stepToApply";
const _sU = "siteUrls";
const _sUe = "seedUrls";
const _sV = "stringValue";
const _sVC = "s3VectorsConfiguration";
const _s_ = "s3";
const _sc = "scope";
const _se = "severity";
const _ser = "server";
const _sm = "smithy.ts.sdk.synthetic.com.amazonaws.bedrockagent";
const _so = "source";
const _sq = "sql";
const _st = "status";
const _sta = "statistics";
const _sto = "storage";
const _t = "type";
const _tC = "toolConfiguration";
const _tCe = "textContent";
const _tCem = "templateConfiguration";
const _tCo = "toolChoice";
const _tF = "textField";
const _tFr = "transformationFunction";
const _tI = "targetInput";
const _tIN = "textIndexName";
const _tIe = "tenantId";
const _tK = "topK";
const _tKa = "tagKeys";
const _tLC = "transformationLambdaConfiguration";
const _tN = "tableName";
const _tNa = "tableNames";
const _tP = "topP";
const _tPT = "textPromptTemplate";
const _tS = "toolSpec";
const _tT = "templateType";
const _ta = "tags";
const _tab = "tables";
const _tar = "target";
const _te = "temperature";
const _tex = "text";
const _to = "tools";
const _too = "tool";
const _tr = "transformations";
const _u = "uri";
const _uA = "updatedAt";
const _uAH = "userAgentHeader";
const _uAs = "userAgent";
const _uC = "urlConfiguration";
const _uCC = "unknownConnectionCondition";
const _uCCn = "unsatisfiedConnectionConditions";
const _uCS = "unknownConnectionSource";
const _uCSO = "unknownConnectionSourceOutput";
const _uCT = "unknownConnectionTarget";
const _uCTI = "unknownConnectionTargetInput";
const _uN = "unreachableNode";
const _uNI = "unfulfilledNodeInput";
const _uNIn = "unknownNodeInput";
const _uNO = "unknownNodeOutput";
const _uPSA = "usernamePasswordSecretArn";
const _un = "unspecified";
const _ur = "url";
const _v = "version";
const _vBA = "vectorBucketArn";
const _vF = "vectorField";
const _vIC = "vectorIngestionConfiguration";
const _vIN = "vectorIndexName";
const _vKBC = "vectorKnowledgeBaseConfiguration";
const _va = "variants";
const _val = "validations";
const _valu = "values";
const _value = "value";
const _vi = "video";
const _wA = "workgroupArn";
const _wC = "webConfiguration";
const n0 = "com.amazonaws.bedrockagent";
var BasePromptTemplate = [0, n0, _BPT, 8, 0];
var ByteContentBlob = [0, n0, _BCB, 8, 21];
var CollaborationInstruction = [0, n0, _CI, 8, 0];
var Data = [0, n0, _D, 8, 0];
var FilteredObjectType = [0, n0, _FOT, 8, 0];
var FilterPattern = [0, n0, _FP, 8, 0];
var FlowConditionExpression = [0, n0, _FCE, 8, 0];
var FlowNodeInputExpression = [0, n0, _FNIE, 8, 0];
var GraphArn = [0, n0, _GA, 8, 0];
var IndexArn = [0, n0, _IA, 8, 0];
var IndexName = [0, n0, _IN, 8, 0];
var InlineCode = [0, n0, _IC, 8, 0];
var Instruction = [0, n0, _I, 8, 0];
var Key = [0, n0, _K, 8, 0];
var KnowledgeBaseTextPrompt = [0, n0, _KBTP, 8, 0];
var NumberValue = [0, n0, _NV, 8, 1];
var OpenSearchManagedClusterIndexName = [0, n0, _OSMCIN, 8, 0];
var Payload = [0, n0, _P, 8, 0];
var PromptMetadataKey = [0, n0, _PMK, 8, 0];
var PromptMetadataValue = [0, n0, _PMV, 8, 0];
var S3Prefix = [0, n0, _SP, 8, 0];
var StringValue = [0, n0, _SV, 8, 0];
var TextPrompt = [0, n0, _TP, 8, 0];
var UserAgent = [0, n0, _UA, 8, 0];
var UserAgentHeader = [0, n0, _UAH, 8, 0];
var VectorBucketArn = [0, n0, _VBA, 8, 0];
var AccessDeniedException$ = [-3, n0, _ADE,
    { [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(AccessDeniedException$, AccessDeniedException);
var ActionGroupSummary$ = [3, n0, _AGS,
    0,
    [_aGI, _aGN, _aGS, _d, _uA],
    [0, 0, 0, 0, 5]
];
var Agent$ = [3, n0, _A,
    0,
    [_aI, _aN, _aA, _aV, _cT, _i, _aS, _fM, _d, _oT, _cO, _iSTTLIS, _aRRA, _cEKA, _cA, _uA, _pA, _fR, _rA, _pOC, _gC, _mC, _aC],
    [0, 0, 0, 0, 0, [() => Instruction, 0], 0, 0, 0, 0, () => CustomOrchestration$, 1, 0, 0, 5, 5, 5, 64 | 0, 64 | 0, [() => PromptOverrideConfiguration$, 0], () => GuardrailConfiguration$, () => MemoryConfiguration$, 0]
];
var AgentActionGroup$ = [3, n0, _AAG,
    0,
    [_aI, _aV, _aGI, _aGN, _cT, _d, _cA, _uA, _pAS, _pAGSP, _aGE, _aSp, _fS, _aGS],
    [0, 0, 0, 0, 0, 0, 5, 5, 0, 128 | 0, () => ActionGroupExecutor$, [() => APISchema$, 0], () => FunctionSchema$, 0]
];
var AgentAlias$ = [3, n0, _AA,
    0,
    [_aI, _aAI, _aAN, _aAA, _cT, _d, _rC, _cA, _uA, _aAHE, _aAS, _fR, _aIS],
    [0, 0, 0, 0, 0, 0, () => AgentAliasRoutingConfiguration, 5, 5, () => AgentAliasHistoryEvents, 0, 64 | 0, 0]
];
var AgentAliasHistoryEvent$ = [3, n0, _AAHE,
    0,
    [_rC, _eD, _sD],
    [() => AgentAliasRoutingConfiguration, 5, 5]
];
var AgentAliasRoutingConfigurationListItem$ = [3, n0, _AARCLI,
    0,
    [_aV, _pT],
    [0, 0]
];
var AgentAliasSummary$ = [3, n0, _AAS,
    0,
    [_aAI, _aAN, _d, _rC, _aAS, _cA, _uA, _aIS],
    [0, 0, 0, () => AgentAliasRoutingConfiguration, 0, 5, 5, 0]
];
var AgentCollaborator$ = [3, n0, _AC,
    0,
    [_aI, _aV, _aD, _cI, _cIo, _cN, _cA, _lUA, _rCH, _cT],
    [0, 0, () => AgentDescriptor$, 0, [() => CollaborationInstruction, 0], 0, 5, 5, 0, 0]
];
var AgentCollaboratorSummary$ = [3, n0, _ACS,
    0,
    [_aI, _aV, _cI, _aD, _cIo, _rCH, _cN, _cA, _lUA],
    [0, 0, 0, () => AgentDescriptor$, [() => CollaborationInstruction, 0], 0, 0, 5, 5]
];
var AgentDescriptor$ = [3, n0, _AD,
    0,
    [_aAl],
    [0]
];
var AgentFlowNodeConfiguration$ = [3, n0, _AFNC,
    0,
    [_aAA],
    [0]
];
var AgentKnowledgeBase$ = [3, n0, _AKB,
    0,
    [_aI, _aV, _kBI, _d, _cA, _uA, _kBS],
    [0, 0, 0, 0, 5, 5, 0]
];
var AgentKnowledgeBaseSummary$ = [3, n0, _AKBS,
    0,
    [_kBI, _d, _kBS, _uA],
    [0, 0, 0, 5]
];
var AgentSummary$ = [3, n0, _AS,
    0,
    [_aI, _aN, _aS, _d, _uA, _lAV, _gC],
    [0, 0, 0, 0, 5, 0, () => GuardrailConfiguration$]
];
var AgentVersion$ = [3, n0, _AV,
    0,
    [_aI, _aN, _aA, _v, _i, _aS, _fM, _d, _iSTTLIS, _aRRA, _cEKA, _cA, _uA, _fR, _rA, _pOC, _gC, _mC, _aC],
    [0, 0, 0, 0, [() => Instruction, 0], 0, 0, 0, 1, 0, 0, 5, 5, 64 | 0, 64 | 0, [() => PromptOverrideConfiguration$, 0], () => GuardrailConfiguration$, () => MemoryConfiguration$, 0]
];
var AgentVersionSummary$ = [3, n0, _AVS,
    0,
    [_aN, _aS, _aV, _cA, _uA, _d, _gC],
    [0, 0, 0, 5, 5, 0, () => GuardrailConfiguration$]
];
var AnyToolChoice$ = [3, n0, _ATC,
    0,
    [],
    []
];
var AssociateAgentCollaboratorRequest$ = [3, n0, _AACR,
    0,
    [_aI, _aV, _aD, _cN, _cIo, _rCH, _cT],
    [[0, 1], [0, 1], () => AgentDescriptor$, 0, [() => CollaborationInstruction, 0], 0, [0, 4]]
];
var AssociateAgentCollaboratorResponse$ = [3, n0, _AACRs,
    0,
    [_aCg],
    [[() => AgentCollaborator$, 0]]
];
var AssociateAgentKnowledgeBaseRequest$ = [3, n0, _AAKBR,
    0,
    [_aI, _aV, _kBI, _d, _kBS],
    [[0, 1], [0, 1], 0, 0, 0]
];
var AssociateAgentKnowledgeBaseResponse$ = [3, n0, _AAKBRs,
    0,
    [_aKB],
    [() => AgentKnowledgeBase$]
];
var AudioConfiguration$ = [3, n0, _ACu,
    0,
    [_sC],
    [() => AudioSegmentationConfiguration$]
];
var AudioSegmentationConfiguration$ = [3, n0, _ASC,
    0,
    [_fLD],
    [1]
];
var AutoToolChoice$ = [3, n0, _ATCu,
    0,
    [],
    []
];
var BedrockDataAutomationConfiguration$ = [3, n0, _BDAC,
    0,
    [_pM],
    [0]
];
var BedrockEmbeddingModelConfiguration$ = [3, n0, _BEMC,
    0,
    [_di, _eDT, _a, _vi],
    [1, 0, () => AudioConfigurations, () => VideoConfigurations]
];
var BedrockFoundationModelConfiguration$ = [3, n0, _BFMC,
    0,
    [_mA, _pP, _pM],
    [0, () => ParsingPrompt$, 0]
];
var BedrockFoundationModelContextEnrichmentConfiguration$ = [3, n0, _BFMCEC,
    0,
    [_eSC, _mA],
    [() => EnrichmentStrategyConfiguration$, 0]
];
var ByteContentDoc$ = [3, n0, _BCD,
    0,
    [_mT, _da],
    [0, [() => ByteContentBlob, 0]]
];
var CachePointBlock$ = [3, n0, _CPB,
    0,
    [_t],
    [0]
];
var ChatPromptTemplateConfiguration$ = [3, n0, _CPTC,
    8,
    [_me, _s, _iV, _tC],
    [[() => Messages, 0], [() => SystemContentBlocks, 0], [() => PromptInputVariablesList, 0], [() => ToolConfiguration$, 0]]
];
var ChunkingConfiguration$ = [3, n0, _CC,
    0,
    [_cS, _fSCC, _hCC, _sCC],
    [0, () => FixedSizeChunkingConfiguration$, () => HierarchicalChunkingConfiguration$, () => SemanticChunkingConfiguration$]
];
var CollectorFlowNodeConfiguration$ = [3, n0, _CFNC,
    0,
    [],
    []
];
var ConditionFlowNodeConfiguration$ = [3, n0, _CFNCo,
    0,
    [_co],
    [[() => FlowConditions, 0]]
];
var ConflictException$ = [-3, n0, _CE,
    { [_e]: _c, [_hE]: 409 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(ConflictException$, ConflictException);
var ConfluenceCrawlerConfiguration$ = [3, n0, _CCC,
    0,
    [_fC],
    [[() => CrawlFilterConfiguration$, 0]]
];
var ConfluenceDataSourceConfiguration$ = [3, n0, _CDSC,
    0,
    [_sCo, _cC],
    [() => ConfluenceSourceConfiguration$, [() => ConfluenceCrawlerConfiguration$, 0]]
];
var ConfluenceSourceConfiguration$ = [3, n0, _CSC,
    0,
    [_hU, _hT, _aT, _cSA],
    [0, 0, 0, 0]
];
var ContextEnrichmentConfiguration$ = [3, n0, _CEC,
    0,
    [_t, _bFMC],
    [0, () => BedrockFoundationModelContextEnrichmentConfiguration$]
];
var CrawlFilterConfiguration$ = [3, n0, _CFC,
    0,
    [_t, _pOF],
    [0, [() => PatternObjectFilterConfiguration$, 0]]
];
var CreateAgentActionGroupRequest$ = [3, n0, _CAAGR,
    0,
    [_aI, _aV, _aGN, _cT, _d, _pAGS, _pAGSP, _aGE, _aSp, _aGS, _fS],
    [[0, 1], [0, 1], 0, [0, 4], 0, 0, 128 | 0, () => ActionGroupExecutor$, [() => APISchema$, 0], 0, () => FunctionSchema$]
];
var CreateAgentActionGroupResponse$ = [3, n0, _CAAGRr,
    0,
    [_aAG],
    [[() => AgentActionGroup$, 0]]
];
var CreateAgentAliasRequest$ = [3, n0, _CAAR,
    0,
    [_aI, _aAN, _cT, _d, _rC, _ta],
    [[0, 1], 0, [0, 4], 0, () => AgentAliasRoutingConfiguration, 128 | 0]
];
var CreateAgentAliasResponse$ = [3, n0, _CAARr,
    0,
    [_aAg],
    [() => AgentAlias$]
];
var CreateAgentRequest$ = [3, n0, _CAR,
    0,
    [_aN, _cT, _i, _fM, _d, _oT, _cO, _iSTTLIS, _aRRA, _cEKA, _ta, _pOC, _gC, _mC, _aC],
    [0, [0, 4], [() => Instruction, 0], 0, 0, 0, () => CustomOrchestration$, 1, 0, 0, 128 | 0, [() => PromptOverrideConfiguration$, 0], () => GuardrailConfiguration$, () => MemoryConfiguration$, 0]
];
var CreateAgentResponse$ = [3, n0, _CARr,
    0,
    [_ag],
    [[() => Agent$, 0]]
];
var CreateDataSourceRequest$ = [3, n0, _CDSR,
    0,
    [_kBI, _cT, _n, _d, _dSC, _dDP, _sSEC, _vIC],
    [[0, 1], [0, 4], 0, 0, [() => DataSourceConfiguration$, 0], 0, () => ServerSideEncryptionConfiguration$, () => VectorIngestionConfiguration$]
];
var CreateDataSourceResponse$ = [3, n0, _CDSRr,
    0,
    [_dS],
    [[() => DataSource$, 0]]
];
var CreateFlowAliasRequest$ = [3, n0, _CFAR,
    0,
    [_n, _d, _rC, _cCo, _fI, _cT, _ta],
    [0, 0, () => FlowAliasRoutingConfiguration, () => FlowAliasConcurrencyConfiguration$, [0, 1], [0, 4], 128 | 0]
];
var CreateFlowAliasResponse$ = [3, n0, _CFARr,
    0,
    [_n, _d, _rC, _cCo, _fIl, _id, _ar, _cA, _uA],
    [0, 0, () => FlowAliasRoutingConfiguration, () => FlowAliasConcurrencyConfiguration$, 0, 0, 0, 5, 5]
];
var CreateFlowRequest$ = [3, n0, _CFR,
    0,
    [_n, _d, _eRA, _cEKA, _de, _cT, _ta],
    [0, 0, 0, 0, [() => FlowDefinition$, 0], [0, 4], 128 | 0]
];
var CreateFlowResponse$ = [3, n0, _CFRr,
    0,
    [_n, _d, _eRA, _cEKA, _id, _ar, _st, _cA, _uA, _v, _de],
    [0, 0, 0, 0, 0, 0, 0, 5, 5, 0, [() => FlowDefinition$, 0]]
];
var CreateFlowVersionRequest$ = [3, n0, _CFVR,
    0,
    [_fI, _d, _cT],
    [[0, 1], 0, [0, 4]]
];
var CreateFlowVersionResponse$ = [3, n0, _CFVRr,
    0,
    [_n, _d, _eRA, _cEKA, _id, _ar, _st, _cA, _v, _de],
    [0, 0, 0, 0, 0, 0, 0, 5, 0, [() => FlowDefinition$, 0]]
];
var CreateKnowledgeBaseRequest$ = [3, n0, _CKBR,
    0,
    [_cT, _n, _d, _rAo, _kBC, _sCt, _ta],
    [[0, 4], 0, 0, 0, [() => KnowledgeBaseConfiguration$, 0], [() => StorageConfiguration$, 0], 128 | 0]
];
var CreateKnowledgeBaseResponse$ = [3, n0, _CKBRr,
    0,
    [_kB],
    [[() => KnowledgeBase$, 0]]
];
var CreatePromptRequest$ = [3, n0, _CPR,
    0,
    [_n, _d, _cEKA, _dV, _va, _cT, _ta],
    [0, 0, 0, 0, [() => PromptVariantList, 0], [0, 4], 128 | 0]
];
var CreatePromptResponse$ = [3, n0, _CPRr,
    0,
    [_n, _d, _cEKA, _dV, _va, _id, _ar, _v, _cA, _uA],
    [0, 0, 0, 0, [() => PromptVariantList, 0], 0, 0, 0, 5, 5]
];
var CreatePromptVersionRequest$ = [3, n0, _CPVR,
    0,
    [_pI, _d, _cT, _ta],
    [[0, 1], 0, [0, 4], 128 | 0]
];
var CreatePromptVersionResponse$ = [3, n0, _CPVRr,
    0,
    [_n, _d, _cEKA, _dV, _va, _id, _ar, _v, _cA, _uA],
    [0, 0, 0, 0, [() => PromptVariantList, 0], 0, 0, 0, 5, 5]
];
var CuratedQuery$ = [3, n0, _CQ,
    0,
    [_nL, _sq],
    [0, 0]
];
var CustomContent$ = [3, n0, _CCu,
    0,
    [_cDI, _sT, _sL, _iC],
    [() => CustomDocumentIdentifier$, 0, () => CustomS3Location$, [() => InlineContent$, 0]]
];
var CustomDocumentIdentifier$ = [3, n0, _CDI,
    0,
    [_id],
    [0]
];
var CustomOrchestration$ = [3, n0, _CO,
    0,
    [_ex],
    [() => OrchestrationExecutor$]
];
var CustomS3Location$ = [3, n0, _CSL,
    0,
    [_u, _bOAI],
    [0, 0]
];
var CustomTransformationConfiguration$ = [3, n0, _CTC,
    0,
    [_iS, _tr],
    [() => IntermediateStorage$, () => Transformations]
];
var CyclicConnectionFlowValidationDetails$ = [3, n0, _CCFVD,
    0,
    [_con],
    [0]
];
var DataSource$ = [3, n0, _DS,
    0,
    [_kBI, _dSI, _n, _st, _d, _dSC, _sSEC, _vIC, _dDP, _cA, _uA, _fR],
    [0, 0, 0, 0, 0, [() => DataSourceConfiguration$, 0], () => ServerSideEncryptionConfiguration$, () => VectorIngestionConfiguration$, 0, 5, 5, 64 | 0]
];
var DataSourceConfiguration$ = [3, n0, _DSC,
    0,
    [_t, _sCon, _wC, _cCon, _sCa, _sPC],
    [0, [() => S3DataSourceConfiguration$, 0], [() => WebDataSourceConfiguration$, 0], [() => ConfluenceDataSourceConfiguration$, 0], [() => SalesforceDataSourceConfiguration$, 0], [() => SharePointDataSourceConfiguration$, 0]]
];
var DataSourceSummary$ = [3, n0, _DSS,
    0,
    [_kBI, _dSI, _n, _st, _d, _uA],
    [0, 0, 0, 0, 0, 5]
];
var DeleteAgentActionGroupRequest$ = [3, n0, _DAAGR,
    0,
    [_aI, _aV, _aGI, _sRIUC],
    [[0, 1], [0, 1], [0, 1], [2, { [_hQ]: _sRIUC }]]
];
var DeleteAgentActionGroupResponse$ = [3, n0, _DAAGRe,
    0,
    [],
    []
];
var DeleteAgentAliasRequest$ = [3, n0, _DAAR,
    0,
    [_aI, _aAI],
    [[0, 1], [0, 1]]
];
var DeleteAgentAliasResponse$ = [3, n0, _DAARe,
    0,
    [_aI, _aAI, _aAS],
    [0, 0, 0]
];
var DeleteAgentRequest$ = [3, n0, _DAR,
    0,
    [_aI, _sRIUC],
    [[0, 1], [2, { [_hQ]: _sRIUC }]]
];
var DeleteAgentResponse$ = [3, n0, _DARe,
    0,
    [_aI, _aS],
    [0, 0]
];
var DeleteAgentVersionRequest$ = [3, n0, _DAVR,
    0,
    [_aI, _aV, _sRIUC],
    [[0, 1], [0, 1], [2, { [_hQ]: _sRIUC }]]
];
var DeleteAgentVersionResponse$ = [3, n0, _DAVRe,
    0,
    [_aI, _aV, _aS],
    [0, 0, 0]
];
var DeleteDataSourceRequest$ = [3, n0, _DDSR,
    0,
    [_kBI, _dSI],
    [[0, 1], [0, 1]]
];
var DeleteDataSourceResponse$ = [3, n0, _DDSRe,
    0,
    [_kBI, _dSI, _st],
    [0, 0, 0]
];
var DeleteFlowAliasRequest$ = [3, n0, _DFAR,
    0,
    [_fI, _aIl],
    [[0, 1], [0, 1]]
];
var DeleteFlowAliasResponse$ = [3, n0, _DFARe,
    0,
    [_fIl, _id],
    [0, 0]
];
var DeleteFlowRequest$ = [3, n0, _DFR,
    0,
    [_fI, _sRIUC],
    [[0, 1], [2, { [_hQ]: _sRIUC }]]
];
var DeleteFlowResponse$ = [3, n0, _DFRe,
    0,
    [_id],
    [0]
];
var DeleteFlowVersionRequest$ = [3, n0, _DFVR,
    0,
    [_fI, _fV, _sRIUC],
    [[0, 1], [0, 1], [2, { [_hQ]: _sRIUC }]]
];
var DeleteFlowVersionResponse$ = [3, n0, _DFVRe,
    0,
    [_id, _v],
    [0, 0]
];
var DeleteKnowledgeBaseDocumentsRequest$ = [3, n0, _DKBDR,
    0,
    [_kBI, _dSI, _cT, _dI],
    [[0, 1], [0, 1], [0, 4], () => DocumentIdentifiers]
];
var DeleteKnowledgeBaseDocumentsResponse$ = [3, n0, _DKBDRe,
    0,
    [_dD],
    [() => KnowledgeBaseDocumentDetails]
];
var DeleteKnowledgeBaseRequest$ = [3, n0, _DKBR,
    0,
    [_kBI],
    [[0, 1]]
];
var DeleteKnowledgeBaseResponse$ = [3, n0, _DKBRe,
    0,
    [_kBI, _st],
    [0, 0]
];
var DeletePromptRequest$ = [3, n0, _DPR,
    0,
    [_pI, _pV],
    [[0, 1], [0, { [_hQ]: _pV }]]
];
var DeletePromptResponse$ = [3, n0, _DPRe,
    0,
    [_id, _v],
    [0, 0]
];
var DisassociateAgentCollaboratorRequest$ = [3, n0, _DACR,
    0,
    [_aI, _aV, _cI],
    [[0, 1], [0, 1], [0, 1]]
];
var DisassociateAgentCollaboratorResponse$ = [3, n0, _DACRi,
    0,
    [],
    []
];
var DisassociateAgentKnowledgeBaseRequest$ = [3, n0, _DAKBR,
    0,
    [_aI, _aV, _kBI],
    [[0, 1], [0, 1], [0, 1]]
];
var DisassociateAgentKnowledgeBaseResponse$ = [3, n0, _DAKBRi,
    0,
    [],
    []
];
var DocumentContent$ = [3, n0, _DC,
    0,
    [_dST, _cu, _s_],
    [0, [() => CustomContent$, 0], () => S3Content$]
];
var DocumentIdentifier$ = [3, n0, _DI,
    0,
    [_dST, _s_, _cu],
    [0, () => S3Location$, () => CustomDocumentIdentifier$]
];
var DocumentMetadata$ = [3, n0, _DM,
    0,
    [_t, _iA, _sL],
    [0, [() => MetadataAttributes, 0], () => CustomS3Location$]
];
var DuplicateConditionExpressionFlowValidationDetails$ = [3, n0, _DCEFVD,
    0,
    [_no, _exp],
    [0, [() => FlowConditionExpression, 0]]
];
var DuplicateConnectionsFlowValidationDetails$ = [3, n0, _DCFVD,
    0,
    [_so, _tar],
    [0, 0]
];
var EmbeddingModelConfiguration$ = [3, n0, _EMC,
    0,
    [_bEMC],
    [() => BedrockEmbeddingModelConfiguration$]
];
var EnrichmentStrategyConfiguration$ = [3, n0, _ESC,
    0,
    [_met],
    [0]
];
var FieldForReranking$ = [3, n0, _FFR,
    0,
    [_fN],
    [0]
];
var FixedSizeChunkingConfiguration$ = [3, n0, _FSCC,
    0,
    [_mTa, _oP],
    [1, 1]
];
var FlowAliasConcurrencyConfiguration$ = [3, n0, _FACC,
    0,
    [_t, _mCa],
    [0, 1]
];
var FlowAliasRoutingConfigurationListItem$ = [3, n0, _FARCLI,
    0,
    [_fV],
    [0]
];
var FlowAliasSummary$ = [3, n0, _FAS,
    0,
    [_n, _d, _rC, _cCo, _fIl, _id, _ar, _cA, _uA],
    [0, 0, () => FlowAliasRoutingConfiguration, () => FlowAliasConcurrencyConfiguration$, 0, 0, 0, 5, 5]
];
var FlowCondition$ = [3, n0, _FC,
    0,
    [_n, _exp],
    [0, [() => FlowConditionExpression, 0]]
];
var FlowConditionalConnectionConfiguration$ = [3, n0, _FCCC,
    0,
    [_cond],
    [0]
];
var FlowConnection$ = [3, n0, _FCl,
    0,
    [_t, _n, _so, _tar, _conf],
    [0, 0, 0, 0, () => FlowConnectionConfiguration$]
];
var FlowDataConnectionConfiguration$ = [3, n0, _FDCC,
    0,
    [_sO, _tI],
    [0, 0]
];
var FlowDefinition$ = [3, n0, _FD,
    8,
    [_nod, _conn],
    [[() => FlowNodes, 0], () => FlowConnections]
];
var FlowNode$ = [3, n0, _FN,
    0,
    [_n, _t, _conf, _in, _o],
    [0, 0, [() => FlowNodeConfiguration$, 0], [() => FlowNodeInputs, 0], () => FlowNodeOutputs]
];
var FlowNodeInput$ = [3, n0, _FNI,
    0,
    [_n, _t, _exp, _ca],
    [0, 0, [() => FlowNodeInputExpression, 0], 0]
];
var FlowNodeOutput$ = [3, n0, _FNO,
    0,
    [_n, _t],
    [0, 0]
];
var FlowSummary$ = [3, n0, _FS,
    0,
    [_n, _d, _id, _ar, _st, _cA, _uA, _v],
    [0, 0, 0, 0, 0, 5, 5, 0]
];
var FlowValidation$ = [3, n0, _FV,
    0,
    [_m, _se, _det, _t],
    [0, 0, [() => FlowValidationDetails$, 0], 0]
];
var FlowVersionSummary$ = [3, n0, _FVS,
    0,
    [_id, _ar, _st, _cA, _v],
    [0, 0, 0, 5, 0]
];
var Function$ = [3, n0, _F,
    0,
    [_n, _d, _p, _rCe],
    [0, 0, () => ParameterMap, 0]
];
var GetAgentActionGroupRequest$ = [3, n0, _GAAGR,
    0,
    [_aI, _aV, _aGI],
    [[0, 1], [0, 1], [0, 1]]
];
var GetAgentActionGroupResponse$ = [3, n0, _GAAGRe,
    0,
    [_aAG],
    [[() => AgentActionGroup$, 0]]
];
var GetAgentAliasRequest$ = [3, n0, _GAAR,
    0,
    [_aI, _aAI],
    [[0, 1], [0, 1]]
];
var GetAgentAliasResponse$ = [3, n0, _GAARe,
    0,
    [_aAg],
    [() => AgentAlias$]
];
var GetAgentCollaboratorRequest$ = [3, n0, _GACR,
    0,
    [_aI, _aV, _cI],
    [[0, 1], [0, 1], [0, 1]]
];
var GetAgentCollaboratorResponse$ = [3, n0, _GACRe,
    0,
    [_aCg],
    [[() => AgentCollaborator$, 0]]
];
var GetAgentKnowledgeBaseRequest$ = [3, n0, _GAKBR,
    0,
    [_aI, _aV, _kBI],
    [[0, 1], [0, 1], [0, 1]]
];
var GetAgentKnowledgeBaseResponse$ = [3, n0, _GAKBRe,
    0,
    [_aKB],
    [() => AgentKnowledgeBase$]
];
var GetAgentRequest$ = [3, n0, _GAR,
    0,
    [_aI],
    [[0, 1]]
];
var GetAgentResponse$ = [3, n0, _GARe,
    0,
    [_ag],
    [[() => Agent$, 0]]
];
var GetAgentVersionRequest$ = [3, n0, _GAVR,
    0,
    [_aI, _aV],
    [[0, 1], [0, 1]]
];
var GetAgentVersionResponse$ = [3, n0, _GAVRe,
    0,
    [_aV],
    [[() => AgentVersion$, 0]]
];
var GetDataSourceRequest$ = [3, n0, _GDSR,
    0,
    [_kBI, _dSI],
    [[0, 1], [0, 1]]
];
var GetDataSourceResponse$ = [3, n0, _GDSRe,
    0,
    [_dS],
    [[() => DataSource$, 0]]
];
var GetFlowAliasRequest$ = [3, n0, _GFAR,
    0,
    [_fI, _aIl],
    [[0, 1], [0, 1]]
];
var GetFlowAliasResponse$ = [3, n0, _GFARe,
    0,
    [_n, _d, _rC, _cCo, _fIl, _id, _ar, _cA, _uA],
    [0, 0, () => FlowAliasRoutingConfiguration, () => FlowAliasConcurrencyConfiguration$, 0, 0, 0, 5, 5]
];
var GetFlowRequest$ = [3, n0, _GFR,
    0,
    [_fI],
    [[0, 1]]
];
var GetFlowResponse$ = [3, n0, _GFRe,
    0,
    [_n, _d, _eRA, _cEKA, _id, _ar, _st, _cA, _uA, _v, _de, _val],
    [0, 0, 0, 0, 0, 0, 0, 5, 5, 0, [() => FlowDefinition$, 0], [() => FlowValidations, 0]]
];
var GetFlowVersionRequest$ = [3, n0, _GFVR,
    0,
    [_fI, _fV],
    [[0, 1], [0, 1]]
];
var GetFlowVersionResponse$ = [3, n0, _GFVRe,
    0,
    [_n, _d, _eRA, _cEKA, _id, _ar, _st, _cA, _v, _de],
    [0, 0, 0, 0, 0, 0, 0, 5, 0, [() => FlowDefinition$, 0]]
];
var GetIngestionJobRequest$ = [3, n0, _GIJR,
    0,
    [_kBI, _dSI, _iJI],
    [[0, 1], [0, 1], [0, 1]]
];
var GetIngestionJobResponse$ = [3, n0, _GIJRe,
    0,
    [_iJ],
    [() => IngestionJob$]
];
var GetKnowledgeBaseDocumentsRequest$ = [3, n0, _GKBDR,
    0,
    [_kBI, _dSI, _dI],
    [[0, 1], [0, 1], () => DocumentIdentifiers]
];
var GetKnowledgeBaseDocumentsResponse$ = [3, n0, _GKBDRe,
    0,
    [_dD],
    [() => KnowledgeBaseDocumentDetails]
];
var GetKnowledgeBaseRequest$ = [3, n0, _GKBR,
    0,
    [_kBI],
    [[0, 1]]
];
var GetKnowledgeBaseResponse$ = [3, n0, _GKBRe,
    0,
    [_kB],
    [[() => KnowledgeBase$, 0]]
];
var GetPromptRequest$ = [3, n0, _GPR,
    0,
    [_pI, _pV],
    [[0, 1], [0, { [_hQ]: _pV }]]
];
var GetPromptResponse$ = [3, n0, _GPRe,
    0,
    [_n, _d, _cEKA, _dV, _va, _id, _ar, _v, _cA, _uA],
    [0, 0, 0, 0, [() => PromptVariantList, 0], 0, 0, 0, 5, 5]
];
var GuardrailConfiguration$ = [3, n0, _GC,
    0,
    [_gI, _gV],
    [0, 0]
];
var HierarchicalChunkingConfiguration$ = [3, n0, _HCC,
    0,
    [_lC, _oTv],
    [() => HierarchicalChunkingLevelConfigurations, 1]
];
var HierarchicalChunkingLevelConfiguration$ = [3, n0, _HCLC,
    0,
    [_mTa],
    [1]
];
var IncompatibleConnectionDataTypeFlowValidationDetails$ = [3, n0, _ICDTFVD,
    0,
    [_con],
    [0]
];
var InferenceConfiguration$ = [3, n0, _ICn,
    0,
    [_te, _tP, _tK, _mL, _sS],
    [1, 1, 1, 1, 64 | 0]
];
var IngestionJob$ = [3, n0, _IJ,
    0,
    [_kBI, _dSI, _iJI, _d, _st, _sta, _fR, _sA, _uA],
    [0, 0, 0, 0, 0, () => IngestionJobStatistics$, 64 | 0, 5, 5]
];
var IngestionJobFilter$ = [3, n0, _IJF,
    0,
    [_at, _op, _valu],
    [0, 0, 64 | 0]
];
var IngestionJobSortBy$ = [3, n0, _IJSB,
    0,
    [_at, _or],
    [0, 0]
];
var IngestionJobStatistics$ = [3, n0, _IJS,
    0,
    [_nODS, _nOMDS, _nONDI, _nOMDI, _nOMDM, _nODD, _nODF],
    [1, 1, 1, 1, 1, 1, 1]
];
var IngestionJobSummary$ = [3, n0, _IJSn,
    0,
    [_kBI, _dSI, _iJI, _d, _st, _sA, _uA, _sta],
    [0, 0, 0, 0, 0, 5, 5, () => IngestionJobStatistics$]
];
var IngestKnowledgeBaseDocumentsRequest$ = [3, n0, _IKBDR,
    0,
    [_kBI, _dSI, _cT, _do],
    [[0, 1], [0, 1], [0, 4], [() => KnowledgeBaseDocuments, 0]]
];
var IngestKnowledgeBaseDocumentsResponse$ = [3, n0, _IKBDRn,
    0,
    [_dD],
    [() => KnowledgeBaseDocumentDetails]
];
var InlineCodeFlowNodeConfiguration$ = [3, n0, _ICFNC,
    0,
    [_cod, _l],
    [[() => InlineCode, 0], 0]
];
var InlineContent$ = [3, n0, _ICnl,
    0,
    [_t, _bC, _tCe],
    [0, [() => ByteContentDoc$, 0], [() => TextContentDoc$, 0]]
];
var InputFlowNodeConfiguration$ = [3, n0, _IFNC,
    0,
    [],
    []
];
var IntermediateStorage$ = [3, n0, _IS,
    0,
    [_sL],
    [() => S3Location$]
];
var InternalServerException$ = [-3, n0, _ISE,
    { [_e]: _ser, [_hE]: 500 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(InternalServerException$, InternalServerException);
var InvalidLoopBoundaryFlowValidationDetails$ = [3, n0, _ILBFVD,
    0,
    [_con, _so, _tar],
    [0, 0, 0]
];
var IteratorFlowNodeConfiguration$ = [3, n0, _IFNCt,
    0,
    [],
    []
];
var KendraKnowledgeBaseConfiguration$ = [3, n0, _KKBC,
    0,
    [_kIA],
    [0]
];
var KnowledgeBase$ = [3, n0, _KB,
    0,
    [_kBI, _n, _kBA, _d, _rAo, _kBC, _sCt, _st, _cA, _uA, _fR],
    [0, 0, 0, 0, 0, [() => KnowledgeBaseConfiguration$, 0], [() => StorageConfiguration$, 0], 0, 5, 5, 64 | 0]
];
var KnowledgeBaseConfiguration$ = [3, n0, _KBC,
    0,
    [_t, _vKBC, _kKBC, _sKBC],
    [0, () => VectorKnowledgeBaseConfiguration$, () => KendraKnowledgeBaseConfiguration$, [() => SqlKnowledgeBaseConfiguration$, 0]]
];
var KnowledgeBaseDocument$ = [3, n0, _KBD,
    0,
    [_meta, _cont],
    [[() => DocumentMetadata$, 0], [() => DocumentContent$, 0]]
];
var KnowledgeBaseDocumentDetail$ = [3, n0, _KBDD,
    0,
    [_kBI, _dSI, _st, _ide, _sR, _uA],
    [0, 0, 0, () => DocumentIdentifier$, 0, 5]
];
var KnowledgeBaseFlowNodeConfiguration$ = [3, n0, _KBFNC,
    0,
    [_kBI, _mI, _gC, _nOR, _pTr, _iCn, _rCer, _oC],
    [0, 0, () => GuardrailConfiguration$, 1, [() => KnowledgeBasePromptTemplate$, 0], () => PromptInferenceConfiguration$, [() => VectorSearchRerankingConfiguration$, 0], [() => KnowledgeBaseOrchestrationConfiguration$, 0]]
];
var KnowledgeBaseOrchestrationConfiguration$ = [3, n0, _KBOC,
    0,
    [_pTr, _iCnf, _aMRF, _pC],
    [[() => KnowledgeBasePromptTemplate$, 0], () => PromptInferenceConfiguration$, 128 | 15, () => PerformanceConfiguration$]
];
var KnowledgeBasePromptTemplate$ = [3, n0, _KBPT,
    0,
    [_tPT],
    [[() => KnowledgeBaseTextPrompt, 0]]
];
var KnowledgeBaseSummary$ = [3, n0, _KBS,
    0,
    [_kBI, _n, _d, _st, _uA],
    [0, 0, 0, 0, 5]
];
var LambdaFunctionFlowNodeConfiguration$ = [3, n0, _LFFNC,
    0,
    [_lA],
    [0]
];
var LexFlowNodeConfiguration$ = [3, n0, _LFNC,
    0,
    [_bAA, _lI],
    [0, 0]
];
var ListAgentActionGroupsRequest$ = [3, n0, _LAAGR,
    0,
    [_aI, _aV, _mR, _nT],
    [[0, 1], [0, 1], 1, 0]
];
var ListAgentActionGroupsResponse$ = [3, n0, _LAAGRi,
    0,
    [_aGSc, _nT],
    [() => ActionGroupSummaries, 0]
];
var ListAgentAliasesRequest$ = [3, n0, _LAAR,
    0,
    [_aI, _mR, _nT],
    [[0, 1], 1, 0]
];
var ListAgentAliasesResponse$ = [3, n0, _LAARi,
    0,
    [_aASg, _nT],
    [() => AgentAliasSummaries, 0]
];
var ListAgentCollaboratorsRequest$ = [3, n0, _LACR,
    0,
    [_aI, _aV, _mR, _nT],
    [[0, 1], [0, 1], 1, 0]
];
var ListAgentCollaboratorsResponse$ = [3, n0, _LACRi,
    0,
    [_aCS, _nT],
    [[() => AgentCollaboratorSummaries, 0], 0]
];
var ListAgentKnowledgeBasesRequest$ = [3, n0, _LAKBR,
    0,
    [_aI, _aV, _mR, _nT],
    [[0, 1], [0, 1], 1, 0]
];
var ListAgentKnowledgeBasesResponse$ = [3, n0, _LAKBRi,
    0,
    [_aKBS, _nT],
    [() => AgentKnowledgeBaseSummaries, 0]
];
var ListAgentsRequest$ = [3, n0, _LAR,
    0,
    [_mR, _nT],
    [1, 0]
];
var ListAgentsResponse$ = [3, n0, _LARi,
    0,
    [_aSg, _nT],
    [() => AgentSummaries, 0]
];
var ListAgentVersionsRequest$ = [3, n0, _LAVR,
    0,
    [_aI, _mR, _nT],
    [[0, 1], 1, 0]
];
var ListAgentVersionsResponse$ = [3, n0, _LAVRi,
    0,
    [_aVS, _nT],
    [() => AgentVersionSummaries, 0]
];
var ListDataSourcesRequest$ = [3, n0, _LDSR,
    0,
    [_kBI, _mR, _nT],
    [[0, 1], 1, 0]
];
var ListDataSourcesResponse$ = [3, n0, _LDSRi,
    0,
    [_dSS, _nT],
    [() => DataSourceSummaries, 0]
];
var ListFlowAliasesRequest$ = [3, n0, _LFAR,
    0,
    [_fI, _mR, _nT],
    [[0, 1], [1, { [_hQ]: _mR }], [0, { [_hQ]: _nT }]]
];
var ListFlowAliasesResponse$ = [3, n0, _LFARi,
    0,
    [_fAS, _nT],
    [() => FlowAliasSummaries, 0]
];
var ListFlowsRequest$ = [3, n0, _LFR,
    0,
    [_mR, _nT],
    [[1, { [_hQ]: _mR }], [0, { [_hQ]: _nT }]]
];
var ListFlowsResponse$ = [3, n0, _LFRi,
    0,
    [_fSl, _nT],
    [() => FlowSummaries, 0]
];
var ListFlowVersionsRequest$ = [3, n0, _LFVR,
    0,
    [_fI, _mR, _nT],
    [[0, 1], [1, { [_hQ]: _mR }], [0, { [_hQ]: _nT }]]
];
var ListFlowVersionsResponse$ = [3, n0, _LFVRi,
    0,
    [_fVS, _nT],
    [() => FlowVersionSummaries, 0]
];
var ListIngestionJobsRequest$ = [3, n0, _LIJR,
    0,
    [_kBI, _dSI, _f, _sB, _mR, _nT],
    [[0, 1], [0, 1], () => IngestionJobFilters, () => IngestionJobSortBy$, 1, 0]
];
var ListIngestionJobsResponse$ = [3, n0, _LIJRi,
    0,
    [_iJS, _nT],
    [() => IngestionJobSummaries, 0]
];
var ListKnowledgeBaseDocumentsRequest$ = [3, n0, _LKBDR,
    0,
    [_kBI, _dSI, _mR, _nT],
    [[0, 1], [0, 1], 1, 0]
];
var ListKnowledgeBaseDocumentsResponse$ = [3, n0, _LKBDRi,
    0,
    [_dD, _nT],
    [() => KnowledgeBaseDocumentDetails, 0]
];
var ListKnowledgeBasesRequest$ = [3, n0, _LKBR,
    0,
    [_mR, _nT],
    [1, 0]
];
var ListKnowledgeBasesResponse$ = [3, n0, _LKBRi,
    0,
    [_kBSn, _nT],
    [() => KnowledgeBaseSummaries, 0]
];
var ListPromptsRequest$ = [3, n0, _LPR,
    0,
    [_pI, _mR, _nT],
    [[0, { [_hQ]: _pI }], [1, { [_hQ]: _mR }], [0, { [_hQ]: _nT }]]
];
var ListPromptsResponse$ = [3, n0, _LPRi,
    0,
    [_pS, _nT],
    [() => PromptSummaries, 0]
];
var ListTagsForResourceRequest$ = [3, n0, _LTFRR,
    0,
    [_rAe],
    [[0, 1]]
];
var ListTagsForResourceResponse$ = [3, n0, _LTFRRi,
    0,
    [_ta],
    [128 | 0]
];
var LoopControllerFlowNodeConfiguration$ = [3, n0, _LCFNC,
    0,
    [_cCont, _mIa],
    [[() => FlowCondition$, 0], 1]
];
var LoopFlowNodeConfiguration$ = [3, n0, _LFNCo,
    0,
    [_de],
    [[() => FlowDefinition$, 0]]
];
var LoopIncompatibleNodeTypeFlowValidationDetails$ = [3, n0, _LINTFVD,
    0,
    [_no, _iNT, _iNN],
    [0, 0, 0]
];
var LoopInputFlowNodeConfiguration$ = [3, n0, _LIFNC,
    0,
    [],
    []
];
var MalformedConditionExpressionFlowValidationDetails$ = [3, n0, _MCEFVD,
    0,
    [_no, _cond, _cau],
    [0, 0, 0]
];
var MalformedNodeInputExpressionFlowValidationDetails$ = [3, n0, _MNIEFVD,
    0,
    [_no, _inp, _cau],
    [0, 0, 0]
];
var MemoryConfiguration$ = [3, n0, _MC,
    0,
    [_eMT, _sDt, _sSC],
    [64 | 0, 1, () => SessionSummaryConfiguration$]
];
var Message$ = [3, n0, _M,
    0,
    [_r, _cont],
    [0, [() => ContentBlocks, 0]]
];
var MetadataAttribute$ = [3, n0, _MA,
    0,
    [_k, _value],
    [[() => Key, 0], [() => MetadataAttributeValue$, 0]]
];
var MetadataAttributeValue$ = [3, n0, _MAV,
    0,
    [_t, _nV, _bV, _sV, _sLV],
    [0, [() => NumberValue, 0], 2, [() => StringValue, 0], [() => StringListValue, 0]]
];
var MetadataConfigurationForReranking$ = [3, n0, _MCFR,
    0,
    [_sM, _sMC],
    [0, [() => RerankingMetadataSelectiveModeConfiguration$, 0]]
];
var MismatchedNodeInputTypeFlowValidationDetails$ = [3, n0, _MNITFVD,
    0,
    [_no, _inp, _eT],
    [0, 0, 0]
];
var MismatchedNodeOutputTypeFlowValidationDetails$ = [3, n0, _MNOTFVD,
    0,
    [_no, _ou, _eT],
    [0, 0, 0]
];
var MissingConnectionConfigurationFlowValidationDetails$ = [3, n0, _MCCFVD,
    0,
    [_con],
    [0]
];
var MissingDefaultConditionFlowValidationDetails$ = [3, n0, _MDCFVD,
    0,
    [_no],
    [0]
];
var MissingEndingNodesFlowValidationDetails$ = [3, n0, _MENFVD,
    0,
    [],
    []
];
var MissingLoopControllerNodeFlowValidationDetails$ = [3, n0, _MLCNFVD,
    0,
    [_lN],
    [0]
];
var MissingLoopInputNodeFlowValidationDetails$ = [3, n0, _MLINFVD,
    0,
    [_lN],
    [0]
];
var MissingNodeConfigurationFlowValidationDetails$ = [3, n0, _MNCFVD,
    0,
    [_no],
    [0]
];
var MissingNodeInputFlowValidationDetails$ = [3, n0, _MNIFVD,
    0,
    [_no, _inp],
    [0, 0]
];
var MissingNodeOutputFlowValidationDetails$ = [3, n0, _MNOFVD,
    0,
    [_no, _ou],
    [0, 0]
];
var MissingStartingNodesFlowValidationDetails$ = [3, n0, _MSNFVD,
    0,
    [],
    []
];
var MongoDbAtlasConfiguration$ = [3, n0, _MDAC,
    0,
    [_en, _dN, _cNo, _vIN, _cSA, _fMi, _eSN, _tIN],
    [0, 0, 0, 0, 0, () => MongoDbAtlasFieldMapping$, 0, 0]
];
var MongoDbAtlasFieldMapping$ = [3, n0, _MDAFM,
    0,
    [_vF, _tF, _mF],
    [0, 0, 0]
];
var MultipleLoopControllerNodesFlowValidationDetails$ = [3, n0, _MLCNFVDu,
    0,
    [_lN],
    [0]
];
var MultipleLoopInputNodesFlowValidationDetails$ = [3, n0, _MLINFVDu,
    0,
    [_lN],
    [0]
];
var MultipleNodeInputConnectionsFlowValidationDetails$ = [3, n0, _MNICFVD,
    0,
    [_no, _inp],
    [0, 0]
];
var NeptuneAnalyticsConfiguration$ = [3, n0, _NAC,
    0,
    [_gA, _fMi],
    [[() => GraphArn, 0], () => NeptuneAnalyticsFieldMapping$]
];
var NeptuneAnalyticsFieldMapping$ = [3, n0, _NAFM,
    0,
    [_tF, _mF],
    [0, 0]
];
var OpenSearchManagedClusterConfiguration$ = [3, n0, _OSMCC,
    0,
    [_dE, _dA, _vIN, _fMi],
    [0, 0, [() => OpenSearchManagedClusterIndexName, 0], () => OpenSearchManagedClusterFieldMapping$]
];
var OpenSearchManagedClusterFieldMapping$ = [3, n0, _OSMCFM,
    0,
    [_vF, _tF, _mF],
    [0, 0, 0]
];
var OpenSearchServerlessConfiguration$ = [3, n0, _OSSC,
    0,
    [_cAo, _vIN, _fMi],
    [0, 0, () => OpenSearchServerlessFieldMapping$]
];
var OpenSearchServerlessFieldMapping$ = [3, n0, _OSSFM,
    0,
    [_vF, _tF, _mF],
    [0, 0, 0]
];
var OutputFlowNodeConfiguration$ = [3, n0, _OFNC,
    0,
    [],
    []
];
var ParameterDetail$ = [3, n0, _PD,
    0,
    [_d, _t, _re],
    [0, 0, 2]
];
var ParsingConfiguration$ = [3, n0, _PC,
    0,
    [_pSa, _bFMC, _bDAC],
    [0, () => BedrockFoundationModelConfiguration$, () => BedrockDataAutomationConfiguration$]
];
var ParsingPrompt$ = [3, n0, _PP,
    0,
    [_pPT],
    [0]
];
var PatternObjectFilter$ = [3, n0, _POF,
    0,
    [_oTb, _iF, _eF],
    [[() => FilteredObjectType, 0], [() => FilterList, 0], [() => FilterList, 0]]
];
var PatternObjectFilterConfiguration$ = [3, n0, _POFC,
    0,
    [_f],
    [[() => PatternObjectFilterList, 0]]
];
var PerformanceConfiguration$ = [3, n0, _PCe,
    0,
    [_la],
    [0]
];
var PineconeConfiguration$ = [3, n0, _PCi,
    0,
    [_cSo, _cSA, _na, _fMi],
    [0, 0, 0, () => PineconeFieldMapping$]
];
var PineconeFieldMapping$ = [3, n0, _PFM,
    0,
    [_tF, _mF],
    [0, 0]
];
var PrepareAgentRequest$ = [3, n0, _PAR,
    0,
    [_aI],
    [[0, 1]]
];
var PrepareAgentResponse$ = [3, n0, _PARr,
    0,
    [_aI, _aS, _aV, _pA],
    [0, 0, 0, 5]
];
var PrepareFlowRequest$ = [3, n0, _PFR,
    0,
    [_fI],
    [[0, 1]]
];
var PrepareFlowResponse$ = [3, n0, _PFRr,
    0,
    [_id, _st],
    [0, 0]
];
var PromptAgentResource$ = [3, n0, _PARro,
    8,
    [_aIg],
    [0]
];
var PromptConfiguration$ = [3, n0, _PCr,
    0,
    [_pTro, _pCM, _pSr, _bPT, _iCn, _pMa, _fM, _aMRF],
    [0, 0, 0, [() => BasePromptTemplate, 0], () => InferenceConfiguration$, 0, 0, 15]
];
var PromptFlowNodeConfiguration$ = [3, n0, _PFNC,
    0,
    [_sCo, _gC],
    [[() => PromptFlowNodeSourceConfiguration$, 0], () => GuardrailConfiguration$]
];
var PromptFlowNodeInlineConfiguration$ = [3, n0, _PFNIC,
    0,
    [_tT, _tCem, _mI, _iCn, _aMRF],
    [0, [() => PromptTemplateConfiguration$, 0], 0, () => PromptInferenceConfiguration$, 15]
];
var PromptFlowNodeResourceConfiguration$ = [3, n0, _PFNRC,
    0,
    [_pAr],
    [0]
];
var PromptInputVariable$ = [3, n0, _PIV,
    0,
    [_n],
    [0]
];
var PromptMetadataEntry$ = [3, n0, _PME,
    8,
    [_k, _value],
    [[() => PromptMetadataKey, 0], [() => PromptMetadataValue, 0]]
];
var PromptModelInferenceConfiguration$ = [3, n0, _PMIC,
    0,
    [_te, _tP, _mTa, _sS],
    [1, 1, 1, 64 | 0]
];
var PromptOverrideConfiguration$ = [3, n0, _POC,
    8,
    [_pCr, _oL],
    [[() => PromptConfigurations, 0], 0]
];
var PromptSummary$ = [3, n0, _PS,
    0,
    [_n, _d, _id, _ar, _v, _cA, _uA],
    [0, 0, 0, 0, 0, 5, 5]
];
var PromptVariant$ = [3, n0, _PV,
    8,
    [_n, _tT, _tCem, _mI, _iCn, _meta, _aMRF, _gAR],
    [0, 0, [() => PromptTemplateConfiguration$, 0], 0, () => PromptInferenceConfiguration$, [() => PromptMetadataList, 0], 15, [() => PromptGenAiResource$, 0]]
];
var QueryGenerationColumn$ = [3, n0, _QGC,
    0,
    [_n, _d, _inc],
    [0, 0, 0]
];
var QueryGenerationConfiguration$ = [3, n0, _QGCu,
    0,
    [_eTS, _gCe],
    [1, [() => QueryGenerationContext$, 0]]
];
var QueryGenerationContext$ = [3, n0, _QGCue,
    8,
    [_tab, _cQ],
    [() => QueryGenerationTables, () => CuratedQueries]
];
var QueryGenerationTable$ = [3, n0, _QGT,
    0,
    [_n, _d, _inc, _col],
    [0, 0, 0, () => QueryGenerationColumns]
];
var RdsConfiguration$ = [3, n0, _RC,
    0,
    [_rAe, _cSA, _dN, _tN, _fMi],
    [0, 0, 0, 0, () => RdsFieldMapping$]
];
var RdsFieldMapping$ = [3, n0, _RFM,
    0,
    [_pKF, _vF, _tF, _mF, _cMF],
    [0, 0, 0, 0, 0]
];
var RedisEnterpriseCloudConfiguration$ = [3, n0, _RECC,
    0,
    [_en, _vIN, _cSA, _fMi],
    [0, 0, 0, () => RedisEnterpriseCloudFieldMapping$]
];
var RedisEnterpriseCloudFieldMapping$ = [3, n0, _RECFM,
    0,
    [_vF, _tF, _mF],
    [0, 0, 0]
];
var RedshiftConfiguration$ = [3, n0, _RCe,
    0,
    [_sCto, _qEC, _qGC],
    [() => RedshiftQueryEngineStorageConfigurations, () => RedshiftQueryEngineConfiguration$, [() => QueryGenerationConfiguration$, 0]]
];
var RedshiftProvisionedAuthConfiguration$ = [3, n0, _RPAC,
    0,
    [_t, _dU, _uPSA],
    [0, 0, 0]
];
var RedshiftProvisionedConfiguration$ = [3, n0, _RPC,
    0,
    [_cIl, _aCu],
    [0, () => RedshiftProvisionedAuthConfiguration$]
];
var RedshiftQueryEngineAwsDataCatalogStorageConfiguration$ = [3, n0, _RQEADCSC,
    0,
    [_tNa],
    [64 | 0]
];
var RedshiftQueryEngineConfiguration$ = [3, n0, _RQEC,
    0,
    [_t, _sCe, _pCro],
    [0, () => RedshiftServerlessConfiguration$, () => RedshiftProvisionedConfiguration$]
];
var RedshiftQueryEngineRedshiftStorageConfiguration$ = [3, n0, _RQERSC,
    0,
    [_dN],
    [0]
];
var RedshiftQueryEngineStorageConfiguration$ = [3, n0, _RQESC,
    0,
    [_t, _aDCC, _rCed],
    [0, () => RedshiftQueryEngineAwsDataCatalogStorageConfiguration$, () => RedshiftQueryEngineRedshiftStorageConfiguration$]
];
var RedshiftServerlessAuthConfiguration$ = [3, n0, _RSAC,
    0,
    [_t, _uPSA],
    [0, 0]
];
var RedshiftServerlessConfiguration$ = [3, n0, _RSC,
    0,
    [_wA, _aCu],
    [0, () => RedshiftServerlessAuthConfiguration$]
];
var ResourceNotFoundException$ = [-3, n0, _RNFE,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(ResourceNotFoundException$, ResourceNotFoundException);
var RetrievalFlowNodeConfiguration$ = [3, n0, _RFNC,
    0,
    [_sCer],
    [() => RetrievalFlowNodeServiceConfiguration$]
];
var RetrievalFlowNodeS3Configuration$ = [3, n0, _RFNSC,
    0,
    [_bN],
    [0]
];
var S3Content$ = [3, n0, _SC,
    0,
    [_sL],
    [() => S3Location$]
];
var S3DataSourceConfiguration$ = [3, n0, _SDSC,
    0,
    [_bA, _iP, _bOAI],
    [0, [() => S3Prefixes, 0], 0]
];
var S3Identifier$ = [3, n0, _SI,
    0,
    [_sBN, _sOK],
    [0, 0]
];
var S3Location$ = [3, n0, _SL,
    0,
    [_u],
    [0]
];
var S3VectorsConfiguration$ = [3, n0, _SVC,
    0,
    [_vBA, _iAn, _iN],
    [[() => VectorBucketArn, 0], [() => IndexArn, 0], [() => IndexName, 0]]
];
var SalesforceCrawlerConfiguration$ = [3, n0, _SCC,
    0,
    [_fC],
    [[() => CrawlFilterConfiguration$, 0]]
];
var SalesforceDataSourceConfiguration$ = [3, n0, _SDSCa,
    0,
    [_sCo, _cC],
    [() => SalesforceSourceConfiguration$, [() => SalesforceCrawlerConfiguration$, 0]]
];
var SalesforceSourceConfiguration$ = [3, n0, _SSC,
    0,
    [_hU, _aT, _cSA],
    [0, 0, 0]
];
var SeedUrl$ = [3, n0, _SU,
    0,
    [_ur],
    [0]
];
var SemanticChunkingConfiguration$ = [3, n0, _SCCe,
    0,
    [_mTa, _bS, _bPTr],
    [1, 1, 1]
];
var ServerSideEncryptionConfiguration$ = [3, n0, _SSEC,
    0,
    [_kKA],
    [0]
];
var ServiceQuotaExceededException$ = [-3, n0, _SQEE,
    { [_e]: _c, [_hE]: 402 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(ServiceQuotaExceededException$, ServiceQuotaExceededException);
var SessionSummaryConfiguration$ = [3, n0, _SSCe,
    0,
    [_mRS],
    [1]
];
var SharePointCrawlerConfiguration$ = [3, n0, _SPCC,
    0,
    [_fC],
    [[() => CrawlFilterConfiguration$, 0]]
];
var SharePointDataSourceConfiguration$ = [3, n0, _SPDSC,
    0,
    [_sCo, _cC],
    [() => SharePointSourceConfiguration$, [() => SharePointCrawlerConfiguration$, 0]]
];
var SharePointSourceConfiguration$ = [3, n0, _SPSC,
    0,
    [_tIe, _dom, _sU, _hT, _aT, _cSA],
    [0, 0, 64 | 0, 0, 0, 0]
];
var SpecificToolChoice$ = [3, n0, _STC,
    0,
    [_n],
    [0]
];
var SqlKnowledgeBaseConfiguration$ = [3, n0, _SKBC,
    0,
    [_t, _rCed],
    [0, [() => RedshiftConfiguration$, 0]]
];
var StartIngestionJobRequest$ = [3, n0, _SIJR,
    0,
    [_kBI, _dSI, _cT, _d],
    [[0, 1], [0, 1], [0, 4], 0]
];
var StartIngestionJobResponse$ = [3, n0, _SIJRt,
    0,
    [_iJ],
    [() => IngestionJob$]
];
var StopIngestionJobRequest$ = [3, n0, _SIJRto,
    0,
    [_kBI, _dSI, _iJI],
    [[0, 1], [0, 1], [0, 1]]
];
var StopIngestionJobResponse$ = [3, n0, _SIJRtop,
    0,
    [_iJ],
    [() => IngestionJob$]
];
var StorageConfiguration$ = [3, n0, _SCt,
    0,
    [_t, _oSC, _oMCC, _pCi, _rECC, _rCd, _mDAC, _nAC, _sVC],
    [0, () => OpenSearchServerlessConfiguration$, [() => OpenSearchManagedClusterConfiguration$, 0], () => PineconeConfiguration$, () => RedisEnterpriseCloudConfiguration$, () => RdsConfiguration$, () => MongoDbAtlasConfiguration$, [() => NeptuneAnalyticsConfiguration$, 0], [() => S3VectorsConfiguration$, 0]]
];
var StorageFlowNodeConfiguration$ = [3, n0, _SFNC,
    0,
    [_sCer],
    [() => StorageFlowNodeServiceConfiguration$]
];
var StorageFlowNodeS3Configuration$ = [3, n0, _SFNSC,
    0,
    [_bN],
    [0]
];
var SupplementalDataStorageConfiguration$ = [3, n0, _SDSCu,
    0,
    [_sLt],
    [() => SupplementalDataStorageLocations]
];
var SupplementalDataStorageLocation$ = [3, n0, _SDSL,
    0,
    [_t, _sL],
    [0, () => S3Location$]
];
var TagResourceRequest$ = [3, n0, _TRR,
    0,
    [_rAe, _ta],
    [[0, 1], 128 | 0]
];
var TagResourceResponse$ = [3, n0, _TRRa,
    0,
    [],
    []
];
var TextContentDoc$ = [3, n0, _TCD,
    0,
    [_da],
    [[() => Data, 0]]
];
var TextPromptTemplateConfiguration$ = [3, n0, _TPTC,
    8,
    [_tex, _cP, _iV],
    [[() => TextPrompt, 0], () => CachePointBlock$, [() => PromptInputVariablesList, 0]]
];
var ThrottlingException$ = [-3, n0, _TE,
    { [_e]: _c, [_hE]: 429 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(ThrottlingException$, ThrottlingException);
var ToolConfiguration$ = [3, n0, _TC,
    0,
    [_to, _tCo],
    [[() => Tools, 0], [() => ToolChoice$, 0]]
];
var ToolSpecification$ = [3, n0, _TS,
    0,
    [_n, _d, _iSn],
    [0, 0, () => ToolInputSchema$]
];
var Transformation$ = [3, n0, _T,
    0,
    [_tFr, _sTA],
    [() => TransformationFunction$, 0]
];
var TransformationFunction$ = [3, n0, _TF,
    0,
    [_tLC],
    [() => TransformationLambdaConfiguration$]
];
var TransformationLambdaConfiguration$ = [3, n0, _TLC,
    0,
    [_lA],
    [0]
];
var UnfulfilledNodeInputFlowValidationDetails$ = [3, n0, _UNIFVD,
    0,
    [_no, _inp],
    [0, 0]
];
var UnknownConnectionConditionFlowValidationDetails$ = [3, n0, _UCCFVD,
    0,
    [_con],
    [0]
];
var UnknownConnectionSourceFlowValidationDetails$ = [3, n0, _UCSFVD,
    0,
    [_con],
    [0]
];
var UnknownConnectionSourceOutputFlowValidationDetails$ = [3, n0, _UCSOFVD,
    0,
    [_con],
    [0]
];
var UnknownConnectionTargetFlowValidationDetails$ = [3, n0, _UCTFVD,
    0,
    [_con],
    [0]
];
var UnknownConnectionTargetInputFlowValidationDetails$ = [3, n0, _UCTIFVD,
    0,
    [_con],
    [0]
];
var UnknownNodeInputFlowValidationDetails$ = [3, n0, _UNIFVDn,
    0,
    [_no, _inp],
    [0, 0]
];
var UnknownNodeOutputFlowValidationDetails$ = [3, n0, _UNOFVD,
    0,
    [_no, _ou],
    [0, 0]
];
var UnreachableNodeFlowValidationDetails$ = [3, n0, _UNFVD,
    0,
    [_no],
    [0]
];
var UnsatisfiedConnectionConditionsFlowValidationDetails$ = [3, n0, _UCCFVDn,
    0,
    [_con],
    [0]
];
var UnspecifiedFlowValidationDetails$ = [3, n0, _UFVD,
    0,
    [],
    []
];
var UntagResourceRequest$ = [3, n0, _URR,
    0,
    [_rAe, _tKa],
    [[0, 1], [64 | 0, { [_hQ]: _tKa }]]
];
var UntagResourceResponse$ = [3, n0, _URRn,
    0,
    [],
    []
];
var UpdateAgentActionGroupRequest$ = [3, n0, _UAAGR,
    0,
    [_aI, _aV, _aGI, _aGN, _d, _pAGS, _pAGSP, _aGE, _aGS, _aSp, _fS],
    [[0, 1], [0, 1], [0, 1], 0, 0, 0, 128 | 0, () => ActionGroupExecutor$, 0, [() => APISchema$, 0], () => FunctionSchema$]
];
var UpdateAgentActionGroupResponse$ = [3, n0, _UAAGRp,
    0,
    [_aAG],
    [[() => AgentActionGroup$, 0]]
];
var UpdateAgentAliasRequest$ = [3, n0, _UAAR,
    0,
    [_aI, _aAI, _aAN, _d, _rC, _aIS],
    [[0, 1], [0, 1], 0, 0, () => AgentAliasRoutingConfiguration, 0]
];
var UpdateAgentAliasResponse$ = [3, n0, _UAARp,
    0,
    [_aAg],
    [() => AgentAlias$]
];
var UpdateAgentCollaboratorRequest$ = [3, n0, _UACR,
    0,
    [_aI, _aV, _cI, _aD, _cN, _cIo, _rCH],
    [[0, 1], [0, 1], [0, 1], () => AgentDescriptor$, 0, [() => CollaborationInstruction, 0], 0]
];
var UpdateAgentCollaboratorResponse$ = [3, n0, _UACRp,
    0,
    [_aCg],
    [[() => AgentCollaborator$, 0]]
];
var UpdateAgentKnowledgeBaseRequest$ = [3, n0, _UAKBR,
    0,
    [_aI, _aV, _kBI, _d, _kBS],
    [[0, 1], [0, 1], [0, 1], 0, 0]
];
var UpdateAgentKnowledgeBaseResponse$ = [3, n0, _UAKBRp,
    0,
    [_aKB],
    [() => AgentKnowledgeBase$]
];
var UpdateAgentRequest$ = [3, n0, _UAR,
    0,
    [_aI, _aN, _i, _fM, _d, _oT, _cO, _iSTTLIS, _aRRA, _cEKA, _pOC, _gC, _mC, _aC],
    [[0, 1], 0, [() => Instruction, 0], 0, 0, 0, () => CustomOrchestration$, 1, 0, 0, [() => PromptOverrideConfiguration$, 0], () => GuardrailConfiguration$, () => MemoryConfiguration$, 0]
];
var UpdateAgentResponse$ = [3, n0, _UARp,
    0,
    [_ag],
    [[() => Agent$, 0]]
];
var UpdateDataSourceRequest$ = [3, n0, _UDSR,
    0,
    [_kBI, _dSI, _n, _d, _dSC, _dDP, _sSEC, _vIC],
    [[0, 1], [0, 1], 0, 0, [() => DataSourceConfiguration$, 0], 0, () => ServerSideEncryptionConfiguration$, () => VectorIngestionConfiguration$]
];
var UpdateDataSourceResponse$ = [3, n0, _UDSRp,
    0,
    [_dS],
    [[() => DataSource$, 0]]
];
var UpdateFlowAliasRequest$ = [3, n0, _UFAR,
    0,
    [_n, _d, _rC, _cCo, _fI, _aIl],
    [0, 0, () => FlowAliasRoutingConfiguration, () => FlowAliasConcurrencyConfiguration$, [0, 1], [0, 1]]
];
var UpdateFlowAliasResponse$ = [3, n0, _UFARp,
    0,
    [_n, _d, _rC, _cCo, _fIl, _id, _ar, _cA, _uA],
    [0, 0, () => FlowAliasRoutingConfiguration, () => FlowAliasConcurrencyConfiguration$, 0, 0, 0, 5, 5]
];
var UpdateFlowRequest$ = [3, n0, _UFR,
    0,
    [_n, _d, _eRA, _cEKA, _de, _fI],
    [0, 0, 0, 0, [() => FlowDefinition$, 0], [0, 1]]
];
var UpdateFlowResponse$ = [3, n0, _UFRp,
    0,
    [_n, _d, _eRA, _cEKA, _id, _ar, _st, _cA, _uA, _v, _de],
    [0, 0, 0, 0, 0, 0, 0, 5, 5, 0, [() => FlowDefinition$, 0]]
];
var UpdateKnowledgeBaseRequest$ = [3, n0, _UKBR,
    0,
    [_kBI, _n, _d, _rAo, _kBC, _sCt],
    [[0, 1], 0, 0, 0, [() => KnowledgeBaseConfiguration$, 0], [() => StorageConfiguration$, 0]]
];
var UpdateKnowledgeBaseResponse$ = [3, n0, _UKBRp,
    0,
    [_kB],
    [[() => KnowledgeBase$, 0]]
];
var UpdatePromptRequest$ = [3, n0, _UPR,
    0,
    [_n, _d, _cEKA, _dV, _va, _pI],
    [0, 0, 0, 0, [() => PromptVariantList, 0], [0, 1]]
];
var UpdatePromptResponse$ = [3, n0, _UPRp,
    0,
    [_n, _d, _cEKA, _dV, _va, _id, _ar, _v, _cA, _uA],
    [0, 0, 0, 0, [() => PromptVariantList, 0], 0, 0, 0, 5, 5]
];
var UrlConfiguration$ = [3, n0, _UC,
    0,
    [_sUe],
    [() => SeedUrls]
];
var ValidateFlowDefinitionRequest$ = [3, n0, _VFDR,
    0,
    [_de],
    [[() => FlowDefinition$, 0]]
];
var ValidateFlowDefinitionResponse$ = [3, n0, _VFDRa,
    0,
    [_val],
    [[() => FlowValidations, 0]]
];
var ValidationException$ = [-3, n0, _VE,
    { [_e]: _c, [_hE]: 400 },
    [_m, _fL],
    [0, () => ValidationExceptionFieldList]
];
schema.TypeRegistry.for(n0).registerError(ValidationException$, ValidationException);
var ValidationExceptionField$ = [3, n0, _VEF,
    0,
    [_n, _m],
    [0, 0]
];
var VectorIngestionConfiguration$ = [3, n0, _VIC,
    0,
    [_cCh, _cTC, _pCa, _cEC],
    [() => ChunkingConfiguration$, () => CustomTransformationConfiguration$, () => ParsingConfiguration$, () => ContextEnrichmentConfiguration$]
];
var VectorKnowledgeBaseConfiguration$ = [3, n0, _VKBC,
    0,
    [_eMA, _eMC, _sDSC],
    [0, () => EmbeddingModelConfiguration$, () => SupplementalDataStorageConfiguration$]
];
var VectorSearchBedrockRerankingConfiguration$ = [3, n0, _VSBRC,
    0,
    [_mCo, _nORR, _mCe],
    [() => VectorSearchBedrockRerankingModelConfiguration$, 1, [() => MetadataConfigurationForReranking$, 0]]
];
var VectorSearchBedrockRerankingModelConfiguration$ = [3, n0, _VSBRMC,
    0,
    [_mA, _aMRF],
    [0, 128 | 15]
];
var VectorSearchRerankingConfiguration$ = [3, n0, _VSRC,
    0,
    [_t, _bRC],
    [0, [() => VectorSearchBedrockRerankingConfiguration$, 0]]
];
var VideoConfiguration$ = [3, n0, _VC,
    0,
    [_sC],
    [() => VideoSegmentationConfiguration$]
];
var VideoSegmentationConfiguration$ = [3, n0, _VSC,
    0,
    [_fLD],
    [1]
];
var WebCrawlerConfiguration$ = [3, n0, _WCC,
    0,
    [_cL, _iF, _eF, _sc, _uAs, _uAH],
    [() => WebCrawlerLimits$, [() => FilterList, 0], [() => FilterList, 0], 0, [() => UserAgent, 0], [() => UserAgentHeader, 0]]
];
var WebCrawlerLimits$ = [3, n0, _WCL,
    0,
    [_rL, _mP],
    [1, 1]
];
var WebDataSourceConfiguration$ = [3, n0, _WDSC,
    0,
    [_sCo, _cC],
    [() => WebSourceConfiguration$, [() => WebCrawlerConfiguration$, 0]]
];
var WebSourceConfiguration$ = [3, n0, _WSC,
    0,
    [_uC],
    [() => UrlConfiguration$]
];
var BedrockAgentServiceException$ = [-3, _sm, "BedrockAgentServiceException", 0, [], []];
schema.TypeRegistry.for(_sm).registerError(BedrockAgentServiceException$, BedrockAgentServiceException);
var ActionGroupSummaries = [1, n0, _AGSc,
    0, () => ActionGroupSummary$
];
var AgentAliasHistoryEvents = [1, n0, _AAHEg,
    0, () => AgentAliasHistoryEvent$
];
var AgentAliasRoutingConfiguration = [1, n0, _AARC,
    0, () => AgentAliasRoutingConfigurationListItem$
];
var AgentAliasSummaries = [1, n0, _AASg,
    0, () => AgentAliasSummary$
];
var AgentCollaboratorSummaries = [1, n0, _ACSg,
    0, [() => AgentCollaboratorSummary$,
        0]
];
var AgentKnowledgeBaseSummaries = [1, n0, _AKBSg,
    0, () => AgentKnowledgeBaseSummary$
];
var AgentSummaries = [1, n0, _ASg,
    0, () => AgentSummary$
];
var AgentVersionSummaries = [1, n0, _AVSg,
    0, () => AgentVersionSummary$
];
var AudioConfigurations = [1, n0, _ACud,
    0, () => AudioConfiguration$
];
var ContentBlocks = [1, n0, _CB,
    0, [() => ContentBlock$,
        0]
];
var CuratedQueries = [1, n0, _CQu,
    0, () => CuratedQuery$
];
var DataSourceSummaries = [1, n0, _DSSa,
    0, () => DataSourceSummary$
];
var DocumentIdentifiers = [1, n0, _DIo,
    0, () => DocumentIdentifier$
];
var FieldsForReranking = [1, n0, _FFRi,
    8, () => FieldForReranking$
];
var FilterList = [1, n0, _FL,
    8, [() => FilterPattern,
        0]
];
var FlowAliasRoutingConfiguration = [1, n0, _FARC,
    0, () => FlowAliasRoutingConfigurationListItem$
];
var FlowAliasSummaries = [1, n0, _FASl,
    0, () => FlowAliasSummary$
];
var FlowConditions = [1, n0, _FClo,
    0, [() => FlowCondition$,
        0]
];
var FlowConnections = [1, n0, _FClow,
    0, () => FlowConnection$
];
var FlowNodeInputs = [1, n0, _FNIl,
    0, [() => FlowNodeInput$,
        0]
];
var FlowNodeOutputs = [1, n0, _FNOl,
    0, () => FlowNodeOutput$
];
var FlowNodes = [1, n0, _FNl,
    0, [() => FlowNode$,
        0]
];
var FlowSummaries = [1, n0, _FSl,
    0, () => FlowSummary$
];
var FlowValidations = [1, n0, _FVl,
    0, [() => FlowValidation$,
        0]
];
var FlowVersionSummaries = [1, n0, _FVSl,
    0, () => FlowVersionSummary$
];
var Functions = [1, n0, _Fu,
    0, () => Function$
];
var HierarchicalChunkingLevelConfigurations = [1, n0, _HCLCi,
    0, () => HierarchicalChunkingLevelConfiguration$
];
var IngestionJobFilters = [1, n0, _IJFn,
    0, () => IngestionJobFilter$
];
var IngestionJobSummaries = [1, n0, _IJSng,
    0, () => IngestionJobSummary$
];
var KnowledgeBaseDocumentDetails = [1, n0, _KBDDn,
    0, () => KnowledgeBaseDocumentDetail$
];
var KnowledgeBaseDocuments = [1, n0, _KBDn,
    0, [() => KnowledgeBaseDocument$,
        0]
];
var KnowledgeBaseSummaries = [1, n0, _KBSn,
    0, () => KnowledgeBaseSummary$
];
var Messages = [1, n0, _Me,
    0, [() => Message$,
        0]
];
var MetadataAttributes = [1, n0, _MAe,
    0, [() => MetadataAttribute$,
        0]
];
var PatternObjectFilterList = [1, n0, _POFL,
    8, [() => PatternObjectFilter$,
        0]
];
var PromptConfigurations = [1, n0, _PCro,
    0, [() => PromptConfiguration$,
        0]
];
var PromptInputVariablesList = [1, n0, _PIVL,
    8, () => PromptInputVariable$
];
var PromptMetadataList = [1, n0, _PML,
    8, [() => PromptMetadataEntry$,
        0]
];
var PromptSummaries = [1, n0, _PSr,
    0, () => PromptSummary$
];
var PromptVariantList = [1, n0, _PVL,
    8, [() => PromptVariant$,
        0]
];
var QueryGenerationColumns = [1, n0, _QGCuer,
    0, () => QueryGenerationColumn$
];
var QueryGenerationTables = [1, n0, _QGTu,
    0, () => QueryGenerationTable$
];
var RedshiftQueryEngineStorageConfigurations = [1, n0, _RQESCe,
    0, () => RedshiftQueryEngineStorageConfiguration$
];
var S3Prefixes = [1, n0, _SPr,
    0, [() => S3Prefix,
        0]
];
var SeedUrls = [1, n0, _SUe,
    0, () => SeedUrl$
];
var StringListValue = [1, n0, _SLV,
    0, [() => StringValue,
        0]
];
var SupplementalDataStorageLocations = [1, n0, _SDSLu,
    0, () => SupplementalDataStorageLocation$
];
var SystemContentBlocks = [1, n0, _SCB,
    0, [() => SystemContentBlock$,
        0]
];
var Tools = [1, n0, _To,
    8, () => Tool$
];
var Transformations = [1, n0, _Tr,
    0, () => Transformation$
];
var ValidationExceptionFieldList = [1, n0, _VEFL,
    0, () => ValidationExceptionField$
];
var VideoConfigurations = [1, n0, _VCi,
    0, () => VideoConfiguration$
];
var ParameterMap = [2, n0, _PM,
    0, 0, () => ParameterDetail$
];
var ActionGroupExecutor$ = [4, n0, _AGE,
    0,
    [_lam, _cCu],
    [0, 0]
];
var APISchema$ = [4, n0, _APIS,
    0,
    [_s_, _pa],
    [() => S3Identifier$, [() => Payload, 0]]
];
var ContentBlock$ = [4, n0, _CBo,
    8,
    [_tex, _cP],
    [0, () => CachePointBlock$]
];
var FlowConnectionConfiguration$ = [4, n0, _FCC,
    0,
    [_da, _condi],
    [() => FlowDataConnectionConfiguration$, () => FlowConditionalConnectionConfiguration$]
];
var FlowNodeConfiguration$ = [4, n0, _FNC,
    0,
    [_inp, _ou, _kB, _cond, _le, _pr, _lF, _sto, _ag, _ret, _it, _coll, _iCnl, _lo, _lIo, _lCo],
    [() => InputFlowNodeConfiguration$, () => OutputFlowNodeConfiguration$, [() => KnowledgeBaseFlowNodeConfiguration$, 0], [() => ConditionFlowNodeConfiguration$, 0], () => LexFlowNodeConfiguration$, [() => PromptFlowNodeConfiguration$, 0], () => LambdaFunctionFlowNodeConfiguration$, () => StorageFlowNodeConfiguration$, () => AgentFlowNodeConfiguration$, () => RetrievalFlowNodeConfiguration$, () => IteratorFlowNodeConfiguration$, () => CollectorFlowNodeConfiguration$, [() => InlineCodeFlowNodeConfiguration$, 0], [() => LoopFlowNodeConfiguration$, 0], () => LoopInputFlowNodeConfiguration$, [() => LoopControllerFlowNodeConfiguration$, 0]]
];
var FlowValidationDetails$ = [4, n0, _FVD,
    0,
    [_cCy, _dC, _dCE, _uN, _uCS, _uCSO, _uCT, _uCTI, _uCC, _mCE, _mNIE, _mNIT, _mNOT, _iCDT, _mCC, _mDC, _mEN, _mNC, _mNI, _mNO, _mSN, _mNIC, _uNI, _uCCn, _un, _uNIn, _uNO, _mLIN, _mLCN, _mLINu, _mLCNu, _lINT, _iLB],
    [() => CyclicConnectionFlowValidationDetails$, () => DuplicateConnectionsFlowValidationDetails$, [() => DuplicateConditionExpressionFlowValidationDetails$, 0], () => UnreachableNodeFlowValidationDetails$, () => UnknownConnectionSourceFlowValidationDetails$, () => UnknownConnectionSourceOutputFlowValidationDetails$, () => UnknownConnectionTargetFlowValidationDetails$, () => UnknownConnectionTargetInputFlowValidationDetails$, () => UnknownConnectionConditionFlowValidationDetails$, () => MalformedConditionExpressionFlowValidationDetails$, () => MalformedNodeInputExpressionFlowValidationDetails$, () => MismatchedNodeInputTypeFlowValidationDetails$, () => MismatchedNodeOutputTypeFlowValidationDetails$, () => IncompatibleConnectionDataTypeFlowValidationDetails$, () => MissingConnectionConfigurationFlowValidationDetails$, () => MissingDefaultConditionFlowValidationDetails$, () => MissingEndingNodesFlowValidationDetails$, () => MissingNodeConfigurationFlowValidationDetails$, () => MissingNodeInputFlowValidationDetails$, () => MissingNodeOutputFlowValidationDetails$, () => MissingStartingNodesFlowValidationDetails$, () => MultipleNodeInputConnectionsFlowValidationDetails$, () => UnfulfilledNodeInputFlowValidationDetails$, () => UnsatisfiedConnectionConditionsFlowValidationDetails$, () => UnspecifiedFlowValidationDetails$, () => UnknownNodeInputFlowValidationDetails$, () => UnknownNodeOutputFlowValidationDetails$, () => MissingLoopInputNodeFlowValidationDetails$, () => MissingLoopControllerNodeFlowValidationDetails$, () => MultipleLoopInputNodesFlowValidationDetails$, () => MultipleLoopControllerNodesFlowValidationDetails$, () => LoopIncompatibleNodeTypeFlowValidationDetails$, () => InvalidLoopBoundaryFlowValidationDetails$]
];
var FunctionSchema$ = [4, n0, _FSu,
    0,
    [_fu],
    [() => Functions]
];
var OrchestrationExecutor$ = [4, n0, _OE,
    0,
    [_lam],
    [0]
];
var PromptFlowNodeSourceConfiguration$ = [4, n0, _PFNSC,
    0,
    [_res, _inl],
    [() => PromptFlowNodeResourceConfiguration$, [() => PromptFlowNodeInlineConfiguration$, 0]]
];
var PromptGenAiResource$ = [4, n0, _PGAR,
    8,
    [_ag],
    [[() => PromptAgentResource$, 0]]
];
var PromptInferenceConfiguration$ = [4, n0, _PIC,
    0,
    [_tex],
    [() => PromptModelInferenceConfiguration$]
];
var PromptTemplateConfiguration$ = [4, n0, _PTC,
    0,
    [_tex, _ch],
    [[() => TextPromptTemplateConfiguration$, 0], [() => ChatPromptTemplateConfiguration$, 0]]
];
var RerankingMetadataSelectiveModeConfiguration$ = [4, n0, _RMSMC,
    0,
    [_fTI, _fTE],
    [[() => FieldsForReranking, 0], [() => FieldsForReranking, 0]]
];
var RetrievalFlowNodeServiceConfiguration$ = [4, n0, _RFNSCe,
    0,
    [_s_],
    [() => RetrievalFlowNodeS3Configuration$]
];
var StorageFlowNodeServiceConfiguration$ = [4, n0, _SFNSCt,
    0,
    [_s_],
    [() => StorageFlowNodeS3Configuration$]
];
var SystemContentBlock$ = [4, n0, _SCBy,
    8,
    [_tex, _cP],
    [0, () => CachePointBlock$]
];
var Tool$ = [4, n0, _Too,
    0,
    [_tS, _cP],
    [() => ToolSpecification$, () => CachePointBlock$]
];
var ToolChoice$ = [4, n0, _TCo,
    8,
    [_au, _an, _too],
    [() => AutoToolChoice$, () => AnyToolChoice$, () => SpecificToolChoice$]
];
var ToolInputSchema$ = [4, n0, _TIS,
    0,
    [_j],
    [15]
];
var AssociateAgentCollaborator$ = [9, n0, _AAC,
    { [_h]: ["PUT", "/agents/{agentId}/agentversions/{agentVersion}/agentcollaborators/", 200] }, () => AssociateAgentCollaboratorRequest$, () => AssociateAgentCollaboratorResponse$
];
var AssociateAgentKnowledgeBase$ = [9, n0, _AAKB,
    { [_h]: ["PUT", "/agents/{agentId}/agentversions/{agentVersion}/knowledgebases/", 200] }, () => AssociateAgentKnowledgeBaseRequest$, () => AssociateAgentKnowledgeBaseResponse$
];
var CreateAgent$ = [9, n0, _CA,
    { [_h]: ["PUT", "/agents/", 202] }, () => CreateAgentRequest$, () => CreateAgentResponse$
];
var CreateAgentActionGroup$ = [9, n0, _CAAG,
    { [_h]: ["PUT", "/agents/{agentId}/agentversions/{agentVersion}/actiongroups/", 200] }, () => CreateAgentActionGroupRequest$, () => CreateAgentActionGroupResponse$
];
var CreateAgentAlias$ = [9, n0, _CAA,
    { [_h]: ["PUT", "/agents/{agentId}/agentaliases/", 202] }, () => CreateAgentAliasRequest$, () => CreateAgentAliasResponse$
];
var CreateDataSource$ = [9, n0, _CDS,
    { [_h]: ["PUT", "/knowledgebases/{knowledgeBaseId}/datasources/", 200] }, () => CreateDataSourceRequest$, () => CreateDataSourceResponse$
];
var CreateFlow$ = [9, n0, _CF,
    { [_h]: ["POST", "/flows/", 201] }, () => CreateFlowRequest$, () => CreateFlowResponse$
];
var CreateFlowAlias$ = [9, n0, _CFA,
    { [_h]: ["POST", "/flows/{flowIdentifier}/aliases", 201] }, () => CreateFlowAliasRequest$, () => CreateFlowAliasResponse$
];
var CreateFlowVersion$ = [9, n0, _CFV,
    { [_h]: ["POST", "/flows/{flowIdentifier}/versions", 201] }, () => CreateFlowVersionRequest$, () => CreateFlowVersionResponse$
];
var CreateKnowledgeBase$ = [9, n0, _CKB,
    { [_h]: ["PUT", "/knowledgebases/", 202] }, () => CreateKnowledgeBaseRequest$, () => CreateKnowledgeBaseResponse$
];
var CreatePrompt$ = [9, n0, _CP,
    { [_h]: ["POST", "/prompts/", 201] }, () => CreatePromptRequest$, () => CreatePromptResponse$
];
var CreatePromptVersion$ = [9, n0, _CPV,
    { [_h]: ["POST", "/prompts/{promptIdentifier}/versions", 201] }, () => CreatePromptVersionRequest$, () => CreatePromptVersionResponse$
];
var DeleteAgent$ = [9, n0, _DA,
    { [_h]: ["DELETE", "/agents/{agentId}/", 202] }, () => DeleteAgentRequest$, () => DeleteAgentResponse$
];
var DeleteAgentActionGroup$ = [9, n0, _DAAG,
    { [_h]: ["DELETE", "/agents/{agentId}/agentversions/{agentVersion}/actiongroups/{actionGroupId}/", 204] }, () => DeleteAgentActionGroupRequest$, () => DeleteAgentActionGroupResponse$
];
var DeleteAgentAlias$ = [9, n0, _DAA,
    { [_h]: ["DELETE", "/agents/{agentId}/agentaliases/{agentAliasId}/", 202] }, () => DeleteAgentAliasRequest$, () => DeleteAgentAliasResponse$
];
var DeleteAgentVersion$ = [9, n0, _DAV,
    { [_h]: ["DELETE", "/agents/{agentId}/agentversions/{agentVersion}/", 202] }, () => DeleteAgentVersionRequest$, () => DeleteAgentVersionResponse$
];
var DeleteDataSource$ = [9, n0, _DDS,
    { [_h]: ["DELETE", "/knowledgebases/{knowledgeBaseId}/datasources/{dataSourceId}", 202] }, () => DeleteDataSourceRequest$, () => DeleteDataSourceResponse$
];
var DeleteFlow$ = [9, n0, _DF,
    { [_h]: ["DELETE", "/flows/{flowIdentifier}/", 200] }, () => DeleteFlowRequest$, () => DeleteFlowResponse$
];
var DeleteFlowAlias$ = [9, n0, _DFA,
    { [_h]: ["DELETE", "/flows/{flowIdentifier}/aliases/{aliasIdentifier}", 200] }, () => DeleteFlowAliasRequest$, () => DeleteFlowAliasResponse$
];
var DeleteFlowVersion$ = [9, n0, _DFV,
    { [_h]: ["DELETE", "/flows/{flowIdentifier}/versions/{flowVersion}/", 200] }, () => DeleteFlowVersionRequest$, () => DeleteFlowVersionResponse$
];
var DeleteKnowledgeBase$ = [9, n0, _DKB,
    { [_h]: ["DELETE", "/knowledgebases/{knowledgeBaseId}", 202] }, () => DeleteKnowledgeBaseRequest$, () => DeleteKnowledgeBaseResponse$
];
var DeleteKnowledgeBaseDocuments$ = [9, n0, _DKBD,
    { [_h]: ["POST", "/knowledgebases/{knowledgeBaseId}/datasources/{dataSourceId}/documents/deleteDocuments", 202] }, () => DeleteKnowledgeBaseDocumentsRequest$, () => DeleteKnowledgeBaseDocumentsResponse$
];
var DeletePrompt$ = [9, n0, _DP,
    { [_h]: ["DELETE", "/prompts/{promptIdentifier}/", 200] }, () => DeletePromptRequest$, () => DeletePromptResponse$
];
var DisassociateAgentCollaborator$ = [9, n0, _DAC,
    { [_h]: ["DELETE", "/agents/{agentId}/agentversions/{agentVersion}/agentcollaborators/{collaboratorId}/", 204] }, () => DisassociateAgentCollaboratorRequest$, () => DisassociateAgentCollaboratorResponse$
];
var DisassociateAgentKnowledgeBase$ = [9, n0, _DAKB,
    { [_h]: ["DELETE", "/agents/{agentId}/agentversions/{agentVersion}/knowledgebases/{knowledgeBaseId}/", 204] }, () => DisassociateAgentKnowledgeBaseRequest$, () => DisassociateAgentKnowledgeBaseResponse$
];
var GetAgent$ = [9, n0, _GAe,
    { [_h]: ["GET", "/agents/{agentId}/", 200] }, () => GetAgentRequest$, () => GetAgentResponse$
];
var GetAgentActionGroup$ = [9, n0, _GAAG,
    { [_h]: ["GET", "/agents/{agentId}/agentversions/{agentVersion}/actiongroups/{actionGroupId}/", 200] }, () => GetAgentActionGroupRequest$, () => GetAgentActionGroupResponse$
];
var GetAgentAlias$ = [9, n0, _GAA,
    { [_h]: ["GET", "/agents/{agentId}/agentaliases/{agentAliasId}/", 200] }, () => GetAgentAliasRequest$, () => GetAgentAliasResponse$
];
var GetAgentCollaborator$ = [9, n0, _GAC,
    { [_h]: ["GET", "/agents/{agentId}/agentversions/{agentVersion}/agentcollaborators/{collaboratorId}/", 200] }, () => GetAgentCollaboratorRequest$, () => GetAgentCollaboratorResponse$
];
var GetAgentKnowledgeBase$ = [9, n0, _GAKB,
    { [_h]: ["GET", "/agents/{agentId}/agentversions/{agentVersion}/knowledgebases/{knowledgeBaseId}/", 200] }, () => GetAgentKnowledgeBaseRequest$, () => GetAgentKnowledgeBaseResponse$
];
var GetAgentVersion$ = [9, n0, _GAV,
    { [_h]: ["GET", "/agents/{agentId}/agentversions/{agentVersion}/", 200] }, () => GetAgentVersionRequest$, () => GetAgentVersionResponse$
];
var GetDataSource$ = [9, n0, _GDS,
    { [_h]: ["GET", "/knowledgebases/{knowledgeBaseId}/datasources/{dataSourceId}", 200] }, () => GetDataSourceRequest$, () => GetDataSourceResponse$
];
var GetFlow$ = [9, n0, _GF,
    { [_h]: ["GET", "/flows/{flowIdentifier}/", 200] }, () => GetFlowRequest$, () => GetFlowResponse$
];
var GetFlowAlias$ = [9, n0, _GFA,
    { [_h]: ["GET", "/flows/{flowIdentifier}/aliases/{aliasIdentifier}", 200] }, () => GetFlowAliasRequest$, () => GetFlowAliasResponse$
];
var GetFlowVersion$ = [9, n0, _GFV,
    { [_h]: ["GET", "/flows/{flowIdentifier}/versions/{flowVersion}/", 200] }, () => GetFlowVersionRequest$, () => GetFlowVersionResponse$
];
var GetIngestionJob$ = [9, n0, _GIJ,
    { [_h]: ["GET", "/knowledgebases/{knowledgeBaseId}/datasources/{dataSourceId}/ingestionjobs/{ingestionJobId}", 200] }, () => GetIngestionJobRequest$, () => GetIngestionJobResponse$
];
var GetKnowledgeBase$ = [9, n0, _GKB,
    { [_h]: ["GET", "/knowledgebases/{knowledgeBaseId}", 200] }, () => GetKnowledgeBaseRequest$, () => GetKnowledgeBaseResponse$
];
var GetKnowledgeBaseDocuments$ = [9, n0, _GKBD,
    { [_h]: ["POST", "/knowledgebases/{knowledgeBaseId}/datasources/{dataSourceId}/documents/getDocuments", 200] }, () => GetKnowledgeBaseDocumentsRequest$, () => GetKnowledgeBaseDocumentsResponse$
];
var GetPrompt$ = [9, n0, _GP,
    { [_h]: ["GET", "/prompts/{promptIdentifier}/", 200] }, () => GetPromptRequest$, () => GetPromptResponse$
];
var IngestKnowledgeBaseDocuments$ = [9, n0, _IKBD,
    { [_h]: ["PUT", "/knowledgebases/{knowledgeBaseId}/datasources/{dataSourceId}/documents", 202] }, () => IngestKnowledgeBaseDocumentsRequest$, () => IngestKnowledgeBaseDocumentsResponse$
];
var ListAgentActionGroups$ = [9, n0, _LAAG,
    { [_h]: ["POST", "/agents/{agentId}/agentversions/{agentVersion}/actiongroups/", 200] }, () => ListAgentActionGroupsRequest$, () => ListAgentActionGroupsResponse$
];
var ListAgentAliases$ = [9, n0, _LAA,
    { [_h]: ["POST", "/agents/{agentId}/agentaliases/", 200] }, () => ListAgentAliasesRequest$, () => ListAgentAliasesResponse$
];
var ListAgentCollaborators$ = [9, n0, _LAC,
    { [_h]: ["POST", "/agents/{agentId}/agentversions/{agentVersion}/agentcollaborators/", 200] }, () => ListAgentCollaboratorsRequest$, () => ListAgentCollaboratorsResponse$
];
var ListAgentKnowledgeBases$ = [9, n0, _LAKB,
    { [_h]: ["POST", "/agents/{agentId}/agentversions/{agentVersion}/knowledgebases/", 200] }, () => ListAgentKnowledgeBasesRequest$, () => ListAgentKnowledgeBasesResponse$
];
var ListAgents$ = [9, n0, _LA,
    { [_h]: ["POST", "/agents/", 200] }, () => ListAgentsRequest$, () => ListAgentsResponse$
];
var ListAgentVersions$ = [9, n0, _LAV,
    { [_h]: ["POST", "/agents/{agentId}/agentversions/", 200] }, () => ListAgentVersionsRequest$, () => ListAgentVersionsResponse$
];
var ListDataSources$ = [9, n0, _LDS,
    { [_h]: ["POST", "/knowledgebases/{knowledgeBaseId}/datasources/", 200] }, () => ListDataSourcesRequest$, () => ListDataSourcesResponse$
];
var ListFlowAliases$ = [9, n0, _LFA,
    { [_h]: ["GET", "/flows/{flowIdentifier}/aliases", 200] }, () => ListFlowAliasesRequest$, () => ListFlowAliasesResponse$
];
var ListFlows$ = [9, n0, _LF,
    { [_h]: ["GET", "/flows/", 200] }, () => ListFlowsRequest$, () => ListFlowsResponse$
];
var ListFlowVersions$ = [9, n0, _LFV,
    { [_h]: ["GET", "/flows/{flowIdentifier}/versions", 200] }, () => ListFlowVersionsRequest$, () => ListFlowVersionsResponse$
];
var ListIngestionJobs$ = [9, n0, _LIJ,
    { [_h]: ["POST", "/knowledgebases/{knowledgeBaseId}/datasources/{dataSourceId}/ingestionjobs/", 200] }, () => ListIngestionJobsRequest$, () => ListIngestionJobsResponse$
];
var ListKnowledgeBaseDocuments$ = [9, n0, _LKBD,
    { [_h]: ["POST", "/knowledgebases/{knowledgeBaseId}/datasources/{dataSourceId}/documents", 200] }, () => ListKnowledgeBaseDocumentsRequest$, () => ListKnowledgeBaseDocumentsResponse$
];
var ListKnowledgeBases$ = [9, n0, _LKB,
    { [_h]: ["POST", "/knowledgebases/", 200] }, () => ListKnowledgeBasesRequest$, () => ListKnowledgeBasesResponse$
];
var ListPrompts$ = [9, n0, _LP,
    { [_h]: ["GET", "/prompts/", 200] }, () => ListPromptsRequest$, () => ListPromptsResponse$
];
var ListTagsForResource$ = [9, n0, _LTFR,
    { [_h]: ["GET", "/tags/{resourceArn}", 200] }, () => ListTagsForResourceRequest$, () => ListTagsForResourceResponse$
];
var PrepareAgent$ = [9, n0, _PA,
    { [_h]: ["POST", "/agents/{agentId}/", 202] }, () => PrepareAgentRequest$, () => PrepareAgentResponse$
];
var PrepareFlow$ = [9, n0, _PF,
    { [_h]: ["POST", "/flows/{flowIdentifier}/", 202] }, () => PrepareFlowRequest$, () => PrepareFlowResponse$
];
var StartIngestionJob$ = [9, n0, _SIJ,
    { [_h]: ["PUT", "/knowledgebases/{knowledgeBaseId}/datasources/{dataSourceId}/ingestionjobs/", 202] }, () => StartIngestionJobRequest$, () => StartIngestionJobResponse$
];
var StopIngestionJob$ = [9, n0, _SIJt,
    { [_h]: ["POST", "/knowledgebases/{knowledgeBaseId}/datasources/{dataSourceId}/ingestionjobs/{ingestionJobId}/stop", 202] }, () => StopIngestionJobRequest$, () => StopIngestionJobResponse$
];
var TagResource$ = [9, n0, _TR,
    { [_h]: ["POST", "/tags/{resourceArn}", 200] }, () => TagResourceRequest$, () => TagResourceResponse$
];
var UntagResource$ = [9, n0, _UR,
    { [_h]: ["DELETE", "/tags/{resourceArn}", 200] }, () => UntagResourceRequest$, () => UntagResourceResponse$
];
var UpdateAgent$ = [9, n0, _UAp,
    { [_h]: ["PUT", "/agents/{agentId}/", 202] }, () => UpdateAgentRequest$, () => UpdateAgentResponse$
];
var UpdateAgentActionGroup$ = [9, n0, _UAAG,
    { [_h]: ["PUT", "/agents/{agentId}/agentversions/{agentVersion}/actiongroups/{actionGroupId}/", 200] }, () => UpdateAgentActionGroupRequest$, () => UpdateAgentActionGroupResponse$
];
var UpdateAgentAlias$ = [9, n0, _UAA,
    { [_h]: ["PUT", "/agents/{agentId}/agentaliases/{agentAliasId}/", 202] }, () => UpdateAgentAliasRequest$, () => UpdateAgentAliasResponse$
];
var UpdateAgentCollaborator$ = [9, n0, _UAC,
    { [_h]: ["PUT", "/agents/{agentId}/agentversions/{agentVersion}/agentcollaborators/{collaboratorId}/", 200] }, () => UpdateAgentCollaboratorRequest$, () => UpdateAgentCollaboratorResponse$
];
var UpdateAgentKnowledgeBase$ = [9, n0, _UAKB,
    { [_h]: ["PUT", "/agents/{agentId}/agentversions/{agentVersion}/knowledgebases/{knowledgeBaseId}/", 200] }, () => UpdateAgentKnowledgeBaseRequest$, () => UpdateAgentKnowledgeBaseResponse$
];
var UpdateDataSource$ = [9, n0, _UDS,
    { [_h]: ["PUT", "/knowledgebases/{knowledgeBaseId}/datasources/{dataSourceId}", 200] }, () => UpdateDataSourceRequest$, () => UpdateDataSourceResponse$
];
var UpdateFlow$ = [9, n0, _UF,
    { [_h]: ["PUT", "/flows/{flowIdentifier}/", 200] }, () => UpdateFlowRequest$, () => UpdateFlowResponse$
];
var UpdateFlowAlias$ = [9, n0, _UFA,
    { [_h]: ["PUT", "/flows/{flowIdentifier}/aliases/{aliasIdentifier}", 200] }, () => UpdateFlowAliasRequest$, () => UpdateFlowAliasResponse$
];
var UpdateKnowledgeBase$ = [9, n0, _UKB,
    { [_h]: ["PUT", "/knowledgebases/{knowledgeBaseId}", 202] }, () => UpdateKnowledgeBaseRequest$, () => UpdateKnowledgeBaseResponse$
];
var UpdatePrompt$ = [9, n0, _UP,
    { [_h]: ["PUT", "/prompts/{promptIdentifier}/", 200] }, () => UpdatePromptRequest$, () => UpdatePromptResponse$
];
var ValidateFlowDefinition$ = [9, n0, _VFD,
    { [_h]: ["POST", "/flows/validate-definition", 200] }, () => ValidateFlowDefinitionRequest$, () => ValidateFlowDefinitionResponse$
];

class AssociateAgentCollaboratorCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "AssociateAgentCollaborator", {})
    .n("BedrockAgentClient", "AssociateAgentCollaboratorCommand")
    .sc(AssociateAgentCollaborator$)
    .build() {
}

class AssociateAgentKnowledgeBaseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "AssociateAgentKnowledgeBase", {})
    .n("BedrockAgentClient", "AssociateAgentKnowledgeBaseCommand")
    .sc(AssociateAgentKnowledgeBase$)
    .build() {
}

class CreateAgentActionGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "CreateAgentActionGroup", {})
    .n("BedrockAgentClient", "CreateAgentActionGroupCommand")
    .sc(CreateAgentActionGroup$)
    .build() {
}

class CreateAgentAliasCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "CreateAgentAlias", {})
    .n("BedrockAgentClient", "CreateAgentAliasCommand")
    .sc(CreateAgentAlias$)
    .build() {
}

class CreateAgentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "CreateAgent", {})
    .n("BedrockAgentClient", "CreateAgentCommand")
    .sc(CreateAgent$)
    .build() {
}

class CreateDataSourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "CreateDataSource", {})
    .n("BedrockAgentClient", "CreateDataSourceCommand")
    .sc(CreateDataSource$)
    .build() {
}

class CreateFlowAliasCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "CreateFlowAlias", {})
    .n("BedrockAgentClient", "CreateFlowAliasCommand")
    .sc(CreateFlowAlias$)
    .build() {
}

class CreateFlowCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "CreateFlow", {})
    .n("BedrockAgentClient", "CreateFlowCommand")
    .sc(CreateFlow$)
    .build() {
}

class CreateFlowVersionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "CreateFlowVersion", {})
    .n("BedrockAgentClient", "CreateFlowVersionCommand")
    .sc(CreateFlowVersion$)
    .build() {
}

class CreateKnowledgeBaseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "CreateKnowledgeBase", {})
    .n("BedrockAgentClient", "CreateKnowledgeBaseCommand")
    .sc(CreateKnowledgeBase$)
    .build() {
}

class CreatePromptCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "CreatePrompt", {})
    .n("BedrockAgentClient", "CreatePromptCommand")
    .sc(CreatePrompt$)
    .build() {
}

class CreatePromptVersionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "CreatePromptVersion", {})
    .n("BedrockAgentClient", "CreatePromptVersionCommand")
    .sc(CreatePromptVersion$)
    .build() {
}

class DeleteAgentActionGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "DeleteAgentActionGroup", {})
    .n("BedrockAgentClient", "DeleteAgentActionGroupCommand")
    .sc(DeleteAgentActionGroup$)
    .build() {
}

class DeleteAgentAliasCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "DeleteAgentAlias", {})
    .n("BedrockAgentClient", "DeleteAgentAliasCommand")
    .sc(DeleteAgentAlias$)
    .build() {
}

class DeleteAgentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "DeleteAgent", {})
    .n("BedrockAgentClient", "DeleteAgentCommand")
    .sc(DeleteAgent$)
    .build() {
}

class DeleteAgentVersionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "DeleteAgentVersion", {})
    .n("BedrockAgentClient", "DeleteAgentVersionCommand")
    .sc(DeleteAgentVersion$)
    .build() {
}

class DeleteDataSourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "DeleteDataSource", {})
    .n("BedrockAgentClient", "DeleteDataSourceCommand")
    .sc(DeleteDataSource$)
    .build() {
}

class DeleteFlowAliasCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "DeleteFlowAlias", {})
    .n("BedrockAgentClient", "DeleteFlowAliasCommand")
    .sc(DeleteFlowAlias$)
    .build() {
}

class DeleteFlowCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "DeleteFlow", {})
    .n("BedrockAgentClient", "DeleteFlowCommand")
    .sc(DeleteFlow$)
    .build() {
}

class DeleteFlowVersionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "DeleteFlowVersion", {})
    .n("BedrockAgentClient", "DeleteFlowVersionCommand")
    .sc(DeleteFlowVersion$)
    .build() {
}

class DeleteKnowledgeBaseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "DeleteKnowledgeBase", {})
    .n("BedrockAgentClient", "DeleteKnowledgeBaseCommand")
    .sc(DeleteKnowledgeBase$)
    .build() {
}

class DeleteKnowledgeBaseDocumentsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "DeleteKnowledgeBaseDocuments", {})
    .n("BedrockAgentClient", "DeleteKnowledgeBaseDocumentsCommand")
    .sc(DeleteKnowledgeBaseDocuments$)
    .build() {
}

class DeletePromptCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "DeletePrompt", {})
    .n("BedrockAgentClient", "DeletePromptCommand")
    .sc(DeletePrompt$)
    .build() {
}

class DisassociateAgentCollaboratorCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "DisassociateAgentCollaborator", {})
    .n("BedrockAgentClient", "DisassociateAgentCollaboratorCommand")
    .sc(DisassociateAgentCollaborator$)
    .build() {
}

class DisassociateAgentKnowledgeBaseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "DisassociateAgentKnowledgeBase", {})
    .n("BedrockAgentClient", "DisassociateAgentKnowledgeBaseCommand")
    .sc(DisassociateAgentKnowledgeBase$)
    .build() {
}

class GetAgentActionGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "GetAgentActionGroup", {})
    .n("BedrockAgentClient", "GetAgentActionGroupCommand")
    .sc(GetAgentActionGroup$)
    .build() {
}

class GetAgentAliasCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "GetAgentAlias", {})
    .n("BedrockAgentClient", "GetAgentAliasCommand")
    .sc(GetAgentAlias$)
    .build() {
}

class GetAgentCollaboratorCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "GetAgentCollaborator", {})
    .n("BedrockAgentClient", "GetAgentCollaboratorCommand")
    .sc(GetAgentCollaborator$)
    .build() {
}

class GetAgentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "GetAgent", {})
    .n("BedrockAgentClient", "GetAgentCommand")
    .sc(GetAgent$)
    .build() {
}

class GetAgentKnowledgeBaseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "GetAgentKnowledgeBase", {})
    .n("BedrockAgentClient", "GetAgentKnowledgeBaseCommand")
    .sc(GetAgentKnowledgeBase$)
    .build() {
}

class GetAgentVersionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "GetAgentVersion", {})
    .n("BedrockAgentClient", "GetAgentVersionCommand")
    .sc(GetAgentVersion$)
    .build() {
}

class GetDataSourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "GetDataSource", {})
    .n("BedrockAgentClient", "GetDataSourceCommand")
    .sc(GetDataSource$)
    .build() {
}

class GetFlowAliasCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "GetFlowAlias", {})
    .n("BedrockAgentClient", "GetFlowAliasCommand")
    .sc(GetFlowAlias$)
    .build() {
}

class GetFlowCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "GetFlow", {})
    .n("BedrockAgentClient", "GetFlowCommand")
    .sc(GetFlow$)
    .build() {
}

class GetFlowVersionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "GetFlowVersion", {})
    .n("BedrockAgentClient", "GetFlowVersionCommand")
    .sc(GetFlowVersion$)
    .build() {
}

class GetIngestionJobCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "GetIngestionJob", {})
    .n("BedrockAgentClient", "GetIngestionJobCommand")
    .sc(GetIngestionJob$)
    .build() {
}

class GetKnowledgeBaseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "GetKnowledgeBase", {})
    .n("BedrockAgentClient", "GetKnowledgeBaseCommand")
    .sc(GetKnowledgeBase$)
    .build() {
}

class GetKnowledgeBaseDocumentsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "GetKnowledgeBaseDocuments", {})
    .n("BedrockAgentClient", "GetKnowledgeBaseDocumentsCommand")
    .sc(GetKnowledgeBaseDocuments$)
    .build() {
}

class GetPromptCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "GetPrompt", {})
    .n("BedrockAgentClient", "GetPromptCommand")
    .sc(GetPrompt$)
    .build() {
}

class IngestKnowledgeBaseDocumentsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "IngestKnowledgeBaseDocuments", {})
    .n("BedrockAgentClient", "IngestKnowledgeBaseDocumentsCommand")
    .sc(IngestKnowledgeBaseDocuments$)
    .build() {
}

class ListAgentActionGroupsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "ListAgentActionGroups", {})
    .n("BedrockAgentClient", "ListAgentActionGroupsCommand")
    .sc(ListAgentActionGroups$)
    .build() {
}

class ListAgentAliasesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "ListAgentAliases", {})
    .n("BedrockAgentClient", "ListAgentAliasesCommand")
    .sc(ListAgentAliases$)
    .build() {
}

class ListAgentCollaboratorsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "ListAgentCollaborators", {})
    .n("BedrockAgentClient", "ListAgentCollaboratorsCommand")
    .sc(ListAgentCollaborators$)
    .build() {
}

class ListAgentKnowledgeBasesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "ListAgentKnowledgeBases", {})
    .n("BedrockAgentClient", "ListAgentKnowledgeBasesCommand")
    .sc(ListAgentKnowledgeBases$)
    .build() {
}

class ListAgentsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "ListAgents", {})
    .n("BedrockAgentClient", "ListAgentsCommand")
    .sc(ListAgents$)
    .build() {
}

class ListAgentVersionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "ListAgentVersions", {})
    .n("BedrockAgentClient", "ListAgentVersionsCommand")
    .sc(ListAgentVersions$)
    .build() {
}

class ListDataSourcesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "ListDataSources", {})
    .n("BedrockAgentClient", "ListDataSourcesCommand")
    .sc(ListDataSources$)
    .build() {
}

class ListFlowAliasesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "ListFlowAliases", {})
    .n("BedrockAgentClient", "ListFlowAliasesCommand")
    .sc(ListFlowAliases$)
    .build() {
}

class ListFlowsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "ListFlows", {})
    .n("BedrockAgentClient", "ListFlowsCommand")
    .sc(ListFlows$)
    .build() {
}

class ListFlowVersionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "ListFlowVersions", {})
    .n("BedrockAgentClient", "ListFlowVersionsCommand")
    .sc(ListFlowVersions$)
    .build() {
}

class ListIngestionJobsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "ListIngestionJobs", {})
    .n("BedrockAgentClient", "ListIngestionJobsCommand")
    .sc(ListIngestionJobs$)
    .build() {
}

class ListKnowledgeBaseDocumentsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "ListKnowledgeBaseDocuments", {})
    .n("BedrockAgentClient", "ListKnowledgeBaseDocumentsCommand")
    .sc(ListKnowledgeBaseDocuments$)
    .build() {
}

class ListKnowledgeBasesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "ListKnowledgeBases", {})
    .n("BedrockAgentClient", "ListKnowledgeBasesCommand")
    .sc(ListKnowledgeBases$)
    .build() {
}

class ListPromptsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "ListPrompts", {})
    .n("BedrockAgentClient", "ListPromptsCommand")
    .sc(ListPrompts$)
    .build() {
}

class ListTagsForResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "ListTagsForResource", {})
    .n("BedrockAgentClient", "ListTagsForResourceCommand")
    .sc(ListTagsForResource$)
    .build() {
}

class PrepareAgentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "PrepareAgent", {})
    .n("BedrockAgentClient", "PrepareAgentCommand")
    .sc(PrepareAgent$)
    .build() {
}

class PrepareFlowCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "PrepareFlow", {})
    .n("BedrockAgentClient", "PrepareFlowCommand")
    .sc(PrepareFlow$)
    .build() {
}

class StartIngestionJobCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "StartIngestionJob", {})
    .n("BedrockAgentClient", "StartIngestionJobCommand")
    .sc(StartIngestionJob$)
    .build() {
}

class StopIngestionJobCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "StopIngestionJob", {})
    .n("BedrockAgentClient", "StopIngestionJobCommand")
    .sc(StopIngestionJob$)
    .build() {
}

class TagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "TagResource", {})
    .n("BedrockAgentClient", "TagResourceCommand")
    .sc(TagResource$)
    .build() {
}

class UntagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "UntagResource", {})
    .n("BedrockAgentClient", "UntagResourceCommand")
    .sc(UntagResource$)
    .build() {
}

class UpdateAgentActionGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "UpdateAgentActionGroup", {})
    .n("BedrockAgentClient", "UpdateAgentActionGroupCommand")
    .sc(UpdateAgentActionGroup$)
    .build() {
}

class UpdateAgentAliasCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "UpdateAgentAlias", {})
    .n("BedrockAgentClient", "UpdateAgentAliasCommand")
    .sc(UpdateAgentAlias$)
    .build() {
}

class UpdateAgentCollaboratorCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "UpdateAgentCollaborator", {})
    .n("BedrockAgentClient", "UpdateAgentCollaboratorCommand")
    .sc(UpdateAgentCollaborator$)
    .build() {
}

class UpdateAgentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "UpdateAgent", {})
    .n("BedrockAgentClient", "UpdateAgentCommand")
    .sc(UpdateAgent$)
    .build() {
}

class UpdateAgentKnowledgeBaseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "UpdateAgentKnowledgeBase", {})
    .n("BedrockAgentClient", "UpdateAgentKnowledgeBaseCommand")
    .sc(UpdateAgentKnowledgeBase$)
    .build() {
}

class UpdateDataSourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "UpdateDataSource", {})
    .n("BedrockAgentClient", "UpdateDataSourceCommand")
    .sc(UpdateDataSource$)
    .build() {
}

class UpdateFlowAliasCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "UpdateFlowAlias", {})
    .n("BedrockAgentClient", "UpdateFlowAliasCommand")
    .sc(UpdateFlowAlias$)
    .build() {
}

class UpdateFlowCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "UpdateFlow", {})
    .n("BedrockAgentClient", "UpdateFlowCommand")
    .sc(UpdateFlow$)
    .build() {
}

class UpdateKnowledgeBaseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "UpdateKnowledgeBase", {})
    .n("BedrockAgentClient", "UpdateKnowledgeBaseCommand")
    .sc(UpdateKnowledgeBase$)
    .build() {
}

class UpdatePromptCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "UpdatePrompt", {})
    .n("BedrockAgentClient", "UpdatePromptCommand")
    .sc(UpdatePrompt$)
    .build() {
}

class ValidateFlowDefinitionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentBuildTimeLambda", "ValidateFlowDefinition", {})
    .n("BedrockAgentClient", "ValidateFlowDefinitionCommand")
    .sc(ValidateFlowDefinition$)
    .build() {
}

const commands = {
    AssociateAgentCollaboratorCommand,
    AssociateAgentKnowledgeBaseCommand,
    CreateAgentCommand,
    CreateAgentActionGroupCommand,
    CreateAgentAliasCommand,
    CreateDataSourceCommand,
    CreateFlowCommand,
    CreateFlowAliasCommand,
    CreateFlowVersionCommand,
    CreateKnowledgeBaseCommand,
    CreatePromptCommand,
    CreatePromptVersionCommand,
    DeleteAgentCommand,
    DeleteAgentActionGroupCommand,
    DeleteAgentAliasCommand,
    DeleteAgentVersionCommand,
    DeleteDataSourceCommand,
    DeleteFlowCommand,
    DeleteFlowAliasCommand,
    DeleteFlowVersionCommand,
    DeleteKnowledgeBaseCommand,
    DeleteKnowledgeBaseDocumentsCommand,
    DeletePromptCommand,
    DisassociateAgentCollaboratorCommand,
    DisassociateAgentKnowledgeBaseCommand,
    GetAgentCommand,
    GetAgentActionGroupCommand,
    GetAgentAliasCommand,
    GetAgentCollaboratorCommand,
    GetAgentKnowledgeBaseCommand,
    GetAgentVersionCommand,
    GetDataSourceCommand,
    GetFlowCommand,
    GetFlowAliasCommand,
    GetFlowVersionCommand,
    GetIngestionJobCommand,
    GetKnowledgeBaseCommand,
    GetKnowledgeBaseDocumentsCommand,
    GetPromptCommand,
    IngestKnowledgeBaseDocumentsCommand,
    ListAgentActionGroupsCommand,
    ListAgentAliasesCommand,
    ListAgentCollaboratorsCommand,
    ListAgentKnowledgeBasesCommand,
    ListAgentsCommand,
    ListAgentVersionsCommand,
    ListDataSourcesCommand,
    ListFlowAliasesCommand,
    ListFlowsCommand,
    ListFlowVersionsCommand,
    ListIngestionJobsCommand,
    ListKnowledgeBaseDocumentsCommand,
    ListKnowledgeBasesCommand,
    ListPromptsCommand,
    ListTagsForResourceCommand,
    PrepareAgentCommand,
    PrepareFlowCommand,
    StartIngestionJobCommand,
    StopIngestionJobCommand,
    TagResourceCommand,
    UntagResourceCommand,
    UpdateAgentCommand,
    UpdateAgentActionGroupCommand,
    UpdateAgentAliasCommand,
    UpdateAgentCollaboratorCommand,
    UpdateAgentKnowledgeBaseCommand,
    UpdateDataSourceCommand,
    UpdateFlowCommand,
    UpdateFlowAliasCommand,
    UpdateKnowledgeBaseCommand,
    UpdatePromptCommand,
    ValidateFlowDefinitionCommand,
};
class BedrockAgent extends BedrockAgentClient {
}
smithyClient.createAggregatedClient(commands, BedrockAgent);

const paginateListAgentActionGroups = core.createPaginator(BedrockAgentClient, ListAgentActionGroupsCommand, "nextToken", "nextToken", "maxResults");

const paginateListAgentAliases = core.createPaginator(BedrockAgentClient, ListAgentAliasesCommand, "nextToken", "nextToken", "maxResults");

const paginateListAgentCollaborators = core.createPaginator(BedrockAgentClient, ListAgentCollaboratorsCommand, "nextToken", "nextToken", "maxResults");

const paginateListAgentKnowledgeBases = core.createPaginator(BedrockAgentClient, ListAgentKnowledgeBasesCommand, "nextToken", "nextToken", "maxResults");

const paginateListAgents = core.createPaginator(BedrockAgentClient, ListAgentsCommand, "nextToken", "nextToken", "maxResults");

const paginateListAgentVersions = core.createPaginator(BedrockAgentClient, ListAgentVersionsCommand, "nextToken", "nextToken", "maxResults");

const paginateListDataSources = core.createPaginator(BedrockAgentClient, ListDataSourcesCommand, "nextToken", "nextToken", "maxResults");

const paginateListFlowAliases = core.createPaginator(BedrockAgentClient, ListFlowAliasesCommand, "nextToken", "nextToken", "maxResults");

const paginateListFlows = core.createPaginator(BedrockAgentClient, ListFlowsCommand, "nextToken", "nextToken", "maxResults");

const paginateListFlowVersions = core.createPaginator(BedrockAgentClient, ListFlowVersionsCommand, "nextToken", "nextToken", "maxResults");

const paginateListIngestionJobs = core.createPaginator(BedrockAgentClient, ListIngestionJobsCommand, "nextToken", "nextToken", "maxResults");

const paginateListKnowledgeBaseDocuments = core.createPaginator(BedrockAgentClient, ListKnowledgeBaseDocumentsCommand, "nextToken", "nextToken", "maxResults");

const paginateListKnowledgeBases = core.createPaginator(BedrockAgentClient, ListKnowledgeBasesCommand, "nextToken", "nextToken", "maxResults");

const paginateListPrompts = core.createPaginator(BedrockAgentClient, ListPromptsCommand, "nextToken", "nextToken", "maxResults");

const CustomControlMethod = {
    RETURN_CONTROL: "RETURN_CONTROL",
};
const ActionGroupState = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const Type = {
    ARRAY: "array",
    BOOLEAN: "boolean",
    INTEGER: "integer",
    NUMBER: "number",
    STRING: "string",
};
const RequireConfirmation = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const ActionGroupSignature = {
    AMAZON_CODEINTERPRETER: "AMAZON.CodeInterpreter",
    AMAZON_USERINPUT: "AMAZON.UserInput",
    ANTHROPIC_BASH: "ANTHROPIC.Bash",
    ANTHROPIC_COMPUTER: "ANTHROPIC.Computer",
    ANTHROPIC_TEXTEDITOR: "ANTHROPIC.TextEditor",
};
const AgentCollaboration = {
    DISABLED: "DISABLED",
    SUPERVISOR: "SUPERVISOR",
    SUPERVISOR_ROUTER: "SUPERVISOR_ROUTER",
};
const AgentStatus = {
    CREATING: "CREATING",
    DELETING: "DELETING",
    FAILED: "FAILED",
    NOT_PREPARED: "NOT_PREPARED",
    PREPARED: "PREPARED",
    PREPARING: "PREPARING",
    UPDATING: "UPDATING",
    VERSIONING: "VERSIONING",
};
const MemoryType = {
    SESSION_SUMMARY: "SESSION_SUMMARY",
};
const OrchestrationType = {
    CUSTOM_ORCHESTRATION: "CUSTOM_ORCHESTRATION",
    DEFAULT: "DEFAULT",
};
const CreationMode = {
    DEFAULT: "DEFAULT",
    OVERRIDDEN: "OVERRIDDEN",
};
const PromptState = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const PromptType = {
    KNOWLEDGE_BASE_RESPONSE_GENERATION: "KNOWLEDGE_BASE_RESPONSE_GENERATION",
    MEMORY_SUMMARIZATION: "MEMORY_SUMMARIZATION",
    ORCHESTRATION: "ORCHESTRATION",
    POST_PROCESSING: "POST_PROCESSING",
    PRE_PROCESSING: "PRE_PROCESSING",
};
const AgentAliasStatus = {
    CREATING: "CREATING",
    DELETING: "DELETING",
    DISSOCIATED: "DISSOCIATED",
    FAILED: "FAILED",
    PREPARED: "PREPARED",
    UPDATING: "UPDATING",
};
const AliasInvocationState = {
    ACCEPT_INVOCATIONS: "ACCEPT_INVOCATIONS",
    REJECT_INVOCATIONS: "REJECT_INVOCATIONS",
};
const RelayConversationHistory = {
    DISABLED: "DISABLED",
    TO_COLLABORATOR: "TO_COLLABORATOR",
};
const KnowledgeBaseState = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const DataDeletionPolicy = {
    DELETE: "DELETE",
    RETAIN: "RETAIN",
};
const CrawlFilterConfigurationType = {
    PATTERN: "PATTERN",
};
const ConfluenceAuthType = {
    BASIC: "BASIC",
    OAUTH2_CLIENT_CREDENTIALS: "OAUTH2_CLIENT_CREDENTIALS",
};
const ConfluenceHostType = {
    SAAS: "SAAS",
};
const SalesforceAuthType = {
    OAUTH2_CLIENT_CREDENTIALS: "OAUTH2_CLIENT_CREDENTIALS",
};
const SharePointAuthType = {
    OAUTH2_CLIENT_CREDENTIALS: "OAUTH2_CLIENT_CREDENTIALS",
    OAUTH2_SHAREPOINT_APP_ONLY_CLIENT_CREDENTIALS: "OAUTH2_SHAREPOINT_APP_ONLY_CLIENT_CREDENTIALS",
};
const SharePointHostType = {
    ONLINE: "ONLINE",
};
const DataSourceType = {
    CONFLUENCE: "CONFLUENCE",
    CUSTOM: "CUSTOM",
    REDSHIFT_METADATA: "REDSHIFT_METADATA",
    S3: "S3",
    SALESFORCE: "SALESFORCE",
    SHAREPOINT: "SHAREPOINT",
    WEB: "WEB",
};
const WebScopeType = {
    HOST_ONLY: "HOST_ONLY",
    SUBDOMAINS: "SUBDOMAINS",
};
const ChunkingStrategy = {
    FIXED_SIZE: "FIXED_SIZE",
    HIERARCHICAL: "HIERARCHICAL",
    NONE: "NONE",
    SEMANTIC: "SEMANTIC",
};
const EnrichmentStrategyMethod = {
    CHUNK_ENTITY_EXTRACTION: "CHUNK_ENTITY_EXTRACTION",
};
const ContextEnrichmentType = {
    BEDROCK_FOUNDATION_MODEL: "BEDROCK_FOUNDATION_MODEL",
};
const StepType = {
    POST_CHUNKING: "POST_CHUNKING",
};
const ParsingModality = {
    MULTIMODAL: "MULTIMODAL",
};
const ParsingStrategy = {
    BEDROCK_DATA_AUTOMATION: "BEDROCK_DATA_AUTOMATION",
    BEDROCK_FOUNDATION_MODEL: "BEDROCK_FOUNDATION_MODEL",
};
const DataSourceStatus = {
    AVAILABLE: "AVAILABLE",
    DELETE_UNSUCCESSFUL: "DELETE_UNSUCCESSFUL",
    DELETING: "DELETING",
};
const FlowConnectionType = {
    CONDITIONAL: "Conditional",
    DATA: "Data",
};
const SupportedLanguages = {
    PYTHON_3: "Python_3",
};
const PerformanceConfigLatency = {
    OPTIMIZED: "optimized",
    STANDARD: "standard",
};
const RerankingMetadataSelectionMode = {
    ALL: "ALL",
    SELECTIVE: "SELECTIVE",
};
const VectorSearchRerankingConfigurationType = {
    BEDROCK_RERANKING_MODEL: "BEDROCK_RERANKING_MODEL",
};
const CachePointType = {
    DEFAULT: "default",
};
const ConversationRole = {
    ASSISTANT: "assistant",
    USER: "user",
};
const PromptTemplateType = {
    CHAT: "CHAT",
    TEXT: "TEXT",
};
const FlowNodeInputCategory = {
    EXIT_LOOP: "ExitLoop",
    LOOP_CONDITION: "LoopCondition",
    RETURN_VALUE_TO_LOOP_START: "ReturnValueToLoopStart",
};
const FlowNodeIODataType = {
    ARRAY: "Array",
    BOOLEAN: "Boolean",
    NUMBER: "Number",
    OBJECT: "Object",
    STRING: "String",
};
const FlowNodeType = {
    AGENT: "Agent",
    COLLECTOR: "Collector",
    CONDITION: "Condition",
    INLINE_CODE: "InlineCode",
    INPUT: "Input",
    ITERATOR: "Iterator",
    KNOWLEDGE_BASE: "KnowledgeBase",
    LAMBDA_FUNCTION: "LambdaFunction",
    LEX: "Lex",
    LOOP: "Loop",
    LOOP_CONTROLLER: "LoopController",
    LOOP_INPUT: "LoopInput",
    OUTPUT: "Output",
    PROMPT: "Prompt",
    RETRIEVAL: "Retrieval",
    STORAGE: "Storage",
};
const FlowStatus = {
    FAILED: "Failed",
    NOT_PREPARED: "NotPrepared",
    PREPARED: "Prepared",
    PREPARING: "Preparing",
};
const ConcurrencyType = {
    AUTOMATIC: "Automatic",
    MANUAL: "Manual",
};
const IncompatibleLoopNodeType = {
    COLLECTOR: "Collector",
    CONDITION: "Condition",
    INPUT: "Input",
    ITERATOR: "Iterator",
};
const FlowValidationSeverity = {
    ERROR: "Error",
    WARNING: "Warning",
};
const FlowValidationType = {
    CYCLIC_CONNECTION: "CyclicConnection",
    DUPLICATE_CONDITION_EXPRESSION: "DuplicateConditionExpression",
    DUPLICATE_CONNECTIONS: "DuplicateConnections",
    INCOMPATIBLE_CONNECTION_DATA_TYPE: "IncompatibleConnectionDataType",
    INVALID_LOOP_BOUNDARY: "InvalidLoopBoundary",
    LOOP_INCOMPATIBLE_NODE_TYPE: "LoopIncompatibleNodeType",
    MALFORMED_CONDITION_EXPRESSION: "MalformedConditionExpression",
    MALFORMED_NODE_INPUT_EXPRESSION: "MalformedNodeInputExpression",
    MISMATCHED_NODE_INPUT_TYPE: "MismatchedNodeInputType",
    MISMATCHED_NODE_OUTPUT_TYPE: "MismatchedNodeOutputType",
    MISSING_CONNECTION_CONFIGURATION: "MissingConnectionConfiguration",
    MISSING_DEFAULT_CONDITION: "MissingDefaultCondition",
    MISSING_ENDING_NODES: "MissingEndingNodes",
    MISSING_LOOP_CONTROLLER_NODE: "MissingLoopControllerNode",
    MISSING_LOOP_INPUT_NODE: "MissingLoopInputNode",
    MISSING_NODE_CONFIGURATION: "MissingNodeConfiguration",
    MISSING_NODE_INPUT: "MissingNodeInput",
    MISSING_NODE_OUTPUT: "MissingNodeOutput",
    MISSING_STARTING_NODES: "MissingStartingNodes",
    MULTIPLE_LOOP_CONTROLLER_NODES: "MultipleLoopControllerNodes",
    MULTIPLE_LOOP_INPUT_NODES: "MultipleLoopInputNodes",
    MULTIPLE_NODE_INPUT_CONNECTIONS: "MultipleNodeInputConnections",
    UNFULFILLED_NODE_INPUT: "UnfulfilledNodeInput",
    UNKNOWN_CONNECTION_CONDITION: "UnknownConnectionCondition",
    UNKNOWN_CONNECTION_SOURCE: "UnknownConnectionSource",
    UNKNOWN_CONNECTION_SOURCE_OUTPUT: "UnknownConnectionSourceOutput",
    UNKNOWN_CONNECTION_TARGET: "UnknownConnectionTarget",
    UNKNOWN_CONNECTION_TARGET_INPUT: "UnknownConnectionTargetInput",
    UNKNOWN_NODE_INPUT: "UnknownNodeInput",
    UNKNOWN_NODE_OUTPUT: "UnknownNodeOutput",
    UNREACHABLE_NODE: "UnreachableNode",
    UNSATISFIED_CONNECTION_CONDITIONS: "UnsatisfiedConnectionConditions",
    UNSPECIFIED: "Unspecified",
};
const IngestionJobStatus = {
    COMPLETE: "COMPLETE",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    STARTING: "STARTING",
    STOPPED: "STOPPED",
    STOPPING: "STOPPING",
};
const IngestionJobFilterAttribute = {
    STATUS: "STATUS",
};
const IngestionJobFilterOperator = {
    EQ: "EQ",
};
const IngestionJobSortByAttribute = {
    STARTED_AT: "STARTED_AT",
    STATUS: "STATUS",
};
const SortOrder = {
    ASCENDING: "ASCENDING",
    DESCENDING: "DESCENDING",
};
const ContentDataSourceType = {
    CUSTOM: "CUSTOM",
    S3: "S3",
};
const DocumentStatus = {
    DELETE_IN_PROGRESS: "DELETE_IN_PROGRESS",
    DELETING: "DELETING",
    FAILED: "FAILED",
    IGNORED: "IGNORED",
    INDEXED: "INDEXED",
    IN_PROGRESS: "IN_PROGRESS",
    METADATA_PARTIALLY_INDEXED: "METADATA_PARTIALLY_INDEXED",
    METADATA_UPDATE_FAILED: "METADATA_UPDATE_FAILED",
    NOT_FOUND: "NOT_FOUND",
    PARTIALLY_INDEXED: "PARTIALLY_INDEXED",
    PENDING: "PENDING",
    STARTING: "STARTING",
};
const InlineContentType = {
    BYTE: "BYTE",
    TEXT: "TEXT",
};
const CustomSourceType = {
    IN_LINE: "IN_LINE",
    S3_LOCATION: "S3_LOCATION",
};
const MetadataValueType = {
    BOOLEAN: "BOOLEAN",
    NUMBER: "NUMBER",
    STRING: "STRING",
    STRING_LIST: "STRING_LIST",
};
const MetadataSourceType = {
    IN_LINE_ATTRIBUTE: "IN_LINE_ATTRIBUTE",
    S3_LOCATION: "S3_LOCATION",
};
const RedshiftProvisionedAuthType = {
    IAM: "IAM",
    USERNAME: "USERNAME",
    USERNAME_PASSWORD: "USERNAME_PASSWORD",
};
const RedshiftServerlessAuthType = {
    IAM: "IAM",
    USERNAME_PASSWORD: "USERNAME_PASSWORD",
};
const RedshiftQueryEngineType = {
    PROVISIONED: "PROVISIONED",
    SERVERLESS: "SERVERLESS",
};
const IncludeExclude = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
};
const RedshiftQueryEngineStorageType = {
    AWS_DATA_CATALOG: "AWS_DATA_CATALOG",
    REDSHIFT: "REDSHIFT",
};
const QueryEngineType = {
    REDSHIFT: "REDSHIFT",
};
const KnowledgeBaseType = {
    KENDRA: "KENDRA",
    SQL: "SQL",
    VECTOR: "VECTOR",
};
const EmbeddingDataType = {
    BINARY: "BINARY",
    FLOAT32: "FLOAT32",
};
const SupplementalDataStorageLocationType = {
    S3: "S3",
};
const KnowledgeBaseStorageType = {
    MONGO_DB_ATLAS: "MONGO_DB_ATLAS",
    NEPTUNE_ANALYTICS: "NEPTUNE_ANALYTICS",
    OPENSEARCH_MANAGED_CLUSTER: "OPENSEARCH_MANAGED_CLUSTER",
    OPENSEARCH_SERVERLESS: "OPENSEARCH_SERVERLESS",
    PINECONE: "PINECONE",
    RDS: "RDS",
    REDIS_ENTERPRISE_CLOUD: "REDIS_ENTERPRISE_CLOUD",
    S3_VECTORS: "S3_VECTORS",
};
const KnowledgeBaseStatus = {
    ACTIVE: "ACTIVE",
    CREATING: "CREATING",
    DELETE_UNSUCCESSFUL: "DELETE_UNSUCCESSFUL",
    DELETING: "DELETING",
    FAILED: "FAILED",
    UPDATING: "UPDATING",
};

Object.defineProperty(exports, "$Command", {
    enumerable: true,
    get: function () { return smithyClient.Command; }
});
Object.defineProperty(exports, "__Client", {
    enumerable: true,
    get: function () { return smithyClient.Client; }
});
exports.APISchema$ = APISchema$;
exports.AccessDeniedException = AccessDeniedException;
exports.AccessDeniedException$ = AccessDeniedException$;
exports.ActionGroupExecutor$ = ActionGroupExecutor$;
exports.ActionGroupSignature = ActionGroupSignature;
exports.ActionGroupState = ActionGroupState;
exports.ActionGroupSummary$ = ActionGroupSummary$;
exports.Agent$ = Agent$;
exports.AgentActionGroup$ = AgentActionGroup$;
exports.AgentAlias$ = AgentAlias$;
exports.AgentAliasHistoryEvent$ = AgentAliasHistoryEvent$;
exports.AgentAliasRoutingConfigurationListItem$ = AgentAliasRoutingConfigurationListItem$;
exports.AgentAliasStatus = AgentAliasStatus;
exports.AgentAliasSummary$ = AgentAliasSummary$;
exports.AgentCollaboration = AgentCollaboration;
exports.AgentCollaborator$ = AgentCollaborator$;
exports.AgentCollaboratorSummary$ = AgentCollaboratorSummary$;
exports.AgentDescriptor$ = AgentDescriptor$;
exports.AgentFlowNodeConfiguration$ = AgentFlowNodeConfiguration$;
exports.AgentKnowledgeBase$ = AgentKnowledgeBase$;
exports.AgentKnowledgeBaseSummary$ = AgentKnowledgeBaseSummary$;
exports.AgentStatus = AgentStatus;
exports.AgentSummary$ = AgentSummary$;
exports.AgentVersion$ = AgentVersion$;
exports.AgentVersionSummary$ = AgentVersionSummary$;
exports.AliasInvocationState = AliasInvocationState;
exports.AnyToolChoice$ = AnyToolChoice$;
exports.AssociateAgentCollaborator$ = AssociateAgentCollaborator$;
exports.AssociateAgentCollaboratorCommand = AssociateAgentCollaboratorCommand;
exports.AssociateAgentCollaboratorRequest$ = AssociateAgentCollaboratorRequest$;
exports.AssociateAgentCollaboratorResponse$ = AssociateAgentCollaboratorResponse$;
exports.AssociateAgentKnowledgeBase$ = AssociateAgentKnowledgeBase$;
exports.AssociateAgentKnowledgeBaseCommand = AssociateAgentKnowledgeBaseCommand;
exports.AssociateAgentKnowledgeBaseRequest$ = AssociateAgentKnowledgeBaseRequest$;
exports.AssociateAgentKnowledgeBaseResponse$ = AssociateAgentKnowledgeBaseResponse$;
exports.AudioConfiguration$ = AudioConfiguration$;
exports.AudioSegmentationConfiguration$ = AudioSegmentationConfiguration$;
exports.AutoToolChoice$ = AutoToolChoice$;
exports.BedrockAgent = BedrockAgent;
exports.BedrockAgentClient = BedrockAgentClient;
exports.BedrockAgentServiceException = BedrockAgentServiceException;
exports.BedrockAgentServiceException$ = BedrockAgentServiceException$;
exports.BedrockDataAutomationConfiguration$ = BedrockDataAutomationConfiguration$;
exports.BedrockEmbeddingModelConfiguration$ = BedrockEmbeddingModelConfiguration$;
exports.BedrockFoundationModelConfiguration$ = BedrockFoundationModelConfiguration$;
exports.BedrockFoundationModelContextEnrichmentConfiguration$ = BedrockFoundationModelContextEnrichmentConfiguration$;
exports.ByteContentDoc$ = ByteContentDoc$;
exports.CachePointBlock$ = CachePointBlock$;
exports.CachePointType = CachePointType;
exports.ChatPromptTemplateConfiguration$ = ChatPromptTemplateConfiguration$;
exports.ChunkingConfiguration$ = ChunkingConfiguration$;
exports.ChunkingStrategy = ChunkingStrategy;
exports.CollectorFlowNodeConfiguration$ = CollectorFlowNodeConfiguration$;
exports.ConcurrencyType = ConcurrencyType;
exports.ConditionFlowNodeConfiguration$ = ConditionFlowNodeConfiguration$;
exports.ConflictException = ConflictException;
exports.ConflictException$ = ConflictException$;
exports.ConfluenceAuthType = ConfluenceAuthType;
exports.ConfluenceCrawlerConfiguration$ = ConfluenceCrawlerConfiguration$;
exports.ConfluenceDataSourceConfiguration$ = ConfluenceDataSourceConfiguration$;
exports.ConfluenceHostType = ConfluenceHostType;
exports.ConfluenceSourceConfiguration$ = ConfluenceSourceConfiguration$;
exports.ContentBlock$ = ContentBlock$;
exports.ContentDataSourceType = ContentDataSourceType;
exports.ContextEnrichmentConfiguration$ = ContextEnrichmentConfiguration$;
exports.ContextEnrichmentType = ContextEnrichmentType;
exports.ConversationRole = ConversationRole;
exports.CrawlFilterConfiguration$ = CrawlFilterConfiguration$;
exports.CrawlFilterConfigurationType = CrawlFilterConfigurationType;
exports.CreateAgent$ = CreateAgent$;
exports.CreateAgentActionGroup$ = CreateAgentActionGroup$;
exports.CreateAgentActionGroupCommand = CreateAgentActionGroupCommand;
exports.CreateAgentActionGroupRequest$ = CreateAgentActionGroupRequest$;
exports.CreateAgentActionGroupResponse$ = CreateAgentActionGroupResponse$;
exports.CreateAgentAlias$ = CreateAgentAlias$;
exports.CreateAgentAliasCommand = CreateAgentAliasCommand;
exports.CreateAgentAliasRequest$ = CreateAgentAliasRequest$;
exports.CreateAgentAliasResponse$ = CreateAgentAliasResponse$;
exports.CreateAgentCommand = CreateAgentCommand;
exports.CreateAgentRequest$ = CreateAgentRequest$;
exports.CreateAgentResponse$ = CreateAgentResponse$;
exports.CreateDataSource$ = CreateDataSource$;
exports.CreateDataSourceCommand = CreateDataSourceCommand;
exports.CreateDataSourceRequest$ = CreateDataSourceRequest$;
exports.CreateDataSourceResponse$ = CreateDataSourceResponse$;
exports.CreateFlow$ = CreateFlow$;
exports.CreateFlowAlias$ = CreateFlowAlias$;
exports.CreateFlowAliasCommand = CreateFlowAliasCommand;
exports.CreateFlowAliasRequest$ = CreateFlowAliasRequest$;
exports.CreateFlowAliasResponse$ = CreateFlowAliasResponse$;
exports.CreateFlowCommand = CreateFlowCommand;
exports.CreateFlowRequest$ = CreateFlowRequest$;
exports.CreateFlowResponse$ = CreateFlowResponse$;
exports.CreateFlowVersion$ = CreateFlowVersion$;
exports.CreateFlowVersionCommand = CreateFlowVersionCommand;
exports.CreateFlowVersionRequest$ = CreateFlowVersionRequest$;
exports.CreateFlowVersionResponse$ = CreateFlowVersionResponse$;
exports.CreateKnowledgeBase$ = CreateKnowledgeBase$;
exports.CreateKnowledgeBaseCommand = CreateKnowledgeBaseCommand;
exports.CreateKnowledgeBaseRequest$ = CreateKnowledgeBaseRequest$;
exports.CreateKnowledgeBaseResponse$ = CreateKnowledgeBaseResponse$;
exports.CreatePrompt$ = CreatePrompt$;
exports.CreatePromptCommand = CreatePromptCommand;
exports.CreatePromptRequest$ = CreatePromptRequest$;
exports.CreatePromptResponse$ = CreatePromptResponse$;
exports.CreatePromptVersion$ = CreatePromptVersion$;
exports.CreatePromptVersionCommand = CreatePromptVersionCommand;
exports.CreatePromptVersionRequest$ = CreatePromptVersionRequest$;
exports.CreatePromptVersionResponse$ = CreatePromptVersionResponse$;
exports.CreationMode = CreationMode;
exports.CuratedQuery$ = CuratedQuery$;
exports.CustomContent$ = CustomContent$;
exports.CustomControlMethod = CustomControlMethod;
exports.CustomDocumentIdentifier$ = CustomDocumentIdentifier$;
exports.CustomOrchestration$ = CustomOrchestration$;
exports.CustomS3Location$ = CustomS3Location$;
exports.CustomSourceType = CustomSourceType;
exports.CustomTransformationConfiguration$ = CustomTransformationConfiguration$;
exports.CyclicConnectionFlowValidationDetails$ = CyclicConnectionFlowValidationDetails$;
exports.DataDeletionPolicy = DataDeletionPolicy;
exports.DataSource$ = DataSource$;
exports.DataSourceConfiguration$ = DataSourceConfiguration$;
exports.DataSourceStatus = DataSourceStatus;
exports.DataSourceSummary$ = DataSourceSummary$;
exports.DataSourceType = DataSourceType;
exports.DeleteAgent$ = DeleteAgent$;
exports.DeleteAgentActionGroup$ = DeleteAgentActionGroup$;
exports.DeleteAgentActionGroupCommand = DeleteAgentActionGroupCommand;
exports.DeleteAgentActionGroupRequest$ = DeleteAgentActionGroupRequest$;
exports.DeleteAgentActionGroupResponse$ = DeleteAgentActionGroupResponse$;
exports.DeleteAgentAlias$ = DeleteAgentAlias$;
exports.DeleteAgentAliasCommand = DeleteAgentAliasCommand;
exports.DeleteAgentAliasRequest$ = DeleteAgentAliasRequest$;
exports.DeleteAgentAliasResponse$ = DeleteAgentAliasResponse$;
exports.DeleteAgentCommand = DeleteAgentCommand;
exports.DeleteAgentRequest$ = DeleteAgentRequest$;
exports.DeleteAgentResponse$ = DeleteAgentResponse$;
exports.DeleteAgentVersion$ = DeleteAgentVersion$;
exports.DeleteAgentVersionCommand = DeleteAgentVersionCommand;
exports.DeleteAgentVersionRequest$ = DeleteAgentVersionRequest$;
exports.DeleteAgentVersionResponse$ = DeleteAgentVersionResponse$;
exports.DeleteDataSource$ = DeleteDataSource$;
exports.DeleteDataSourceCommand = DeleteDataSourceCommand;
exports.DeleteDataSourceRequest$ = DeleteDataSourceRequest$;
exports.DeleteDataSourceResponse$ = DeleteDataSourceResponse$;
exports.DeleteFlow$ = DeleteFlow$;
exports.DeleteFlowAlias$ = DeleteFlowAlias$;
exports.DeleteFlowAliasCommand = DeleteFlowAliasCommand;
exports.DeleteFlowAliasRequest$ = DeleteFlowAliasRequest$;
exports.DeleteFlowAliasResponse$ = DeleteFlowAliasResponse$;
exports.DeleteFlowCommand = DeleteFlowCommand;
exports.DeleteFlowRequest$ = DeleteFlowRequest$;
exports.DeleteFlowResponse$ = DeleteFlowResponse$;
exports.DeleteFlowVersion$ = DeleteFlowVersion$;
exports.DeleteFlowVersionCommand = DeleteFlowVersionCommand;
exports.DeleteFlowVersionRequest$ = DeleteFlowVersionRequest$;
exports.DeleteFlowVersionResponse$ = DeleteFlowVersionResponse$;
exports.DeleteKnowledgeBase$ = DeleteKnowledgeBase$;
exports.DeleteKnowledgeBaseCommand = DeleteKnowledgeBaseCommand;
exports.DeleteKnowledgeBaseDocuments$ = DeleteKnowledgeBaseDocuments$;
exports.DeleteKnowledgeBaseDocumentsCommand = DeleteKnowledgeBaseDocumentsCommand;
exports.DeleteKnowledgeBaseDocumentsRequest$ = DeleteKnowledgeBaseDocumentsRequest$;
exports.DeleteKnowledgeBaseDocumentsResponse$ = DeleteKnowledgeBaseDocumentsResponse$;
exports.DeleteKnowledgeBaseRequest$ = DeleteKnowledgeBaseRequest$;
exports.DeleteKnowledgeBaseResponse$ = DeleteKnowledgeBaseResponse$;
exports.DeletePrompt$ = DeletePrompt$;
exports.DeletePromptCommand = DeletePromptCommand;
exports.DeletePromptRequest$ = DeletePromptRequest$;
exports.DeletePromptResponse$ = DeletePromptResponse$;
exports.DisassociateAgentCollaborator$ = DisassociateAgentCollaborator$;
exports.DisassociateAgentCollaboratorCommand = DisassociateAgentCollaboratorCommand;
exports.DisassociateAgentCollaboratorRequest$ = DisassociateAgentCollaboratorRequest$;
exports.DisassociateAgentCollaboratorResponse$ = DisassociateAgentCollaboratorResponse$;
exports.DisassociateAgentKnowledgeBase$ = DisassociateAgentKnowledgeBase$;
exports.DisassociateAgentKnowledgeBaseCommand = DisassociateAgentKnowledgeBaseCommand;
exports.DisassociateAgentKnowledgeBaseRequest$ = DisassociateAgentKnowledgeBaseRequest$;
exports.DisassociateAgentKnowledgeBaseResponse$ = DisassociateAgentKnowledgeBaseResponse$;
exports.DocumentContent$ = DocumentContent$;
exports.DocumentIdentifier$ = DocumentIdentifier$;
exports.DocumentMetadata$ = DocumentMetadata$;
exports.DocumentStatus = DocumentStatus;
exports.DuplicateConditionExpressionFlowValidationDetails$ = DuplicateConditionExpressionFlowValidationDetails$;
exports.DuplicateConnectionsFlowValidationDetails$ = DuplicateConnectionsFlowValidationDetails$;
exports.EmbeddingDataType = EmbeddingDataType;
exports.EmbeddingModelConfiguration$ = EmbeddingModelConfiguration$;
exports.EnrichmentStrategyConfiguration$ = EnrichmentStrategyConfiguration$;
exports.EnrichmentStrategyMethod = EnrichmentStrategyMethod;
exports.FieldForReranking$ = FieldForReranking$;
exports.FixedSizeChunkingConfiguration$ = FixedSizeChunkingConfiguration$;
exports.FlowAliasConcurrencyConfiguration$ = FlowAliasConcurrencyConfiguration$;
exports.FlowAliasRoutingConfigurationListItem$ = FlowAliasRoutingConfigurationListItem$;
exports.FlowAliasSummary$ = FlowAliasSummary$;
exports.FlowCondition$ = FlowCondition$;
exports.FlowConditionalConnectionConfiguration$ = FlowConditionalConnectionConfiguration$;
exports.FlowConnection$ = FlowConnection$;
exports.FlowConnectionConfiguration$ = FlowConnectionConfiguration$;
exports.FlowConnectionType = FlowConnectionType;
exports.FlowDataConnectionConfiguration$ = FlowDataConnectionConfiguration$;
exports.FlowDefinition$ = FlowDefinition$;
exports.FlowNode$ = FlowNode$;
exports.FlowNodeConfiguration$ = FlowNodeConfiguration$;
exports.FlowNodeIODataType = FlowNodeIODataType;
exports.FlowNodeInput$ = FlowNodeInput$;
exports.FlowNodeInputCategory = FlowNodeInputCategory;
exports.FlowNodeOutput$ = FlowNodeOutput$;
exports.FlowNodeType = FlowNodeType;
exports.FlowStatus = FlowStatus;
exports.FlowSummary$ = FlowSummary$;
exports.FlowValidation$ = FlowValidation$;
exports.FlowValidationDetails$ = FlowValidationDetails$;
exports.FlowValidationSeverity = FlowValidationSeverity;
exports.FlowValidationType = FlowValidationType;
exports.FlowVersionSummary$ = FlowVersionSummary$;
exports.Function$ = Function$;
exports.FunctionSchema$ = FunctionSchema$;
exports.GetAgent$ = GetAgent$;
exports.GetAgentActionGroup$ = GetAgentActionGroup$;
exports.GetAgentActionGroupCommand = GetAgentActionGroupCommand;
exports.GetAgentActionGroupRequest$ = GetAgentActionGroupRequest$;
exports.GetAgentActionGroupResponse$ = GetAgentActionGroupResponse$;
exports.GetAgentAlias$ = GetAgentAlias$;
exports.GetAgentAliasCommand = GetAgentAliasCommand;
exports.GetAgentAliasRequest$ = GetAgentAliasRequest$;
exports.GetAgentAliasResponse$ = GetAgentAliasResponse$;
exports.GetAgentCollaborator$ = GetAgentCollaborator$;
exports.GetAgentCollaboratorCommand = GetAgentCollaboratorCommand;
exports.GetAgentCollaboratorRequest$ = GetAgentCollaboratorRequest$;
exports.GetAgentCollaboratorResponse$ = GetAgentCollaboratorResponse$;
exports.GetAgentCommand = GetAgentCommand;
exports.GetAgentKnowledgeBase$ = GetAgentKnowledgeBase$;
exports.GetAgentKnowledgeBaseCommand = GetAgentKnowledgeBaseCommand;
exports.GetAgentKnowledgeBaseRequest$ = GetAgentKnowledgeBaseRequest$;
exports.GetAgentKnowledgeBaseResponse$ = GetAgentKnowledgeBaseResponse$;
exports.GetAgentRequest$ = GetAgentRequest$;
exports.GetAgentResponse$ = GetAgentResponse$;
exports.GetAgentVersion$ = GetAgentVersion$;
exports.GetAgentVersionCommand = GetAgentVersionCommand;
exports.GetAgentVersionRequest$ = GetAgentVersionRequest$;
exports.GetAgentVersionResponse$ = GetAgentVersionResponse$;
exports.GetDataSource$ = GetDataSource$;
exports.GetDataSourceCommand = GetDataSourceCommand;
exports.GetDataSourceRequest$ = GetDataSourceRequest$;
exports.GetDataSourceResponse$ = GetDataSourceResponse$;
exports.GetFlow$ = GetFlow$;
exports.GetFlowAlias$ = GetFlowAlias$;
exports.GetFlowAliasCommand = GetFlowAliasCommand;
exports.GetFlowAliasRequest$ = GetFlowAliasRequest$;
exports.GetFlowAliasResponse$ = GetFlowAliasResponse$;
exports.GetFlowCommand = GetFlowCommand;
exports.GetFlowRequest$ = GetFlowRequest$;
exports.GetFlowResponse$ = GetFlowResponse$;
exports.GetFlowVersion$ = GetFlowVersion$;
exports.GetFlowVersionCommand = GetFlowVersionCommand;
exports.GetFlowVersionRequest$ = GetFlowVersionRequest$;
exports.GetFlowVersionResponse$ = GetFlowVersionResponse$;
exports.GetIngestionJob$ = GetIngestionJob$;
exports.GetIngestionJobCommand = GetIngestionJobCommand;
exports.GetIngestionJobRequest$ = GetIngestionJobRequest$;
exports.GetIngestionJobResponse$ = GetIngestionJobResponse$;
exports.GetKnowledgeBase$ = GetKnowledgeBase$;
exports.GetKnowledgeBaseCommand = GetKnowledgeBaseCommand;
exports.GetKnowledgeBaseDocuments$ = GetKnowledgeBaseDocuments$;
exports.GetKnowledgeBaseDocumentsCommand = GetKnowledgeBaseDocumentsCommand;
exports.GetKnowledgeBaseDocumentsRequest$ = GetKnowledgeBaseDocumentsRequest$;
exports.GetKnowledgeBaseDocumentsResponse$ = GetKnowledgeBaseDocumentsResponse$;
exports.GetKnowledgeBaseRequest$ = GetKnowledgeBaseRequest$;
exports.GetKnowledgeBaseResponse$ = GetKnowledgeBaseResponse$;
exports.GetPrompt$ = GetPrompt$;
exports.GetPromptCommand = GetPromptCommand;
exports.GetPromptRequest$ = GetPromptRequest$;
exports.GetPromptResponse$ = GetPromptResponse$;
exports.GuardrailConfiguration$ = GuardrailConfiguration$;
exports.HierarchicalChunkingConfiguration$ = HierarchicalChunkingConfiguration$;
exports.HierarchicalChunkingLevelConfiguration$ = HierarchicalChunkingLevelConfiguration$;
exports.IncludeExclude = IncludeExclude;
exports.IncompatibleConnectionDataTypeFlowValidationDetails$ = IncompatibleConnectionDataTypeFlowValidationDetails$;
exports.IncompatibleLoopNodeType = IncompatibleLoopNodeType;
exports.InferenceConfiguration$ = InferenceConfiguration$;
exports.IngestKnowledgeBaseDocuments$ = IngestKnowledgeBaseDocuments$;
exports.IngestKnowledgeBaseDocumentsCommand = IngestKnowledgeBaseDocumentsCommand;
exports.IngestKnowledgeBaseDocumentsRequest$ = IngestKnowledgeBaseDocumentsRequest$;
exports.IngestKnowledgeBaseDocumentsResponse$ = IngestKnowledgeBaseDocumentsResponse$;
exports.IngestionJob$ = IngestionJob$;
exports.IngestionJobFilter$ = IngestionJobFilter$;
exports.IngestionJobFilterAttribute = IngestionJobFilterAttribute;
exports.IngestionJobFilterOperator = IngestionJobFilterOperator;
exports.IngestionJobSortBy$ = IngestionJobSortBy$;
exports.IngestionJobSortByAttribute = IngestionJobSortByAttribute;
exports.IngestionJobStatistics$ = IngestionJobStatistics$;
exports.IngestionJobStatus = IngestionJobStatus;
exports.IngestionJobSummary$ = IngestionJobSummary$;
exports.InlineCodeFlowNodeConfiguration$ = InlineCodeFlowNodeConfiguration$;
exports.InlineContent$ = InlineContent$;
exports.InlineContentType = InlineContentType;
exports.InputFlowNodeConfiguration$ = InputFlowNodeConfiguration$;
exports.IntermediateStorage$ = IntermediateStorage$;
exports.InternalServerException = InternalServerException;
exports.InternalServerException$ = InternalServerException$;
exports.InvalidLoopBoundaryFlowValidationDetails$ = InvalidLoopBoundaryFlowValidationDetails$;
exports.IteratorFlowNodeConfiguration$ = IteratorFlowNodeConfiguration$;
exports.KendraKnowledgeBaseConfiguration$ = KendraKnowledgeBaseConfiguration$;
exports.KnowledgeBase$ = KnowledgeBase$;
exports.KnowledgeBaseConfiguration$ = KnowledgeBaseConfiguration$;
exports.KnowledgeBaseDocument$ = KnowledgeBaseDocument$;
exports.KnowledgeBaseDocumentDetail$ = KnowledgeBaseDocumentDetail$;
exports.KnowledgeBaseFlowNodeConfiguration$ = KnowledgeBaseFlowNodeConfiguration$;
exports.KnowledgeBaseOrchestrationConfiguration$ = KnowledgeBaseOrchestrationConfiguration$;
exports.KnowledgeBasePromptTemplate$ = KnowledgeBasePromptTemplate$;
exports.KnowledgeBaseState = KnowledgeBaseState;
exports.KnowledgeBaseStatus = KnowledgeBaseStatus;
exports.KnowledgeBaseStorageType = KnowledgeBaseStorageType;
exports.KnowledgeBaseSummary$ = KnowledgeBaseSummary$;
exports.KnowledgeBaseType = KnowledgeBaseType;
exports.LambdaFunctionFlowNodeConfiguration$ = LambdaFunctionFlowNodeConfiguration$;
exports.LexFlowNodeConfiguration$ = LexFlowNodeConfiguration$;
exports.ListAgentActionGroups$ = ListAgentActionGroups$;
exports.ListAgentActionGroupsCommand = ListAgentActionGroupsCommand;
exports.ListAgentActionGroupsRequest$ = ListAgentActionGroupsRequest$;
exports.ListAgentActionGroupsResponse$ = ListAgentActionGroupsResponse$;
exports.ListAgentAliases$ = ListAgentAliases$;
exports.ListAgentAliasesCommand = ListAgentAliasesCommand;
exports.ListAgentAliasesRequest$ = ListAgentAliasesRequest$;
exports.ListAgentAliasesResponse$ = ListAgentAliasesResponse$;
exports.ListAgentCollaborators$ = ListAgentCollaborators$;
exports.ListAgentCollaboratorsCommand = ListAgentCollaboratorsCommand;
exports.ListAgentCollaboratorsRequest$ = ListAgentCollaboratorsRequest$;
exports.ListAgentCollaboratorsResponse$ = ListAgentCollaboratorsResponse$;
exports.ListAgentKnowledgeBases$ = ListAgentKnowledgeBases$;
exports.ListAgentKnowledgeBasesCommand = ListAgentKnowledgeBasesCommand;
exports.ListAgentKnowledgeBasesRequest$ = ListAgentKnowledgeBasesRequest$;
exports.ListAgentKnowledgeBasesResponse$ = ListAgentKnowledgeBasesResponse$;
exports.ListAgentVersions$ = ListAgentVersions$;
exports.ListAgentVersionsCommand = ListAgentVersionsCommand;
exports.ListAgentVersionsRequest$ = ListAgentVersionsRequest$;
exports.ListAgentVersionsResponse$ = ListAgentVersionsResponse$;
exports.ListAgents$ = ListAgents$;
exports.ListAgentsCommand = ListAgentsCommand;
exports.ListAgentsRequest$ = ListAgentsRequest$;
exports.ListAgentsResponse$ = ListAgentsResponse$;
exports.ListDataSources$ = ListDataSources$;
exports.ListDataSourcesCommand = ListDataSourcesCommand;
exports.ListDataSourcesRequest$ = ListDataSourcesRequest$;
exports.ListDataSourcesResponse$ = ListDataSourcesResponse$;
exports.ListFlowAliases$ = ListFlowAliases$;
exports.ListFlowAliasesCommand = ListFlowAliasesCommand;
exports.ListFlowAliasesRequest$ = ListFlowAliasesRequest$;
exports.ListFlowAliasesResponse$ = ListFlowAliasesResponse$;
exports.ListFlowVersions$ = ListFlowVersions$;
exports.ListFlowVersionsCommand = ListFlowVersionsCommand;
exports.ListFlowVersionsRequest$ = ListFlowVersionsRequest$;
exports.ListFlowVersionsResponse$ = ListFlowVersionsResponse$;
exports.ListFlows$ = ListFlows$;
exports.ListFlowsCommand = ListFlowsCommand;
exports.ListFlowsRequest$ = ListFlowsRequest$;
exports.ListFlowsResponse$ = ListFlowsResponse$;
exports.ListIngestionJobs$ = ListIngestionJobs$;
exports.ListIngestionJobsCommand = ListIngestionJobsCommand;
exports.ListIngestionJobsRequest$ = ListIngestionJobsRequest$;
exports.ListIngestionJobsResponse$ = ListIngestionJobsResponse$;
exports.ListKnowledgeBaseDocuments$ = ListKnowledgeBaseDocuments$;
exports.ListKnowledgeBaseDocumentsCommand = ListKnowledgeBaseDocumentsCommand;
exports.ListKnowledgeBaseDocumentsRequest$ = ListKnowledgeBaseDocumentsRequest$;
exports.ListKnowledgeBaseDocumentsResponse$ = ListKnowledgeBaseDocumentsResponse$;
exports.ListKnowledgeBases$ = ListKnowledgeBases$;
exports.ListKnowledgeBasesCommand = ListKnowledgeBasesCommand;
exports.ListKnowledgeBasesRequest$ = ListKnowledgeBasesRequest$;
exports.ListKnowledgeBasesResponse$ = ListKnowledgeBasesResponse$;
exports.ListPrompts$ = ListPrompts$;
exports.ListPromptsCommand = ListPromptsCommand;
exports.ListPromptsRequest$ = ListPromptsRequest$;
exports.ListPromptsResponse$ = ListPromptsResponse$;
exports.ListTagsForResource$ = ListTagsForResource$;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.ListTagsForResourceRequest$ = ListTagsForResourceRequest$;
exports.ListTagsForResourceResponse$ = ListTagsForResourceResponse$;
exports.LoopControllerFlowNodeConfiguration$ = LoopControllerFlowNodeConfiguration$;
exports.LoopFlowNodeConfiguration$ = LoopFlowNodeConfiguration$;
exports.LoopIncompatibleNodeTypeFlowValidationDetails$ = LoopIncompatibleNodeTypeFlowValidationDetails$;
exports.LoopInputFlowNodeConfiguration$ = LoopInputFlowNodeConfiguration$;
exports.MalformedConditionExpressionFlowValidationDetails$ = MalformedConditionExpressionFlowValidationDetails$;
exports.MalformedNodeInputExpressionFlowValidationDetails$ = MalformedNodeInputExpressionFlowValidationDetails$;
exports.MemoryConfiguration$ = MemoryConfiguration$;
exports.MemoryType = MemoryType;
exports.Message$ = Message$;
exports.MetadataAttribute$ = MetadataAttribute$;
exports.MetadataAttributeValue$ = MetadataAttributeValue$;
exports.MetadataConfigurationForReranking$ = MetadataConfigurationForReranking$;
exports.MetadataSourceType = MetadataSourceType;
exports.MetadataValueType = MetadataValueType;
exports.MismatchedNodeInputTypeFlowValidationDetails$ = MismatchedNodeInputTypeFlowValidationDetails$;
exports.MismatchedNodeOutputTypeFlowValidationDetails$ = MismatchedNodeOutputTypeFlowValidationDetails$;
exports.MissingConnectionConfigurationFlowValidationDetails$ = MissingConnectionConfigurationFlowValidationDetails$;
exports.MissingDefaultConditionFlowValidationDetails$ = MissingDefaultConditionFlowValidationDetails$;
exports.MissingEndingNodesFlowValidationDetails$ = MissingEndingNodesFlowValidationDetails$;
exports.MissingLoopControllerNodeFlowValidationDetails$ = MissingLoopControllerNodeFlowValidationDetails$;
exports.MissingLoopInputNodeFlowValidationDetails$ = MissingLoopInputNodeFlowValidationDetails$;
exports.MissingNodeConfigurationFlowValidationDetails$ = MissingNodeConfigurationFlowValidationDetails$;
exports.MissingNodeInputFlowValidationDetails$ = MissingNodeInputFlowValidationDetails$;
exports.MissingNodeOutputFlowValidationDetails$ = MissingNodeOutputFlowValidationDetails$;
exports.MissingStartingNodesFlowValidationDetails$ = MissingStartingNodesFlowValidationDetails$;
exports.MongoDbAtlasConfiguration$ = MongoDbAtlasConfiguration$;
exports.MongoDbAtlasFieldMapping$ = MongoDbAtlasFieldMapping$;
exports.MultipleLoopControllerNodesFlowValidationDetails$ = MultipleLoopControllerNodesFlowValidationDetails$;
exports.MultipleLoopInputNodesFlowValidationDetails$ = MultipleLoopInputNodesFlowValidationDetails$;
exports.MultipleNodeInputConnectionsFlowValidationDetails$ = MultipleNodeInputConnectionsFlowValidationDetails$;
exports.NeptuneAnalyticsConfiguration$ = NeptuneAnalyticsConfiguration$;
exports.NeptuneAnalyticsFieldMapping$ = NeptuneAnalyticsFieldMapping$;
exports.OpenSearchManagedClusterConfiguration$ = OpenSearchManagedClusterConfiguration$;
exports.OpenSearchManagedClusterFieldMapping$ = OpenSearchManagedClusterFieldMapping$;
exports.OpenSearchServerlessConfiguration$ = OpenSearchServerlessConfiguration$;
exports.OpenSearchServerlessFieldMapping$ = OpenSearchServerlessFieldMapping$;
exports.OrchestrationExecutor$ = OrchestrationExecutor$;
exports.OrchestrationType = OrchestrationType;
exports.OutputFlowNodeConfiguration$ = OutputFlowNodeConfiguration$;
exports.ParameterDetail$ = ParameterDetail$;
exports.ParsingConfiguration$ = ParsingConfiguration$;
exports.ParsingModality = ParsingModality;
exports.ParsingPrompt$ = ParsingPrompt$;
exports.ParsingStrategy = ParsingStrategy;
exports.PatternObjectFilter$ = PatternObjectFilter$;
exports.PatternObjectFilterConfiguration$ = PatternObjectFilterConfiguration$;
exports.PerformanceConfigLatency = PerformanceConfigLatency;
exports.PerformanceConfiguration$ = PerformanceConfiguration$;
exports.PineconeConfiguration$ = PineconeConfiguration$;
exports.PineconeFieldMapping$ = PineconeFieldMapping$;
exports.PrepareAgent$ = PrepareAgent$;
exports.PrepareAgentCommand = PrepareAgentCommand;
exports.PrepareAgentRequest$ = PrepareAgentRequest$;
exports.PrepareAgentResponse$ = PrepareAgentResponse$;
exports.PrepareFlow$ = PrepareFlow$;
exports.PrepareFlowCommand = PrepareFlowCommand;
exports.PrepareFlowRequest$ = PrepareFlowRequest$;
exports.PrepareFlowResponse$ = PrepareFlowResponse$;
exports.PromptAgentResource$ = PromptAgentResource$;
exports.PromptConfiguration$ = PromptConfiguration$;
exports.PromptFlowNodeConfiguration$ = PromptFlowNodeConfiguration$;
exports.PromptFlowNodeInlineConfiguration$ = PromptFlowNodeInlineConfiguration$;
exports.PromptFlowNodeResourceConfiguration$ = PromptFlowNodeResourceConfiguration$;
exports.PromptFlowNodeSourceConfiguration$ = PromptFlowNodeSourceConfiguration$;
exports.PromptGenAiResource$ = PromptGenAiResource$;
exports.PromptInferenceConfiguration$ = PromptInferenceConfiguration$;
exports.PromptInputVariable$ = PromptInputVariable$;
exports.PromptMetadataEntry$ = PromptMetadataEntry$;
exports.PromptModelInferenceConfiguration$ = PromptModelInferenceConfiguration$;
exports.PromptOverrideConfiguration$ = PromptOverrideConfiguration$;
exports.PromptState = PromptState;
exports.PromptSummary$ = PromptSummary$;
exports.PromptTemplateConfiguration$ = PromptTemplateConfiguration$;
exports.PromptTemplateType = PromptTemplateType;
exports.PromptType = PromptType;
exports.PromptVariant$ = PromptVariant$;
exports.QueryEngineType = QueryEngineType;
exports.QueryGenerationColumn$ = QueryGenerationColumn$;
exports.QueryGenerationConfiguration$ = QueryGenerationConfiguration$;
exports.QueryGenerationContext$ = QueryGenerationContext$;
exports.QueryGenerationTable$ = QueryGenerationTable$;
exports.RdsConfiguration$ = RdsConfiguration$;
exports.RdsFieldMapping$ = RdsFieldMapping$;
exports.RedisEnterpriseCloudConfiguration$ = RedisEnterpriseCloudConfiguration$;
exports.RedisEnterpriseCloudFieldMapping$ = RedisEnterpriseCloudFieldMapping$;
exports.RedshiftConfiguration$ = RedshiftConfiguration$;
exports.RedshiftProvisionedAuthConfiguration$ = RedshiftProvisionedAuthConfiguration$;
exports.RedshiftProvisionedAuthType = RedshiftProvisionedAuthType;
exports.RedshiftProvisionedConfiguration$ = RedshiftProvisionedConfiguration$;
exports.RedshiftQueryEngineAwsDataCatalogStorageConfiguration$ = RedshiftQueryEngineAwsDataCatalogStorageConfiguration$;
exports.RedshiftQueryEngineConfiguration$ = RedshiftQueryEngineConfiguration$;
exports.RedshiftQueryEngineRedshiftStorageConfiguration$ = RedshiftQueryEngineRedshiftStorageConfiguration$;
exports.RedshiftQueryEngineStorageConfiguration$ = RedshiftQueryEngineStorageConfiguration$;
exports.RedshiftQueryEngineStorageType = RedshiftQueryEngineStorageType;
exports.RedshiftQueryEngineType = RedshiftQueryEngineType;
exports.RedshiftServerlessAuthConfiguration$ = RedshiftServerlessAuthConfiguration$;
exports.RedshiftServerlessAuthType = RedshiftServerlessAuthType;
exports.RedshiftServerlessConfiguration$ = RedshiftServerlessConfiguration$;
exports.RelayConversationHistory = RelayConversationHistory;
exports.RequireConfirmation = RequireConfirmation;
exports.RerankingMetadataSelectionMode = RerankingMetadataSelectionMode;
exports.RerankingMetadataSelectiveModeConfiguration$ = RerankingMetadataSelectiveModeConfiguration$;
exports.ResourceNotFoundException = ResourceNotFoundException;
exports.ResourceNotFoundException$ = ResourceNotFoundException$;
exports.RetrievalFlowNodeConfiguration$ = RetrievalFlowNodeConfiguration$;
exports.RetrievalFlowNodeS3Configuration$ = RetrievalFlowNodeS3Configuration$;
exports.RetrievalFlowNodeServiceConfiguration$ = RetrievalFlowNodeServiceConfiguration$;
exports.S3Content$ = S3Content$;
exports.S3DataSourceConfiguration$ = S3DataSourceConfiguration$;
exports.S3Identifier$ = S3Identifier$;
exports.S3Location$ = S3Location$;
exports.S3VectorsConfiguration$ = S3VectorsConfiguration$;
exports.SalesforceAuthType = SalesforceAuthType;
exports.SalesforceCrawlerConfiguration$ = SalesforceCrawlerConfiguration$;
exports.SalesforceDataSourceConfiguration$ = SalesforceDataSourceConfiguration$;
exports.SalesforceSourceConfiguration$ = SalesforceSourceConfiguration$;
exports.SeedUrl$ = SeedUrl$;
exports.SemanticChunkingConfiguration$ = SemanticChunkingConfiguration$;
exports.ServerSideEncryptionConfiguration$ = ServerSideEncryptionConfiguration$;
exports.ServiceQuotaExceededException = ServiceQuotaExceededException;
exports.ServiceQuotaExceededException$ = ServiceQuotaExceededException$;
exports.SessionSummaryConfiguration$ = SessionSummaryConfiguration$;
exports.SharePointAuthType = SharePointAuthType;
exports.SharePointCrawlerConfiguration$ = SharePointCrawlerConfiguration$;
exports.SharePointDataSourceConfiguration$ = SharePointDataSourceConfiguration$;
exports.SharePointHostType = SharePointHostType;
exports.SharePointSourceConfiguration$ = SharePointSourceConfiguration$;
exports.SortOrder = SortOrder;
exports.SpecificToolChoice$ = SpecificToolChoice$;
exports.SqlKnowledgeBaseConfiguration$ = SqlKnowledgeBaseConfiguration$;
exports.StartIngestionJob$ = StartIngestionJob$;
exports.StartIngestionJobCommand = StartIngestionJobCommand;
exports.StartIngestionJobRequest$ = StartIngestionJobRequest$;
exports.StartIngestionJobResponse$ = StartIngestionJobResponse$;
exports.StepType = StepType;
exports.StopIngestionJob$ = StopIngestionJob$;
exports.StopIngestionJobCommand = StopIngestionJobCommand;
exports.StopIngestionJobRequest$ = StopIngestionJobRequest$;
exports.StopIngestionJobResponse$ = StopIngestionJobResponse$;
exports.StorageConfiguration$ = StorageConfiguration$;
exports.StorageFlowNodeConfiguration$ = StorageFlowNodeConfiguration$;
exports.StorageFlowNodeS3Configuration$ = StorageFlowNodeS3Configuration$;
exports.StorageFlowNodeServiceConfiguration$ = StorageFlowNodeServiceConfiguration$;
exports.SupplementalDataStorageConfiguration$ = SupplementalDataStorageConfiguration$;
exports.SupplementalDataStorageLocation$ = SupplementalDataStorageLocation$;
exports.SupplementalDataStorageLocationType = SupplementalDataStorageLocationType;
exports.SupportedLanguages = SupportedLanguages;
exports.SystemContentBlock$ = SystemContentBlock$;
exports.TagResource$ = TagResource$;
exports.TagResourceCommand = TagResourceCommand;
exports.TagResourceRequest$ = TagResourceRequest$;
exports.TagResourceResponse$ = TagResourceResponse$;
exports.TextContentDoc$ = TextContentDoc$;
exports.TextPromptTemplateConfiguration$ = TextPromptTemplateConfiguration$;
exports.ThrottlingException = ThrottlingException;
exports.ThrottlingException$ = ThrottlingException$;
exports.Tool$ = Tool$;
exports.ToolChoice$ = ToolChoice$;
exports.ToolConfiguration$ = ToolConfiguration$;
exports.ToolInputSchema$ = ToolInputSchema$;
exports.ToolSpecification$ = ToolSpecification$;
exports.Transformation$ = Transformation$;
exports.TransformationFunction$ = TransformationFunction$;
exports.TransformationLambdaConfiguration$ = TransformationLambdaConfiguration$;
exports.Type = Type;
exports.UnfulfilledNodeInputFlowValidationDetails$ = UnfulfilledNodeInputFlowValidationDetails$;
exports.UnknownConnectionConditionFlowValidationDetails$ = UnknownConnectionConditionFlowValidationDetails$;
exports.UnknownConnectionSourceFlowValidationDetails$ = UnknownConnectionSourceFlowValidationDetails$;
exports.UnknownConnectionSourceOutputFlowValidationDetails$ = UnknownConnectionSourceOutputFlowValidationDetails$;
exports.UnknownConnectionTargetFlowValidationDetails$ = UnknownConnectionTargetFlowValidationDetails$;
exports.UnknownConnectionTargetInputFlowValidationDetails$ = UnknownConnectionTargetInputFlowValidationDetails$;
exports.UnknownNodeInputFlowValidationDetails$ = UnknownNodeInputFlowValidationDetails$;
exports.UnknownNodeOutputFlowValidationDetails$ = UnknownNodeOutputFlowValidationDetails$;
exports.UnreachableNodeFlowValidationDetails$ = UnreachableNodeFlowValidationDetails$;
exports.UnsatisfiedConnectionConditionsFlowValidationDetails$ = UnsatisfiedConnectionConditionsFlowValidationDetails$;
exports.UnspecifiedFlowValidationDetails$ = UnspecifiedFlowValidationDetails$;
exports.UntagResource$ = UntagResource$;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UntagResourceRequest$ = UntagResourceRequest$;
exports.UntagResourceResponse$ = UntagResourceResponse$;
exports.UpdateAgent$ = UpdateAgent$;
exports.UpdateAgentActionGroup$ = UpdateAgentActionGroup$;
exports.UpdateAgentActionGroupCommand = UpdateAgentActionGroupCommand;
exports.UpdateAgentActionGroupRequest$ = UpdateAgentActionGroupRequest$;
exports.UpdateAgentActionGroupResponse$ = UpdateAgentActionGroupResponse$;
exports.UpdateAgentAlias$ = UpdateAgentAlias$;
exports.UpdateAgentAliasCommand = UpdateAgentAliasCommand;
exports.UpdateAgentAliasRequest$ = UpdateAgentAliasRequest$;
exports.UpdateAgentAliasResponse$ = UpdateAgentAliasResponse$;
exports.UpdateAgentCollaborator$ = UpdateAgentCollaborator$;
exports.UpdateAgentCollaboratorCommand = UpdateAgentCollaboratorCommand;
exports.UpdateAgentCollaboratorRequest$ = UpdateAgentCollaboratorRequest$;
exports.UpdateAgentCollaboratorResponse$ = UpdateAgentCollaboratorResponse$;
exports.UpdateAgentCommand = UpdateAgentCommand;
exports.UpdateAgentKnowledgeBase$ = UpdateAgentKnowledgeBase$;
exports.UpdateAgentKnowledgeBaseCommand = UpdateAgentKnowledgeBaseCommand;
exports.UpdateAgentKnowledgeBaseRequest$ = UpdateAgentKnowledgeBaseRequest$;
exports.UpdateAgentKnowledgeBaseResponse$ = UpdateAgentKnowledgeBaseResponse$;
exports.UpdateAgentRequest$ = UpdateAgentRequest$;
exports.UpdateAgentResponse$ = UpdateAgentResponse$;
exports.UpdateDataSource$ = UpdateDataSource$;
exports.UpdateDataSourceCommand = UpdateDataSourceCommand;
exports.UpdateDataSourceRequest$ = UpdateDataSourceRequest$;
exports.UpdateDataSourceResponse$ = UpdateDataSourceResponse$;
exports.UpdateFlow$ = UpdateFlow$;
exports.UpdateFlowAlias$ = UpdateFlowAlias$;
exports.UpdateFlowAliasCommand = UpdateFlowAliasCommand;
exports.UpdateFlowAliasRequest$ = UpdateFlowAliasRequest$;
exports.UpdateFlowAliasResponse$ = UpdateFlowAliasResponse$;
exports.UpdateFlowCommand = UpdateFlowCommand;
exports.UpdateFlowRequest$ = UpdateFlowRequest$;
exports.UpdateFlowResponse$ = UpdateFlowResponse$;
exports.UpdateKnowledgeBase$ = UpdateKnowledgeBase$;
exports.UpdateKnowledgeBaseCommand = UpdateKnowledgeBaseCommand;
exports.UpdateKnowledgeBaseRequest$ = UpdateKnowledgeBaseRequest$;
exports.UpdateKnowledgeBaseResponse$ = UpdateKnowledgeBaseResponse$;
exports.UpdatePrompt$ = UpdatePrompt$;
exports.UpdatePromptCommand = UpdatePromptCommand;
exports.UpdatePromptRequest$ = UpdatePromptRequest$;
exports.UpdatePromptResponse$ = UpdatePromptResponse$;
exports.UrlConfiguration$ = UrlConfiguration$;
exports.ValidateFlowDefinition$ = ValidateFlowDefinition$;
exports.ValidateFlowDefinitionCommand = ValidateFlowDefinitionCommand;
exports.ValidateFlowDefinitionRequest$ = ValidateFlowDefinitionRequest$;
exports.ValidateFlowDefinitionResponse$ = ValidateFlowDefinitionResponse$;
exports.ValidationException = ValidationException;
exports.ValidationException$ = ValidationException$;
exports.ValidationExceptionField$ = ValidationExceptionField$;
exports.VectorIngestionConfiguration$ = VectorIngestionConfiguration$;
exports.VectorKnowledgeBaseConfiguration$ = VectorKnowledgeBaseConfiguration$;
exports.VectorSearchBedrockRerankingConfiguration$ = VectorSearchBedrockRerankingConfiguration$;
exports.VectorSearchBedrockRerankingModelConfiguration$ = VectorSearchBedrockRerankingModelConfiguration$;
exports.VectorSearchRerankingConfiguration$ = VectorSearchRerankingConfiguration$;
exports.VectorSearchRerankingConfigurationType = VectorSearchRerankingConfigurationType;
exports.VideoConfiguration$ = VideoConfiguration$;
exports.VideoSegmentationConfiguration$ = VideoSegmentationConfiguration$;
exports.WebCrawlerConfiguration$ = WebCrawlerConfiguration$;
exports.WebCrawlerLimits$ = WebCrawlerLimits$;
exports.WebDataSourceConfiguration$ = WebDataSourceConfiguration$;
exports.WebScopeType = WebScopeType;
exports.WebSourceConfiguration$ = WebSourceConfiguration$;
exports.paginateListAgentActionGroups = paginateListAgentActionGroups;
exports.paginateListAgentAliases = paginateListAgentAliases;
exports.paginateListAgentCollaborators = paginateListAgentCollaborators;
exports.paginateListAgentKnowledgeBases = paginateListAgentKnowledgeBases;
exports.paginateListAgentVersions = paginateListAgentVersions;
exports.paginateListAgents = paginateListAgents;
exports.paginateListDataSources = paginateListDataSources;
exports.paginateListFlowAliases = paginateListFlowAliases;
exports.paginateListFlowVersions = paginateListFlowVersions;
exports.paginateListFlows = paginateListFlows;
exports.paginateListIngestionJobs = paginateListIngestionJobs;
exports.paginateListKnowledgeBaseDocuments = paginateListKnowledgeBaseDocuments;
exports.paginateListKnowledgeBases = paginateListKnowledgeBases;
exports.paginateListPrompts = paginateListPrompts;
